\chapter{Additional for Dissertation}
\thispagestyle{empty}
\emph{The last chapter containing some additional things needed for a complete 
dissertation document required for a promotion at the University of Groningen.}
\null
\vfill

\newpage

\section*{General Discussion}
\lipsum[1]
\newpage

\section*{Dutch Summary}
\addcontentsline{toc}{section}{Dutch Summary}
Systems genetics is the inter disciplinairy field which deals with the consequences of genetic 
variation on a biological system. The goal of systems genetics is to partition this variation 
into three major categories: Genetic, Environment and Error variation, and explain how complex
phenotypes arise from a combination of these three major factors.  Natural variation (or 
genetic pertubation) is used to interogate the genetic component of phenotype variation. 
Combined with environmental pertubation we can investigate the influence different 
environments and the interaction between genetics and environment. Experimental design and 
statistics are used to minimize and estimate error variance.

Pheno2Geno (Chapter 2) deals with the creation of genetic maps from large scale omics data. 
The theory of genetic map construction is ~100 years old. Software available for genetic map 
construction has not been adapted yet to make use of new technologies such as multi threading 
or cluster computing. Pheno2Geno aims to provide analysis of data from tilling arrays or RNAseq 
to generate gene based expression markers and create high density genetic maps.

Chapter 3 describes the implementation of the Multiple QTL mapping routine by R. C. Jansen into 
R/qtl. Adding a 'new' algorithm to the R/qtl toolset which aims to provide a range of QTL 
mapping tools for inbred crosses. R/qtl is the basis of a toolset build around a unified 
data structure allowing easy adaptation and extension of the software. R/qtl allows researchers 
to analyse data from different sources or to quickly compare different approaches.

In chapter 4 we show case our ideas for a generic storage and computation platform for systems 
genetics. Our demo system xQTL workbench is currently being used as a backend to the 
WormQTL and WormQTL-HD database. xQTL workbench allows users to store and share their data in 
a local or web environment, and run analysis across datasets using the power of distributed 
computing. It comes standard with QTL mapping tools such as: R/qtl, PLINK and qtlbim but also
provides a web inferfaces, data importers, APIs and visualizations.

Current work on using differences in correlation to generate interaction networks and detect 
cell type specific QTL effects is described in chapter 5. Correlated Traits Locus analysis 
(or CTL mapping) enables researchers to find genetic loci controlling correlation differences 
in segregating phenotypes. A variation on this method has proven valuable in discovering cell 
type specific eQTL effects. These effects can be used to untangle cell mixtures seen in whole 
blood gene expression data.

%In short this proposed method creates a proxy for cell counts in an unrelated cohort. this proxy 
%is then used during meta analysis as artifical cell count across many different cohorts. The 
%interaction effects between QTL and Proxy are used to assign a cell type to each eQTL. In the final 
%stage diseases in which cell type enrichment is prominent are tested by using publicly available data.

Last we have some additional required chapters, such as this summary in Dutch, a CV and publication 
list and bibliography.\\

I hope you enjoy reading this thesis as much as I enjoyed creating it during the last four years,\\\\

Danny Arends (Aug 2013)

\newpage
\section*{Abbreviations and Acronyms}
\addcontentsline{toc}{section}{Abbreviations and Acronyms}
{\footnotesize
\begin{tabular}{ l l }
API:         & Application Programming Interface\\
BC:          & Backcross \\
bp:          & Base pair(s) \\
cM:          & centi Morgan \\
CPNN:        & Collaborative computing project for NMR\\
CSV:         & Comma separated values\\
CTL:         & Correlated traits locus \\
DesignGG:    & Experimental design of genetical genomics software\\
DRY:         & Principle of don't repeat yourself\\
DSL:         & Domain Specific Language\\
EBI:         & European Bioinformatics Institute\\
FINDIS:      & Finish disease database\\
GEMs:        & Gene expression based genetic marker\\
GEN2PHEN:    & EU project to unify human and model organism genetic variation databases\\
GMOD:        & Generic model organism database project\\
GUI:         & Graphical user interface\\
GWAS:        & Genome Wide Association Study\\
GWL:         & Genome Wide Linkage analysis\\
HGVBaseG2P:  & Human genome variation database of genotype-to-phenotype information\\
HTML:        & Hypertext markup language\\
IDE:         & Integrated Development Environment\\
JAR:         & Java Software Archive\\
LGPL:        & Lesser general public license\\
MAGE-TAB:    & Microarray gene expression tab delimited file format\\
Mbp:         & Mega base pairs = 1.000.000 bp \\
MOLGENIS:    & Molecular genetics information systems toolkit\\
NordicDB:    & Nordic Control Cohort Database\\
OBF:         & Open Bioinformatics Foundation\\
OntoCAT:     & Ontology common API toolkit\\
PEAA:        & Patterns for enterprise application architecture\\
QTL:         & Quantitative trait locus\\
RDF:         & Resource description format\\
REST:        & Representative state transfer web services\\
RIL:         & Recombinant inbred line \\
SNP:         & Single Nucleotide Polymorphism\\
SOAP:        & Simple Object Access Protocol\\
SQL:         & Structured Query Language\\
UML:         & Uniform data Modeling Language\\
WAR:         & Web Application aRchive file\\
XML:         & Extensible Markup Language\\
XGAP:        & Extensible genotype and phenotype software platform. 
\end{tabular}
}
\newpage

\section*{Acknowledgements}
\addcontentsline{toc}{section}{Acknowledgements}
Knowledge is in the end based on acknowledgement\\
- Ludwig Wittgenstein (1889 - 1951)\\\\

There are many people who need to be thanked and acknowledged, 
here are the people who made the cut:

{\bf Ritsert C. Jansen} to whom I owe my life in science, thank you for giving 
me this chance to work in your group and develop my skills in research and 
education. {\bf Anna Mulder} my girlfriend for all the times I left you and 
{\bf Oscar} (our cat) for my science endavours. Thank you for your trust in 
me and giving me a home to come home to. {\bf Pjotr Prins} Always forgotten, Never 
Ignored. All the people past and present at the {\bf Groningen BioInformatics Centre} 
(GBIC) group for putting up with me. {\bf Yang Li} for your mentorship and all 
the vibrant discussions we had at GBIC, {\bf Joeri v/d Velde} my friend and coworker 
for all the good times hacking on Java and R in Haren. My student {\bf Konrad Zych} 
for his dedication and help with chapter \ref{chap:pheno2geno}. {\bf Morris Swertz} 
my co-promotor, you were a great help when writing chapter \ref{chap:xqtlwormbench} 
and always made me feel at home at GCC. Also all the guys from the {\bf Genomics Coordination 
Centre} (GCC) department for all the End of Sprints, presentations, discussions and beers. 
{\bf Lude Franke} and {\bf Harm-Jan Westra} for the very informative collaboration on the 
human GWA study (chapter \ref{sec:cellspecificeqtl}), I learned a LOT from you, and am 
extremely thankful for your help with the final chapters of this thesis. {\bf Wilko ligterink}, 
{\bf Henk Hilhort} and {\bf Ronny Joosen} for doing the \emph{Arabidopsis thaliana} 
experiments, and trusted me to analyze their data. I learned a lot especially about 
the more 'bio' parts of this thesis. The {\bf University of Groningen} (RUG) for 
educating me and providing me with an place to do my research. Finally the {\bf 
University Medical Centre Groningen} (UMCG) for the pleasant working atmosphere 
and good lunches.

\newpage

\section*{About the author}
\addcontentsline{toc}{section}{About the author}
\subsection*{Curriculum Vitae}
\addcontentsline{toc}{subsection}{Curriculum Vitae}

\begin{wrapfigure}{r}{0.5\textwidth}
  \centering
  \includegraphics[width=0.4\textwidth]{eps/image_6_1.eps}
\end{wrapfigure}

Danny Arends was born on the 15th of Juli 1983 in the city of Zwolle located in the heart of the 
Netherlands. After moving twice, Danny went to an elementary school situated in Heiligerlee, a minuscule 
hamlet in the north of Holland in the province of Groningen. The small school was a perfect 
match for small Danny, whom quickly fell in love with mathematics. With the advent of 
computers at the elementary school, also the love for computer science began to develop. 
Heiligerlee is located next to a small forrest called 'De Hoogte', this forrest was used 
by Danny and friends on a daily basis to build tree huts and wage war on kids from the other 
school in Heiligerlee using snow balls in winter. Combined with growing up amidst the animals 
on a small farm-like residence also sparked young Danny his intrest in biology 

After elementary school Danny went to the Ubbo Emmius lyceum in Stadskanaal (Groningen). 
The large scale VWO was a big change from the small elementary school. Combined with a 
long bus ride to and from school, made for long days. Fortunately new friends were made in 
the class room and during the long bus rides. Danny finished high school after 6 years with a 
Cum Laude, taking mostly exact courses such as mathematics, physics and chemistry.

At 17, Danny thought that a university degree should be the next step in his career.
Because of his interests in computers, Danny decided that computer science would be a good 
match. This turned out to be not so true. While deeply devoted to the computational machines, 
Danny was not satisfied by just studying the workings of a machine build by man.

After two years of Computer science at the University of Groningen, Danny decided it was time 
for a change. Computer science was replaced by Life Science \& Technologie, a bachelor which 
was recently formed as a collaboration between the Biology faculty and Medical Sciences.

He finished his bachelor in record tempo, partly due to the exemptions obtained from 
doing two years of computer science. A Master in molecular biology was quickly selected after
being introduced to bioinformatics at GBIC during a previous bachelor project. The molecular 
biology master allowed for customization of the courses followed, and bioinformatics became the main 
theme in all of the master theses produced. The first thesis: "Machine learning to predict 
transcriptional regulation in prokaryotes" was produced in the group of Oscar Kuipers. 
The second "R/QTL, MQM algorithm" was done in the lab of Ritsert C. Jansen.

Parts of this second master thesis are also in this thesis (chapter \ref{chap:mqm}).
After finishing his master Cum Laude, a PHD was started with Ritsert C. Jansen at the 
Groningen Bioinformatics Centre with a focus on the use of bioinformatic tools to handle 
current challenges in genetics and statistics. The four years of research at GBIC have 
lead to the thesis you are currently reading.\\\\

\newpage

\subsection*{List of Publications}
\addcontentsline{toc}{subsection}{List of Publications}

\subsection*{Authored:}
   R/qtl: high throughput Multiple QTL mapping\\
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  \bold{Bioinformatics} (2010)\\\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, 
           Henk Hilhorst and Ritsert C. Jansen}\\
  \bold{Plant Physiology} (2011)\\\\
  xQTL workbench: a scalable web environment for multi-level QTL analysis\\
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, 
           Steffen Moller, Ritsert C. Jansen and Morris A. Swertz}\\
  \bold{Bioinformatics} (2012)\\\\
  WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp\\
  \authors{L. Basten Snoek*, K. Joeri Van der Velde*, Danny Arends*, Yang Li*, 
           Antje Beyer, Mark Elvin, Jasmin Fisher, Alex Hajnal, Michael O 
           Hengartner, Gino B. Poulin, Miriam Rodriguez, Tobias Schmid, 
           Sabine Schrimpf, Feng Xue, Ritsert C. Jansen, Jan E. Kammenga 
           and Morris A. Swertz}\\
  \bold{Nucleic Acids Research} (2012)\\\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds 
  using Generalized Genetical Genomics\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J. B. Keurentjes, Wilco Ligterink, 
           Ritsert C. Jansen, Henk Hilhorst}\\
  \bold{Plant Physiology} (2013)

\subsection*{Co-Authored:}
  The MOLGENIS toolkit: rapid prototyping of biosoftware at the push of a button\\
  \authors{Morris A Swertz, Martijn Dijkstra, Tomasz Adamusiak,  Joeri K van der Velde, 
           Alexandros Kanterakis, Erik T. Roos, Joris Lops, Gudmundur A. Thorisson, 
           Danny Arends, George Byelas, Juha Muilu, Anthony J. Brookes, Engbert O. de Brock, 
           Ritsert C Jansen and Helen Parkinson}\\
  \bold{BMC Bioinformatics} (2010)\\\\
  XGAP: a uniform and extensible data model and software platform for genotype and phenotype experiments\\
  \authors{Morris A Swertz, K. Joeri van der Velde, Bruno M Tesson, Richard A Scheltema, 
           Danny Arends, Gonzalo Vera, Rudi Alberts, Martijn Dijkstra, Paul Schofield, 
           Klaus Schughart, John M. Hancock, Damian Smedley, Katy Wolstencroft, Carole 
           Goble, Engbert O. de Brock, Andrew R Jones, Helen E Parkinson and Ritsert C Jansen}\\
  \bold{Genome Biology} (2010)\\\\
  SYSGENET: a meeting report from a new European network for systems genetics\\
  \authors{Klaus Schughart, Danny Arends, P. Andreux, R. Balling, Pjotr Prins, et al.}\\
  \bold{Mammalian Genome} (2010)\\\\
  Trans-eQTLs Reveal that Independent Genetic Variants Associated With a Complex Phenotype Converge on 
  Intermediate Genes, with a Major Role for the HLA\\
  \authors{Rudolf SN Fehrmann, Ritsert C. Jansen, Jan H. Veldink, Harm-Jan Westra, Danny Arends,
           Marc Jan Bonder, Jingyuan Fu, Patrick Deelen, Harry J. M. Groen, Asia Smolonska, 
           Rinse K. Weersma, Robert M. W. Hofstra, Wim A. Buurman, ... , Lude Franke}\\
  \bold{Plos Genetics} (2011)\\\\
  Bioinformatics tools and database resources for systems genetics analysis in mice - a short review 
  and an evaluation of future needs\\
  \authors{Caroline Durrant, Morris A. Swertz, Rudi Alberts, Danny Arends, Steffen MÃ¶ller, 
           Richard Mott, Pjotr Prins, K. Joeri van der Velde, Ritsert C. Jansen and 
           Klaus Schughart}\\
  \bold{Briefings in Bioinformatics} (2011)\\\\
  WormQTLHD - a web database for linking human disease to natural variation data in C. elegans\\
  \authors{K. Joeri van der Velde*, Mark de Haan, Konrad Zych, Danny Arends, L. Basten Snoek, 
           Jan E. Kammenga, Ritsert C. Jansen, Morris A. Swertz and Yang Li}\\
  \bold{Nucleic Acids Research} (2014)

\subsection*{Under Review / In Press:}
  Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes\\
  \authors{Konrad Zych, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink, Ritsert C Jansen 
           and Danny Arends}\\
  \bold{BMC Bioinformatics} (201X)\\\\
  TiQS: web environment for expression QTL analysis\\
  \authors{Steffen M\"oller, Ren\'e Sch\"onfelder, Hajo Krabbenh\"oft, Benedikt Bauer, Yask Gupta, 
           Pjotr Prins, Danny Arends, et al.}\\
  \bold{BMC Bioinformatics} (201X)\\\\
  Multiple QTL mapping of cardiac collagen deposition in an F2 population of Scn5a mutant mice reveals 
  interaction between Fgf1 and Pdlim3, Gpr158 \& Itga6\\
  \authors{Elisabeth M. Lodder, Brendon P. Scicluna, L. Beekman, Danny Arends, et al.}\\
  \bold{Genome Research} (201X)

\subsection*{In Preparation:}
  Correlated Traits Locus mapping\\
  \authors{Danny Arends, Pjotr Prins, Harm-Jan Westra, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  \bold{Unknown} (201X)

\subsection*{Acknowledged in:}
  DesignGG: an R-package and web tool for the optimal design of genetical genomics experiments\\
  \authors{Yang Li, Morris A Swertz, Gonzalo Vera, Jingyuan Fu, Rainer Breitling and Ritsert C Jansen}\\
  \bold{BMC Bioinformatics}, 10:188 (2009)\\\\
  Generalizing genetical genomics: getting added value from environmental perturbation\\
  \authors{Yang Li, Rainer Breitling and Ritsert C. Jansen}\\
  \bold{Trends in Genetics}, 24:518-524 (2008)

\subsection*{List of Presentations}
\addcontentsline{toc}{subsection}{List of Presentations}
R/xqtl: High throughput modeling, mapping and exploration of Big Data\\
SYSGENET meeting - Braunschweig, April 2010\\\\
Introduction into QTL analysis\\
Dynamic Presentation - University of Groningen, Aug 2010\\\\
MQM and HPC for R/qtl\\
CSBG meeting - Wageningen University, Sept 2010\\\\
Introduction into QTL mapping - Bioinformatics I\\
University of Groningen, June 2011\\\\
(Re)Construction of genetic maps from gene expression data\\
GBIC - University of Groningen, July 2011\\\\
R/qtl for Big Data\\
MIT Department of Biology, invited by Jeroen PJ Saeij\\
MIT, Boston (MA), May 2011\\\\
The Challenge of Big Data Genetical Genomics\\
NCSA, invited by Victor Jongeneel and Chris Fields\\
NCSA, Urbana (IL), May 2011\\\\
Introduction into QTL mapping - Learning From nature (LFN)\\
Wageningen University, Feb 2012\\\\
Computer practical / tutorial - Learning From nature (LFN)\\
Wageningen University, Feb 2012\\\\
Introduction into R and R/qtl\\
Summer Course - King's college, Sept 2013\\\\

\subsection*{List of Posters}
\addcontentsline{toc}{subsection}{List of Posters}
User friendly cluster computing for QTL analysis\\
  \authors{Danny Arends, Joeri v/d velde}\\
  \bold{ISMB} July 2010 - Boston, USA\\\\
Multiple QTL mapping poster\\
  \authors{Danny Arends, Pjotr Prins,Karl W. Broman and Ritsert C. Jansen}\\
  \bold{GBIC day} September 2010 - Groningen, The Netherlands\\\\
Pheno2Geno poster\\
  \authors{Konrad Zych, Danny Arends, Ritsert C. Jansen}\\
  \bold{NBIC} April 2012 - Lunteren, The Netherlands\\\\
Pheno2Geno poster\\
  \authors{Konrad Zych, Danny Arends, Ritsert C. Jansen}\\
  \bold{ECCB / ESCS} Sept 2012 - Basel, Swiss\\\\
GWAS in potato\\
  \authors{Konrad Zych, Danny Arends, Ritsert C. Jansen}\\
  \bold{Kings College} Sept 2013 - London, UK

\subsection*{Awards}
\addcontentsline{toc}{subsection}{Awards}
1st place Poster Award 'Pheno2Geno' at ESCS2012 Basel (Swiss) (2012)\\
Travel Grant by NBIC for ISMB in Boston (MA, USA) (2010)\\
2nd place Poster Award xQTL workbench NBIC Conference (Lunteren) (2010)\\
Travel Grant 'Short course on system genetics' in Bar Harbor (MA, USA) (2009) by JAX laboratory\\
