\section{Metabolites in a DesignGG experiment}
A complex phenotype such as seed germination is the resultant of several genetic and environmental 
cues and requires the concerted action of many genes. The use of well-structured recombinant inbred 
lines in combination with omics analysis can help to disentangle the genetic basis of such 
quantitative traits. This so called genetical genomics approach can effectively capture both 
genetic (G) and epistatic interactions (G:G). However, to understand how the environment interacts 
with genomic encoded information (G:E) a better understanding of the perception and processing of 
environmental signals is needed. In a classical genetical genomics setup this requires replication 
of the whole experiment in different environmental conditions. A novel generalized setup overcomes 
this limitation and includes environmental perturbation within a single experimental design. 

We developed a dedicated QTL mapping procedure to implement this approach and used existing 
phenotypical data to demonstrate its power. Additionally, we studied the genetic regulation of 
primary metabolism in dry and imbibed Arabidopsis seeds. Many changes were observed in the 
metabolome which are both under environmental and genetic control and their interactions. 
This concept offers unique reduction of experimental load with minimal compromise of statistical 
power and is of great potential in the field of systems genetics which requires a broad 
understanding of both plasticity and dynamic regulation.

\subsection{Introduction}
The use of natural variation to disentangle the genetic (G) mechanisms underlying phenotypic differences 
has been very successful both in crop plants and in the model plant Arabidopsis (Arabidopsis thaliana; 
\cite{Alonso-Blanco:2009}. Most of the variation within wild or domesticated plant species is of 
quantitative nature determined by G polymorphisms at multiple loci. Such quantitative trait loci (QTL) 
can beanalyzed efficiently using experimental mapping populations such as recombinant inbred lines (RILs)
derived from directed crosses. Nowadays, many wellstructured RIL populations are available, often 
accompanied with detailed studies of phenotypic variation (Mitchell-Olds and Schmitt, 2006). The complexity 
ofquantitative traits is further determined by the interactions between genomic loci (i.e. epistasis) and 
between the genotype and the environment (genetic X environmental [G:E]). While epistasis can be effectively
identified in QTL analyses, albeit with lower power than main effects, the detection of G:E interactions 
requires experimentation in multiple conditions of interest. Because of the large population sizes often 
needed to obtain sufficient statistical power for QTL detection, G:E interactions are usually ignored in 
experimental setups. However, a better understanding of the perception and processing of environmental (E)
signals is greatly needed, because interactions provide important insights in adaptation mechanisms and
evolutionary constraints such as balancing and disruptive selection. To obtain a more detailed view of the
molecular mechanisms underlying phenotypic variation, genetical genomics studies, in which molecular traits
are genetically analyzed, have been successfully applied to enhance a directed strategy to identify causal
relationships (Kliebens tein et al., 2006; Keurentjes et al., 2007a; van Leeuwen et al., 2007; 
Wentzell et al., 2007; West et al., 2007; Rowe et al., 2008). The observed phenotype is often the resultant 
of a functional cascade of gene transcription followed by protein translation and modification, which 
finally leads to a highly dynamic metabolome underlying emergent properties (Kooke and Keurentjes, 2011). 
With the technological advances made in genomic analytical platforms, such as transcriptomics, proteomics, 
and metabolomics, the large-scale, high-throughput analyses needed for quantitative G approaches have 
become feasible (Jansen and Nap, 2001; Keurentjes et al., 2008). 

Incorporating developmental and E  perturbation in the often expensive and laborious omic analyses, an 
alternative experimental setup, coined generalized genetical genomics (GGG), using balanced fractions 
of a RIL population has been proposed (Li et al., 2008). It provides a cost-effective experimental setup 
for hypothesis-generating research in multiple environments. Such an approach aims for the creation of 
subpopulations of RILs, one for each environment to be tested, with an optimal distribution of parental 
alleles over all available markers (Li et al., 2009). When these subpopulations are subjected to E 
perturbation, the emerging phenotypes can be explained by several sources of variation: G variation, 
E variation, and G:E variation. Whenever the resulting phenotype is not or only mildly affected by E 
interactions (G:E), the analysis of the different subpopulations can be combined, gaining the full power 
of a complete population. However, when a trait shows strong G:E interaction (e.g. those that only express G variation in specific 
environments), the power to detect QTL is dependent on those subpopulations expressing the G variation. 
Although G:E interactions have been detected previously in genetical genomics studies for expression 
(Li et al., 2006; Smith and Kruglyak, 2008; Gerrits et al., 2009; Yeung et al., 2011) and metabolite 
content (Zhu et al., 2012) by analyzing all lines in a population under different environments, the 
GGG concept offers an effective way of studying a combination of G and E perturbations and is of great 
potential in the field of systems genetics, in which a broad understanding of both plasticity and
dynamics is required (Li et al., 2008). The fundamental basis of the experimental design and data 
analysis using a full model ($Y = E + G + G:E + e$), where $Y$ is the observed phenotype and $e$
is residual error, is generally valid and frequently used (Churchill, 2002; Li et al., 2006; Gerrits
et al., 2009). As a proof of principle, we present experimental data on the G regulation of primary 
metabolism in dry and imbibed Arabidopsis seeds using a GGG design and discuss the application and 
implications of such a strategy.

Plants are extremely rich in biochemical compounds, and major roles in plant development, adaptation, 
and defense have been identified for biosynthesis pathways and their products (Binder, 2010). The 
biosynthetic pathways of primary metabolites are well studied and often well conserved between 
different taxa (Peregr√≠n-Alvarez et al., 2009). Nonetheless, quantitative variation for many of 
these compounds can be observed between natural variants, which might be reflected in their different 
growth characteristics. The

