\chapter{Introduction}
\thispagestyle{empty}
\label{chap:introduction}

\emph{This first chapter introduces the reader to concepts such as gene, chromosome 
and DNA all the way up to Multiple QTL mapping (MQM) and Genome Wide Association Studies 
(GWAS). }

\null
\vfill
\newpage

\section{Introduction}
Systems genetics is the interdisciplinary field which deals with the effects of perturbation 
and natural variation measured at a all possible levels of a biological system. It takes a 
more holistic approach to understanding biological systems. In the last 20 years our ability 
to take large scale measurements from the transcriptome, metabolome and proteome have 
improved dramatically leading to a virtual avalanche of biological data.

Phenotype variation / diversity observed is the consequence of a mixture of variation 
originating at the DNA level (DNA), the environment (Env), and an error component (Err).

$$ Phenotype = Env + DNA + DNA * Env + Err $$

Partitioning of observed variation into this formula is of critical importance 
to understanding the contribution of genetics and environment to the biological 
processes observed. Understanding how a system works and reacts to different 
environments, allows modification of the system to better suit our needs or 
environmental requirements. Knowledge about DNA variation allows to optimize 
breeding of animals and plants, explore the genetic basis of human diseases, 
improved production of chemicals, and so forth.

Variation observed at the DNA level can be measured using genotyping (AFLP, RFLP, 
SNPs, Sequencing) techniques, allowing to create detailed maps of DNA inheritance 
within a population and track the origin of DNA through many generation.

At the phenotype end of the spectrum new high throughput technologies have also 
been developed to mass produce phenotypes measurements using automated phenotyping. 
Automated phenotyping allow thousands of simultaneous measurements at all the 
different endophenotype levels such as: metabolome, proteome, transcriptome. These 
new technologies such as tiling arrays, RNA-seq, High throughput proteomics and 
metabolomics now enable us to measure all levels from the DNA to the observed 
phenotype at minimal costs.

How the activity of DNA is regulated by signals from all molecular levels and modified by 
signals from the environment is the focus of systems genetics. The use of computational 
tools in understanding how genetics and environment intertwine is the central in the 
field of systems genetics to deal with the large amounts of phenotype and genotype data.

Variation observed at levels 'further away' from the genome such as: transcriptome, 
proteome, metabolome, all the way up to classical phenotypes form a complex interplay 
between DNA and environment. This interplay has proven to be challenging to unravel 
on a genome-wide scale. The following chapters will go into our solutions to these issues, 
infrastructure created, methodologies developed, and their application in current 
research. First some (historical) background to provide a solid foundation  and outline
the context of the following chapters. 

This chapter is by no means meant as an exhaustive account of the history of systems genetics, 
only a short introduction to highlight previous work which is of use to this thesis.

\section{History (1800-1930)}

The field of genetics is a relatively old field, and is founded when Gregor 
Mendel publishes his work on the inheritance of phenotypes in pea plants. 
His work described in 'Versuche \"uber Pflanzen-Hybriden' in 1865-1866 
\cite{Mendel:1866} gives us Mendel's Laws of Inheritance. He observed that 
crossing two plants with different colors of flowers leads to an offspring 
population with predictable color ratios. He postulated the idea of heredity 
units and stated that each individual carries two of these resulting in a 
single phenotype. Each heritable units is received from one of the parents, 
but which one of the two units from the parents gets passed to a child is random 
(Law 1 - Segregation).

These two units are not equal, a dominant unit (color 1) overruled another 
unit (color 2 - recessive). Mendel studied more phenotypes in his pea plants 
besides color. Now it is known that Mendel was fortunate to select 
phenotypes which are caused by a single genes and are inherited independently. 
Mendel also observed this when comparing inheritance of multiple phenotypes. 
He concluded that a unit of inheritance is passed from parent to child in an 
independent fashion compared to other units of inheritance 
(Law 2 - Independent Assortment).

Mendel's work was not fully recognized until 30 years later, when his work was 
rediscovered by Hugo de Vries, Carl Correns en Erich von Tschermak who (re) defined the 
rules for Mendelian Genetics \cite{deVries:1889} providing biologist for the first 
time with a mathematical framework to study heritability of phenotypes caused by a 
single genetic unit.

Twenty years after the rediscovery of Mendel's Laws in 1913, careful observations of 
Thomas Hunt Morgan introduced an addition to Mendel's theoretical work. He observed 
that some phenotypes in his \emph{Drosophila melanogaster} mutant flies are inherited 
together, in conflict with Mendel's Laws (which state independent inheritance of 
genetic units). He called this phenomenon genetic linkage and when phenotypes are 
linked to the sex of the animal sex based inheritance \cite{Morgan:1915, Morgan:1916}.

\begin{wrapfigure}{r}{0.5\textwidth}
  \centering
  \includegraphics[width=0.4\textwidth]{eps/image_1_2.eps}
  \caption[Example of pedigree based linkage analysis.]
    {Example of a hypothetical pedigree on which linkage analysis can be performed. Here we show a 
    hypothetical pedigree of individuals within a family (2 generations), squares represent 
    male individuals while circle are females. The phenotype is shown encoded by the fill 
    color of the shape (Black = affected, White = not affected).  In this example pedigree, 
    the "A" allele segregates with the disease. It is shared identical-by-descent in all 
    affected individuals. }
    \label{fig:pedigree}
\end{wrapfigure}

This concept is also commonly explained as beads on a string: in which multiple genetic 
units linked together. Two beads close together are tightly linked are and almost always 
inherited together. While two beads far apart are less linked and can be separated by 
meiosis. Different strands are independently inherited allowing for Mendelian 
inheritance by two beads on two different strings.

Sturtevant, a student of Morgan used his mentors linkage theory between phenotypes 
as a distance measurement between units of inheritance. He created the first genetic 
map of \emph{Drosophila melanogaster} 40 years before the discovery of the DNA molecule \cite{Morgan:1915}, 
while molecular mechanisms were still unknown. Sturtevant genetic map, based on 
phenotypes, allowed genetics based on careful observation of segregating phenotypes in 
large populations of flies, and then determining their distance relative to other 
phenotypes.

In 1918 Ronald Aylmer Fisher published a paper with the title 'The Correlation Between 
Relatives on the Supposition of Mendelian Inheritance' in this paper he showed that a 
large number of small effects from discrete genetic units would additively sum up to 
the continuous phenotypes observed \cite{Fisher:1918}. With this observation we move 
to the next section, in which a book by R. Fischer will again mean a major jump in 
the development of population genetics.

\section{DNA and QTLs (1930-1990)}
In 1930 the book 'The Genetical Theory of Natural Selection' \cite{Fisher:1930} by R. 
Fisher was published. The book describes how Mendelian genetics is consistent with the 
idea of evolution by natural selection. This removes one of the last barrier for large 
scale adoption of Mendel's theories in biology.

The discovery of DNA as the carrier of heritability in 1944 by Avery, MacLeod and McCarty 
\cite{Avery:1944} and the discovery of its structure by Watson and Crick.in 1953 
\cite{Watson:1953} was the long sought after molecular basis for genetics. After the 
discovery of restriction enzymes by Luria and Human \cite{Luria:1952} it was possible 
to cut and paste lengths of DNA. This allowed to experimentally work with DNA and 
investigate the different (lengths of) fragments produced. In 1970 the first techniques 
were developed to sequence DNA. This was first done by two-dimensional chromatography, 
and later using fluorescence based sequencing methods \cite{Pettersson:2009}.

In the end of the 1980s everything was there for the next big step in genetics. Three 
papers were published which detailed the use of RFLP linkage maps to localize genes 
responsible for variation in quantitative phenotypes. These three papers by David Botstein 
and Eric Lander \cite{Lander:1986, Lander:1987, Lander:1989} form the basis for modern 
day linkage analysis and genome wide association studies. 

\begin{wrapfigure}{r}{0.5\textwidth}
  \centering
  \includegraphics[width=0.5\textwidth]{eps/image_1_1}
  \caption[Effect scan across the genome.]
    {An example of how effect scans are performed using a single phenotype measured on 9 individuals (colored numbers)
    and genotypes represented by rows in the matrix (A or B). 20 markers (columns) were assessed: A and B are genetic 
    variants obtained from either parental or maternal line. }
    \label{fig:effectscan}
\end{wrapfigure}

In linkage analysis, DNA is tracked from parents to children. This allows to build so called 
genetic family trees or pedigrees (See fig. \ref{fig:pedigree}) in which segregation of a phenotype (or 
disease) is displayed. If we find a genetic region which co-segregates with the disease we can assume this DNA and 
disease are linked, and can asses the strength of the linkage observed \cite{Rosyara:2009}. By using experimental 
crosses derived from inbred parents the need to build up complex pedigrees is avoided, while 
retaining the advantage of being able to trace back the DNA to the founder strains. Also inbred 
populations provide high statistical power at the cost of reduced resolution for QTL mapping \cite{Jansen:2001a}. 
This reduction in resolution is caused by the lack of recombinations in populations which are 
generally small (20 to 1000 individuals).

Association analysis is done in outbred populations, where tracking of the origin of the 
DNA is hard or impossible. However molecular markers still allows to get information 
about the underlying DNA. We can use \emph{e.g.} Single Nucleotide Polymorphisms (SNPs) and associate 
a phenotype with this genetic marker \cite{Mehta:2013}. The advantage of this approach is that resolution is 
very high (around 200 - 300 kb in humans \cite{HapMap:2005}). This comes from the fact that outbred populations 
have a high number of recombinations compared to inbred populations. However because of the large amount of 
markers statistical power to detect effects are limited because of multiple testing problems.

Heritable phenotypes could be mapped to genomic locations using a combination of DNA 
restriction enzymes, Mendel's inheritance laws and Hunt's linkage theory. Together these 
methodologies and theories provide the experimental and statistical background to 
analyze heritability in any population. An example of a basic effect scan is shown in Figure 
\ref{fig:effectscan}. here at each marker the mean of the phenotype for AA and BB are 
calculated, and the difference plotted per marker. This gives us an initial idea which markers 
might be interesting bio markers for our phenotype. 

Linkage and Association analysis are still the foundation of population genetics research 
today. More sophisticated tools and algorithms have been developed and are used, but the 
basic theory remains the same. Some of these more advanced methods are discussed in the next 
section to sketch a background for the work presented in this thesis.

\section{From Phenotypes to Genetical Omics and GWAS (1990-2010)}

Observed phenotype expression is often more complex than a single causative gene \cite{Sinha:2006, 
West:2007}. To model these more complex interactions between multiple regulators on e.g. gene expression 
extension of the basic model for QTL mapping is necessary. 

Extending the model can be done by incorporating sources of variation as co-factors. This allows us 
to associate / partition the observed variance to either environmental factors (E) or genetic loci 
(G). To include additional genetic components into the model, and then scanning for QTLs conditional 
on these other genetic effects Multiple QTL Mapping (MQM) can be used. MQM belongs to a family 
of QTL mapping methods, that include Haley-Knott regression \cite{Haley:1992} and composite 
interval mapping (CIM) \cite{Zeng:1994}. MQM combines the strengths of generalized linear model 
regression with those of interval mapping \cite{Jansen:1993, Jansen:1994b}. 

In the beginning of the QTL mapping period genetic maps were of poor quality, with a lot of missing 
data and large gaps between markers. To overcome this problem interval mapping was developed. Interval 
mapping uses prior knowledge about linkage in inbred populations to map QTLs between genetic markers. 
Using hidden Markov models (HMMs) interval mapping estimates the unobserved genotypes in the population 
based on the surrounding genotype observations improving QTL mapping resolution \cite{Jansen:1993, Zeng:1994}.

\begin{table}[h]
  \centering
  {\footnotesize
  \begin{tabular}{ | c | l | l | }
    \hline
    {\bf Molecular level} & {\bf Molecule} & {\bf Technology}\\
    \hline
    \hline
\rowcolor{gray!35}    Genome          & DNA                & RFLP \cite{Lander:1986} \\
\rowcolor{gray!35}    Genome          & DNA                & SNP chips \cite{Hacia:1999} \\
\rowcolor{gray!35}    Genome          & DNA                & DNA sequencing \cite{Mardis:2008} \\
    \hline
    EpiGenome       & DNA methylation    & Bisulfite sequencing \cite{Hayatsu:2007} \\
    EpiGenome       & DNA methylation    & ChIP-on-chip \cite{Collas:2010} \\
    EpiGenome       & DNA methylation    & ChIP-Seq \cite{Park:2009} \\
    \hline
    \hline
\rowcolor{gray!35}    Transcriptome   & RNA          & Microarray \cite{Lashkari:1997}, Tiling array \cite{Lee:2013} \\
\rowcolor{gray!35}    Transcriptome   & RNA          & RNA-Seq \cite{Wang:2009}\\
    \hline
    Proteome        & Proteins     & 2D gel electrophoresis \cite{O'Farrell:1975}\\
    Proteome        & Proteins     & Mass Spectometry \cite{Deshaies:2001}\\
    Proteome        & Proteins     & Antibody protein chip \cite{Fasolo:2009} \\
    \hline
\rowcolor{gray!35}    Metabolome      & Metabolites  & Mass Spectometry \cite{Aebersold:2003} \\
\rowcolor{gray!35}    Metabolome      & Metabolites  & Nuclear magnetic resonance \cite{Espina:2009} \\
    \hline
  \end{tabular}
  }
  \caption[Overview]{Overview of current technologies to interrogate the molecular basis for life.}
    \label{tbl:overview}
\end{table}

Genetical Genomics is the concept that views endophenotypes (RNA, Protein and Metabolite abundance) 
as common phenotypes which can be mapped in bulk to the genome similar to classical 
phenotypes \cite{Jansen:2001a}. Using natural occurring variation and new omics tools allows us 
to track variation from the genotype all the way up to the classical phenotypes (see Table 
\ref{tbl:overview} for an short overview of methods). Tracing the effects of variation from 
genome to phenotype allows genetics to go from individual QTLs to a system wide approach of 
analyzing QTLs at all known molecular levels called Systems Genetics \cite{Threadgill:2006, Nadeau:2011}.
Studies in model organisms have shown high heritability's for endophenotypes such as gene 
expression making these phenotypes ideal targets for QTL mapping \cite{Brem:2002, Yvert:2003, Morley:2004}. 

In 2003 R/qtl was developed to provide reference implementations for QTL mapping. R/qtl is 
an extensible, interactive environment to map quantitative trait loci (QTL) in experimental 
crosses. It is implemented as an add-on package for the freely available and widely used 
statistical language/software R \cite{R:2009}. The main focus of R/qtl is to provide 
the mouse community with different QTL mapping methodologies, and allow to deal with the 
the aberrant segregation X chromosome. Furthermore it supports different types of inbred 
populations such as backcross (BC), F2, Recombinant inbred lines (RIL) and 4-way RILs. \cite{Broman:2003}

When mapping gene expression or protein abundance current knowledge of protein and DNA sequence 
allows us to locate their template on the genome. When QTL mapping resolution allows we can even 
distinguish between traits mapping in the proximity of their respective gene (\emph{cis}-eQTL) 
or to other regions in the genome (\emph{trans}-eQTL). This information can be summarized into 
so called cis-trans plots, where the X-axis is the location of the eQTL and the Y-axis the genetic 
location of the trait. Often so called trans-bands are observed, hotspots of many \emph{trans}-
eQTL mapping to a common region in the genome \cite{Breitling:2008a}, which are used to infer 
biological meaning and reconstruction co-expression and/or co-regulatory networks.

In the last decade Genome Wide Association Study (GWAS) have identified thousands of genetic 
variants that are associated with human disease\cite{Hindorff:2009}. For reliable results GWAS 
needs a large cohort of genotyped and phenotyped individuals. Large consortia are working 
together to gather large amounts of human expression data from many different tissues. This 
data is then used in meta analysis leading to eQTL GWA studies with even larger sizes 
(5000+ individuals) \cite{Lude:2011}, leading to more reliable results and enabling discovery 
of new modifiers of human gene expression.

It is known that many factors, such as effects on intermediate molecular phenotypes, influence 
the relationship between genotype and the eventual development of disease. It has also been 
observation that many of the disease-predisposing variant are non-coding, which suggests that 
these variants have a regulatory function. Furthermore it has been shown that many disease-predisposing variants 
(\emph{e.g.} single nucleotide polymorphisms (SNPs)) affect the expression of nearby genes (\emph{i.e.} 
\emph{cis}-eQTLs) \cite{Powell:2012, Lude:2011, Zeller:2010}. While this could be due to sequence 
polymorphisms 

\begin{figure}[h!]
 \centering
    \includegraphics[width=0.6\textwidth]{eps/image_1_3}
  \caption[ThesisLayout.]
    {Schematic figure showing to which molecular level each chapter is related. Each molecular 
    level here depicted are drawn from genome (bottom) to phenome (top), arrows indicate 
    how genetic information is transferred from level to the level as by the current by 
    the biological 'dogma'.}
    \label{fig:layout}
\end{figure}

\section{Thesis contributions (2010-2014)}
The following chapters will highlight the contributions made to the field of life 
sciences during the 4 years of my PHD research at the University of Groningen. A schema 
can be found in Figure \ref{fig:layout}.

Chapter \ref{chap:pheno2geno} details how Pheno2Geno an R package was developed for 
the high-throughput generation of genetic markers and maps from molecular phenotypes. 
Pheno2Geno selects suitable phenotypes that show clear differential expression in the founders. Pheno2Geno 
uses mixture modeling to select phenotypes showing segregation ratios close to the 
expected Mendelian segregation ratios and transform them into genetic markers suitable 
for map construction and/or saturation. Pheno2Geno analyzes the candidate genetic 
markers and excludes those showing multiple QTL, epistatically interacting QTL, and QTL 
by environment interactions to provide a set of robust markers for QTL mapping protecting 
against genetic markers from a non genetic origin.

Chapter \ref{chap:mqm} highlights the integration of the Multiple QTL mapping algorithm 
into R/qtl. In sections (\ref{chap:mqm}.1 and  \ref{chap:mqm}.2) we show the performance 
of MQM on experimental data from a cross of \emph{A. thaliana} Bayreuth x Shahdara. We 
focus on the developing seeds in this cross and try to find genetic factors underlying 
changes we observed in the phenome and metabolome.

Chapter \ref{chap:ctlmapping} -  Shows our current work on understanding differences in 
correlation observed when mapping two traits onto the genome. We show that there is a high 
overlap between CTL mapping and using an G:E interaction model, and use this interaction 
model in data form a human genome wide association study (GWAS) to detect cell type specific 
eQTL in whole blood. We show that low effect QTLs observed in samples can be explained by 
compensating for the relative number of neutrophils observed, or predicted. Improving our 
power to detect QTLs and enabling us to assign cell type labels to observed \emph{cis}-eQTL 
effects.

Chapter \ref{chap:xqtlwormbench} - Details our work to provide infrastructure for the Life 
Sciences. We advocate the use of Generators to create software and propose a data model (XGAP) 
to store phenotype and genotype data. Combining these two approaches we developed: xQTL 
workbench a scalable web platform for the mapping of quantitative trait loci (QTLs) at 
multiple levels such as gene expression (eQTL), protein abundance (pQTL), metabolite 
abundance (mQTL) and phenotype (phQTL) data. Popular QTL mapping methods for model organism 
and human populations are accessible via the web user interface. Large calculations scale 
easily on to multi-core computers, clusters and cloud computing.

Well as you can see enough to cover, we start by improving the genetic map for our model organism 
\emph{A. thaliana}. We introduce Multiple QTL mapping (MQM), which we use to find genetic loci 
involved in seed quality traits. We expand our methodology by looking at different environments 
and realize cell type is just another factors which can be handled in a similar way as environments.

