\documentclass[8pt, twoside, a5paper]{report}
\usepackage[usenames, dvipsnames, svgnames]{xcolor}
\usepackage{titlesec, blindtext, color, graphicx}
\usepackage[light,condensed,math]{kurier}
\usepackage[T1]{fontenc}
\usepackage{tcolorbox}
\usepackage{lipsum}
\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}
\definecolor{GREEN}{rgb}{0, 0.5, 0.1}

%\usepackage{lmodern} % to remove warnings of size substitution
\usepackage[top=0.6in, bottom=0.6in, left=0.6in, right=0.4in]{geometry}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}

\newenvironment{myexampleblock}[1]{%
    \tcolorbox[beamer,%
    noparskip,breakable,
    colback=LightGreen,colframe=DarkGreen,%
    colbacklower=LimeGreen!75!LightGreen,%
    title=#1]}%
    {\endtcolorbox}

\newcommand{\hsp}{\hspace{7pt}}
\newcommand{\hssp}{\hspace{3pt}}

\titleformat{\chapter}[hang]{\vspace{-2 cm}\huge\bfseries}{{\fontsize{50}{60}\selectfont \thechapter \hsp\textcolor{GREEN}{|}\hsp}}{0pt}{\huge\bfseries}
\titleformat{\section}[hang]{\large}{\thesection\hssp}{0pt}{\large\bfseries}

\newcommand{\authors}[1]{\small{#1}}
\newcommand{\italic}[1]{\textit{#1}}
\newcommand{\bold}[1]{{\bfseries #1}}
\newcommand{\code}[1]{{\tt #1}}
\newcommand{\mytitle}[1]{{\LARGE #1}}

\pagestyle{headings}
\renewcommand{\contentsname}{Table of content\vspace{-30pt}}

\begin{document}
\pagenumbering{arabic}
  \thispagestyle{empty}
  \begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
  \end{center}
\newpage
  \thispagestyle{empty}
  \noindent The work described in this thesis was carried out at the Groningen Bioinformatics Centre, University of Groningen, The Netherlands.
  This research was financially supported by\\
  \vspace{130 mm}
  
  \noindent (c) 2013 by GBIC - Danny Arends\\
  Printed by:\\
  ISBN:\\
\newpage

\thispagestyle{empty}
\begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
    \vspace{10 mm}
  \bold{Proefschrift}\\
    \vspace{10 mm}
  Ter verkrijging van het doctoraat in de \\
  Wiskunde en Natuurwetenschappen \\
  aan de Rijksuniversiteit Groningen \\
  op gezag van de \\
  Rector Magnificus, dr. E. Sterken\\ 
  in het openbaar te verdedigen op \\
  Maandag 1 September 2013 \\
  om 12.00 uur \\
    \vspace{10 mm}
  door\\
    \vspace{10 mm}
  Derk Arends\\
    \vspace{10 mm}
  Geboren op 15 juli 1983\\
  te Zwolle
\end{center}

\newpage
\thispagestyle{empty}
\begin{tabular}{ l l }
Promotor:             & Prof. dr. R.C. Jansen \\
                      & \\
Co-Promotors:         & dr. F. Johannes \\
                      & dr. M. Swertz \\
                      & \\
Beoordelingcommisie:  & Prof. dr. \\
                      & Prof. dr. \\
                      & Prof. dr. \\
\end{tabular}
\tableofcontents

\chapter{Introduction}
\lipsum[1-3]

\chapter{Phenotypes and Genotypes}

\emph{Using prior knowledge about phenotypes and how they arise / mix in sexual reproduction allows us to saturate 
existing genetic maps or create them de-novo when enough data is available. Using a sensitive pre-selection 
and mixture modeling approach the resolution of the genetic map of A. thaliana was improved.}

\null
\vfill

\begin{myexampleblock}{Under review:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)
\end{myexampleblock}
\newpage

\section{Pheno2Geno}
Pheno2geno is a package for the computational generation of markers and construction of genetic maps from 
molecular phenotypes in segregating inbred line populations, what is possible because phenotypes with a 
clearly separated bimodal or trimodal expression distribution can be used as genetic markers. Pheno2geno 
offers: de novo map construction, saturation of existing maps and detection of sample mix-ups. 

Pheno2geno starts by finding the phenotypes that are suitable as markers. These phenotypes should show 
differential expression between founders and are selected using a Student's t-test or RankProd. In the 
next step mixture modeling is used to select phenotypes that show bimodal (e.g. for RIL, BC) or trimodal 
(e.g. for F2) expression patterns with mixing proportions comparable to the expected segregation 
frequencies (e.g. 1 to 1 in RIL) across all the offspring individuals. Only these phenotypes are 
transformed from continuous measurements into discrete markers.

After marker detection the markers are used to create de novo map. Additional information, like known physical 
and/or genetic positions for all/some of the markers could be used and it will improve the quality of the 
resulting map. When a genetic map is available pheno2geno can be used to saturate it. For each of the markers 
it is assessed whether it has a single significant QTL. If so, it is being placed on the map in the position 
of this peak, if not it is dropped from the analysis.

Pheno2geno can additionally be used to point out sample mix-ups and errors. Using QTL information the package 
compares the observed phenotype value with the expected value. This results in calculating a mismatch score, 
and obtaining phenotype-based idealized genotypes. 

\section{Phenotypes}
\lipsum

\section{Markers and Maps}
\lipsum

\chapter{(Multiple) QTL mapping}

\emph{QTL mapping is the main analysis method used in the analysis of quantitative traits. This approach has been 
the analysis method to study quantitative traits since its discovery in 1980. Many variations to the basic single 
marker approach can be made, but none so fundamental as multiple QTL mapping using generalized lineair models. 
Computational and sample size problems were limiting the application of the approach. Novel experimental 
design and decreased costs have made sample size less of an issue, and by 'refreshing' the orginal algorithm 
more wide-spread application of this important algorithm is possible.}
\null
\vfill

\begin{myexampleblock}{Originally published as:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  \emph{R/qtl: High throughput Multiple QTL mapping}\\
  \bold{Bioinformatics} (2010) \\

  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\

  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, 
  Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using 
  Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)
\end{myexampleblock}

\newpage

\section{Single marker QTL mapping (R/qtl)}

R/qtl is an extensible, interactive environment for the mapping of quantitative trait loci (QTL) in experimental 
crosses. It is implemented as an add-on package for the freely available and widely used statistical language/
software R \cite{R:2009}. Since its introduction, R/qtl \cite{Broman:2003} has become a
reference implementation with an extensive guide on QTL mapping \cite{RQTLGuide:2009}. R/qtl development is 
continuous, with input from multiple collaborators and users.  We have introduced a full testing environment with 
regression testing, updated the license to the GPL version 3, and hosted the source code repository on Github,
which gives R/qtl software development high visibility and transparency. 

The development of R/qtl reflects trends in quantitative genetics, in particular the use of larger datasets, larger 
calculations and requirements for controlling the false discovery rate. These developments are partly driven by 
high-throughput genetical genomics---the name coined for the study of gene expression QTL (eQTL)\cite{Jansen:2001}, 
metabolite QTL (mQTL), protein QTL (pQTL).

\section{Multiple QTL mapping}

Multiple QTL Mapping (MQM) belongs to a family of QTL mapping methods, that include Haley-Knott regression
\cite{Haley:1992} and composite interval mapping CIM \cite{Zeng:1994}. MQM combines the strengths of generalized 
linear model regression ith those of interval mapping \cite{Jansen:1993, Jansen:1994b}. 

Recent developments in QTL mapping include Bayesian modelling of multiple QTL e.g. R/qtlbim package
\cite{Yandell:2007, Banerjee:2008}. Bayesian modelling, however, is computationally expensive, and arguably has 
little additional power when applied to high density maps, and (nearly) complete genotype data\cite{Handbook:Jansen:2007}. 
Still, we intend to combine the strengths of the different methods in future versions of R/qtl.

These days, with most experimenal setups and high density maps, improving precision may be achieved by increasing 
the population size first. For more information on QTL mapping and Bayesian analysis we refer to the `Handbook of 
Statisical Genetics` \cite{Handbook:2007}. MQM makes used of generalized linear models, thereby potentially 
providing unified analysis of non-normal traits.

MQM provides a practical, relevant and sensitive approach for mapping QTL in experimental populations. The 
theoretical framework of MQM was introduced and explored by R.C. Jansen\cite{Jansen:1994a} and explained in the 
`Handbook of Statistical Genetics' \cite{Handbook:Jansen:2007}. MQM has one known commercial implementation
\cite{Mapqtl:2002}, which has been used effectively in practical research, resulting in hundreds of papers, e.g., 
in mouse, plant, and fish, respectively\cite{DeMooij:2009, Jeuken:2009, Kitano:2009}.  Now, with MQM for R/qtl, 
we present the first free and open source implementation of MQM, that is multi-platform, scalable and suitable 
for automated procedures and large datasets.

\subsection{Features}
MQM for R/qtl is an automated three-stage procedure in which, in the first stage, missing genotype data is 'augmented'. 
In other words, rather than guessing one likely genotype, multiple genotypes are modelled with their estimated 
probabilities. In the second stage, important marker cofactors are selected by multiple regression and backward 
elimination. The third stage, a QTL is moved along the chromosomes using these pre-selected markers as cofactors. 
QTL are interval-mapped using the most informative model selected by either maximum likelihood or restricted maximum 
likelihood. A refined and automated procedure for cases with large numbers of marker cofactors is included. 

The method lets users test different QTL models by elimination of non-significant cofactors. MQM for R/qtl brings the 
following advantages to QTL mapping:
\begin{enumerate}\itemsep1pt
\item Higher power, as long as the QTL explain a reasonable amount of variation.
\item Protection against over-fitting, because MQM fixes the residual variance from the full model, which allows the 
use of more cofactors than may be used in, for example, composite interval mapping\cite{Zeng:1994}.
\item Prevention of ghost QTL detection (between two QTL in coupling phase).
\item Detection of negating QTL (QTL in repulsion phase). 
\item MQM gives (compared to CIM) a reduction in type I and type II error \cite{Handbook:Jansen:2007}.
\item A pragmatic permutation strategy for controlling the false discovery rate (FDR) and prevention of locating 
false QTL hot spots, as discussed in Breitling et al.\cite{Breitling:2008a}. Marker data is permuted, while keeping 
the correlation structure in the trait data.
\item High-performance computing by scaling on multi-CPU computers, as well as clustered computers, by calculating 
phenotypes in parallel, through the Message Passing Interface (MPI) of the SNOW package for R\cite{Tierney:2009}.
\item Visualizations for exploring interactions in a genomic circle plot (Fig. XXX) and cis- and trans-regulation (Fig. XXX).
\end{enumerate}

A 40-page tutorial for MQM explores, both the automated procedure, and the manual procedure of adding and removing 
cofactors, in an \emph{Arabidopsis thaliana} recombinant inbred line (RIL) metabolite (mQTL) dataset with 24 metabolites 
as phenotypes\cite{Fu:2007}. In addition, the tutorial visually explains the effects of data augmentation, cofactor 
selection, model selection, and tweaking of input parameters, such as cofactor significance. Genetic interactions 
(epistasis) are explored through effect plots, and an example is given of parallel computation. The tutorial is part 
of the software distribution of R/qtl and is available online.

\subsection{Conclusions}
MQM for R/qtl is a significant addition to the QTL mapper's toolbox. R/qtl provides the user with the most frequently 
used statistical analysis methods: single-marker analysis, interval mapping, Haley-Knott regression \cite{Haley:1992}, 
CIM \cite{Zeng:1994} and MQM \cite{Jansen:1994a}.  MQM has improved handling of missing data and allows more powerful 
and precise detection of QTL, compared to many other methods. Not only is this new implementation of MQM available in the
statistical R environment, which allows scripting for pipe-lined setups, it is also highly scalable through 
parallelisation and paves the way for high-throughput QTL analysis. With MQM, R/qtl is a free and high-performance 
comprehensive QTL mapping toolbox for the analysis of experimental populations. R/qtl now includes permutation strategies 
for determining thresholds of significance relevant for QTL and QTL hot spots; the first step towards causal inference and
network analysis.

\section{Classical phenotypes in a DesignGG experiment}
\lipsum

\section{Metabolites in a DesignGG experiment}
\lipsum

\chapter{High throughput data analysis}

\emph{Problems in bioinformatics are the huge amount of data gathered and the multitude of technologies used. We 
developed the xQTL workbench system\cite{Arends:2012} to store large amounts of phenotype and genotype data 
in the XGAP\cite{Swertz:2010a} dataformat. xQTL workbench can easily be adapted to a multitude of input and 
storage formats using the MOLGENIS\cite{Swertz:2004} generator. xQTL workbench uses the power of distibuted 
computing to allow massive parrallel QTL analysis and allows new analysis tools to be quickly added to the 
system, an application of the xQTL system is found in the WormQTL database\cite{Snoek:2012}.}

\null
\vfill

\begin{myexampleblock}{Originally published as:}
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  \emph{xQTL workbench: a scalable web environment for multi-level QTL analysis}\\
  \bold{Bioinformatics} (2012)\\\\

  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, et al.}\\
  \emph{WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp}\\
  \bold{Nucleic Acids Research} (2012)
\end{myexampleblock}

\newpage

\section{Storage (Molgenis, XGAP)}
\lipsum

\section{Analysis (xQTL workbench)}
\lipsum

\section{WormQTL}
We present WormQTL, a public web portal for the management of all these new data and integrated development 
of suitable analysis tools. The web server provides a rich set of analysis tools available to use directly, 
based on R/qtl \cite{Broman:2003, Arends:2010}. Users can upload and share new R scripts as 'plugin' for 
colleagues in the community to use directly. New data can be uploaded and downloaded using XGAP-extensible 
text format for genotype and phenotypes\cite{Swertz:2010a}. All data and tools can be accessed via web user 
interfaces and programming interfaces to R, REST, and SOAP web services. Large consortia as well as individual 
researchers, can have a private area that is under embargo for publication. All software is free for download 
as MOLGENIS 'app'\cite{Swertz:2010b}. WormQTL is freely accessible without registration and is hosted on a 
large computational cluster enabling high throughput analyses to all at http://www.wormqtl.org.

  \subsection{WormQTL - introduction}
  New biotechnologies enable exciting new experiments producing valuable data on genetic variation of gene 
  expression, proteins, metabolites and many other complex traits. Over the past decade increased efforts 
  have been made to explore the model species C. elegans as a platform for molecular quantitative genetics 
  and the systems biology of natural variation \cite{Gaertner:2010, Kammenga:2008}. These efforts have 
  resulted in a huge amount of phenotypic and genotypic data across developmental worm stages and environments. 
  To facilitate the accessibility, comparative analysis and meta-analysis of all these data \cite{Swertz:2007} 


\chapter{Mapping correlation}

\emph{In this chapter we take one step further then in the previous chapters and develop a new methodology for
quantitative genetics called Correlated Traits Locus (CTL) mapping, a method complementairy to QTL mapping. 
Where QTL associates differences in mean, CTL, associate differences in correlation to genetic variation, i.e. 
CTL identify regions in the genome for which one genotype leads to correlated expression between a pair of 
traits, while the other genotype shows none (or significantly different) correlation.}

\null
\vfill

\begin{myexampleblock}{In press:}
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  \emph{CTL mapping}\\
  \bold{BMC Bioinformatics} (2013)
\end{myexampleblock}

\newpage

\section{What is a CTL?}
\lipsum[1]

\section{Combining CTL and QTL information}
\lipsum

\section{Examples of CTL mapping}
\lipsum
\subsection{CTL mapping in an \emph{A. thaliana} RIL population}

\subsection{CTL mapping using human GWA data}

\chapter{Thesis summary}
\lipsum[1-3]

\chapter{Additional for Dissertation}
\section*{Nederlandse Samenvatting / Dutch Summary}
\addcontentsline{toc}{section}{Nederlandse Samenvatting / Dutch Summary}
\lipsum[1]

\section*{Abbreviations and acronyms}
\addcontentsline{toc}{section}{Abbreviations and acronyms}
\begin{tabular}{ l l }
BC:          & Backcross \\
bp:          & Base pair(s) \\
cM:          & centi Morgan \\
CTL:         & Correlated traits locus \\
Mbp:         & Mega base pairs = 1.000.000 bp \\
RIL:         & Recombinant inbred line \\
QTL:         & Quantitative trait locus \\
GWA:         & Genome Wide Association \\
\end{tabular}

\newpage

\section*{Acknowledgements}
\addcontentsline{toc}{section}{Acknowledgements}
\lipsum[1]

\newpage

\section*{List of publications}
\addcontentsline{toc}{section}{List of publications}
\subsection*{Authored:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  R/qtl: high throughput Multiple QTL mapping\\
  \bold{Bioinformatics} (2010)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\\\
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  xQTL workbench: a scalable web environment for multi-level QTL analysis\\
  \bold{Bioinformatics} (2012)\\\\
  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, Mark Elvin, et al.}\\
  WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp\\
  \bold{Nucleic Acids Research} (2012)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)

\subsection*{Co-Authored:}
  \authors{Morris A Swertz, Martijn Dijkstra, Tomasz Adamusiak, Danny Arends, et al.}\\
  The MOLGENIS toolkit: rapid prototyping of biosoftware at the push of a button\\
  \bold{BMC Bioinformatics} (2010)\\\\
  \authors{Morris A Swertz, K Joeri van der Velde, Bruno M Tesson, Danny Arends, et al.}\\
  XGAP: a uniform and extensible data model and software platform for genotype and phenotype experiments\\
  \bold{Genome Biology} (2010)\\\\
  \authors{Klaus Schughart, Danny Arends, P. Andreux, R. Balling, Pjotr Prins, et al.}\\
  SYSGENET: a meeting report from a new European network for systems genetics\\
  \bold{Mammalian Genome} (2010)\\\\
  \authors{Rudolf SN Fehrmann, Ritsert C. Jansen, Jan H. Veldink, Harm-Jan Westra, Danny Arends, et al.}\\
  Trans-eQTLs Reveal that Independent Genetic Variants Associated With a Complex Phenotype Converge on Intermediate Genes, with a Major Role for the HLA\\
  \bold{Plos Genetics} (2011)\\\\
  \authors{Caroline Durrant, Morris A. Swertz, Rudi Alberts, Danny Arends, Klaus Schughart, et al.}\\
  Bioinformatics tools and database resources for systems genetics analysis in mice - a short review and an evaluation of future needs\\
  \bold{Briefings in Bioinformatics} (2011)

\subsection*{In Press:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)\\\\
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  CTL mapping\\
  \bold{BMC Bioinformatics} (2013)

\subsection*{Acknowledged in:}
  \authors{Yang Li, Morris A Swertz, Gonzalo Vera, Jingyuan Fu, Rainer Breitling and Ritsert C Jansen}\\
  DesignGG: an R-package and web tool for the optimal design of genetical genomics experiments\\
  \bold{BMC Bioinformatics}, 10:188 (2009)\\\\
  \authors{Yang Li, Rainer Breitling and Ritsert C. Jansen}\\
  Generalizing genetical genomics: getting added value from environmental perturbation\\
  \bold{Trends in Genetics}, 24:518-524 (2008)

\bibliographystyle{plain}
\addcontentsline{toc}{chapter}{Bibliography}
\bibliography{Thesis}

\end{document}

