\documentclass[8pt, twoside, a5paper]{report}
\usepackage[usenames, dvipsnames, svgnames]{xcolor}
\usepackage{titlesec, blindtext, color, graphicx}
\usepackage[light,condensed,math]{kurier}
\usepackage[T1]{fontenc}
\usepackage{tcolorbox}
\usepackage{lipsum}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}
\definecolor{GREEN}{rgb}{0, 0.5, 0.1}

%\usepackage{lmodern} % to remove warnings of size substitution
\usepackage[top=0.6in, bottom=0.6in, left=0.6in, right=0.4in]{geometry}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}

\newenvironment{myexampleblock}[1]{%
    \tcolorbox[beamer,%
    noparskip,breakable,
    colback=LightGreen,colframe=DarkGreen,%
    colbacklower=LimeGreen!75!LightGreen,%
    title=#1]}%
    {\endtcolorbox}

\newcommand{\hsp}{\hspace{7pt}}
\newcommand{\hssp}{\hspace{3pt}}

\titleformat{\chapter}[hang]{\vspace{-2 cm}\huge\bfseries}{{\fontsize{50}{60}\selectfont \thechapter \hsp\textcolor{GREEN}{|}\hsp}}{0pt}{\huge\bfseries}
\titleformat{\section}[hang]{\large}{\thesection\hssp}{0pt}{\large\bfseries}

\newcommand{\authors}[1]{\small{#1}}
\newcommand{\italic}[1]{\textit{#1}}
\newcommand{\bold}[1]{{\bfseries #1}}
\newcommand{\code}[1]{{\tt #1}}
\newcommand{\mytitle}[1]{{\LARGE #1}}

\newcommand{\xqtlwb}{{\it x}QTL workbench }
\newcommand{\molgenis}{Molgenis} % don't use all caps, see journal guidelines
\newcommand{\xgap}{XGAP} % don't use all caps, see journal guidelines

\pagestyle{headings}
\renewcommand{\contentsname}{Table of content\vspace{-30pt}}

\begin{document}
\pagenumbering{arabic}
  \thispagestyle{empty}
  \begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
  \end{center}
\newpage
  \thispagestyle{empty}
  \noindent The work described in this thesis was carried out at the Groningen Bioinformatics Centre, University of Groningen, The Netherlands.
  This research was financially supported by the Centre for BioSystems Genomics (CBSG) and the Netherlands Consortium of Systems Biology (NCSB), both of which are part of the Netherlands Genomics Initiative / Netherlands Organisation for Scientific Research\\
  \vspace{130 mm}
  
  \noindent (c) 2013 by GBIC - Danny Arends\\
  Printed by:\\
  ISBN:\\
\newpage

\thispagestyle{empty}
\begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
    \vspace{10 mm}
  \bold{Proefschrift}\\
    \vspace{10 mm}
  Ter verkrijging van het doctoraat in de \\
  Wiskunde en Natuurwetenschappen \\
  aan de Rijksuniversiteit Groningen \\
  op gezag van de \\
  Rector Magnificus, dr. E. Sterken\\ 
  in het openbaar te verdedigen op \\
  Maandag 1 September 2013 \\
  om 12.00 uur \\
    \vspace{10 mm}
  door\\
    \vspace{10 mm}
  Derk Arends\\
    \vspace{10 mm}
  Geboren op 15 juli 1983\\
  te Zwolle
\end{center}

\newpage
\thispagestyle{empty}
\begin{tabular}{ l l }
Promotor:             & Prof. dr. R.C. Jansen \\
                      & \\
Co-Promotors:         & dr. F. Johannes \\
                      & dr. M. Swertz \\
                      & \\
Beoordelingcommisie:  & Prof. dr. \\
                      & Prof. dr. \\
                      & Prof. dr. \\
\end{tabular}
\tableofcontents

\input{chapter1.tex}

\input{chapter2.tex}

\input{chapter3.tex}

\input{chapter3.1.tex}

\input{chapter3.2.tex}

\input{chapter4.tex}

\input{chapter5.tex}

\chapter{Thesis summary}
\lipsum[1-3]

\chapter{Additional for Dissertation}
\section*{Nederlandse Samenvatting / Dutch Summary}
\addcontentsline{toc}{section}{Nederlandse Samenvatting / Dutch Summary}
\lipsum[1]

\section*{Abbreviations and acronyms}
\addcontentsline{toc}{section}{Abbreviations and acronyms}
{\footnotesize
\begin{tabular}{ l l }
API:         & Application Programming Interface\\
BC:          & Backcross \\
bp:          & Base pair(s) \\
cM:          & centi Morgan \\
CPNN:        & Collaborative computing project for NMR\\
CSV:         & Comma separated values\\
CTL:         & Correlated traits locus \\
DesignGG:    & Experimental design of genetical genomics software\\
DRY:         & Principle of don't repeat yourself\\
DSL:         & Domain Specific Language\\
EBI:         & European Bioinformatics Institute\\
FINDIS:      & Finish disease database\\
GEN2PHEN:    & EU project to unify human and model organism genetic variation databases\\
GMOD:        & Generic model organism database project\\
GUI:         & Graphical user interface\\
GWAS:        & Genome Wide Association Study\\
GWL:         & Genome Wide Linkage analysis\\
HGVBaseG2P:  & Human genome variation database of genotype-to-phenotype information\\
HTML:        & Hypertext markup language\\
IDE:         & Integrated Development Environment\\
JAR:         & Java Software Archive\\
LGPL:        & Lesser general public license\\
MAGE-TAB:    & Microarray gene expression tab delimited file format\\
Mbp:         & Mega base pairs = 1.000.000 bp \\
MOLGENIS:    & Molecular genetics information systems toolkit\\
NordicDB:    & Nordic Control Cohort Database\\
OBF:         & Open Bioinformatics Foundation\\
OntoCAT:     & Ontology common API toolkit\\
PEAA:        & Patterns for enterprise application architecture\\
QTL:         & Quantitative trait locus\\
RDF:         & Resource description format\\
REST:        & Representative state transfer web services\\
RIL:         & Recombinant inbred line \\
SNP:         & Single Nucleotide Polymorphism\\
SOAP:        & Simple Object Access Protocol\\
SQL:         & Structured Query Language\\
UML:         & Uniform data Modeling Language\\
WAR:         & Web Application aRchive file\\
XML:         & Extensible Markup Language\\
XGAP:        & Extensible genotype and phenotype software platform. 
\end{tabular}
}
\newpage

\section*{Acknowledgements}
\addcontentsline{toc}{section}{Acknowledgements}
\lipsum[1]

\newpage

\section*{List of publications}
\addcontentsline{toc}{section}{List of publications}
\subsection*{Authored:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  R/qtl: high throughput Multiple QTL mapping\\
  \bold{Bioinformatics} (2010)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\\\
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  xQTL workbench: a scalable web environment for multi-level QTL analysis\\
  \bold{Bioinformatics} (2012)\\\\
  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, Mark Elvin, et al.}\\
  WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp\\
  \bold{Nucleic Acids Research} (2012)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)

\subsection*{Co-Authored:}
  \authors{Morris A Swertz, Martijn Dijkstra, Tomasz Adamusiak, Danny Arends, et al.}\\
  The MOLGENIS toolkit: rapid prototyping of biosoftware at the push of a button\\
  \bold{BMC Bioinformatics} (2010)\\\\
  \authors{Morris A Swertz, K Joeri van der Velde, Bruno M Tesson, Danny Arends, et al.}\\
  XGAP: a uniform and extensible data model and software platform for genotype and phenotype experiments\\
  \bold{Genome Biology} (2010)\\\\
  \authors{Klaus Schughart, Danny Arends, P. Andreux, R. Balling, Pjotr Prins, et al.}\\
  SYSGENET: a meeting report from a new European network for systems genetics\\
  \bold{Mammalian Genome} (2010)\\\\
  \authors{Rudolf SN Fehrmann, Ritsert C. Jansen, Jan H. Veldink, Harm-Jan Westra, Danny Arends, et al.}\\
  Trans-eQTLs Reveal that Independent Genetic Variants Associated With a Complex Phenotype Converge on Intermediate Genes, with a Major Role for the HLA\\
  \bold{Plos Genetics} (2011)\\\\
  \authors{Caroline Durrant, Morris A. Swertz, Rudi Alberts, Danny Arends, Klaus Schughart, et al.}\\
  Bioinformatics tools and database resources for systems genetics analysis in mice - a short review and an evaluation of future needs\\
  \bold{Briefings in Bioinformatics} (2011)

\subsection*{In Press:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)\\\\
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  CTL mapping\\
  \bold{BMC Bioinformatics} (2013)

\subsection*{Acknowledged in:}
  \authors{Yang Li, Morris A Swertz, Gonzalo Vera, Jingyuan Fu, Rainer Breitling and Ritsert C Jansen}\\
  DesignGG: an R-package and web tool for the optimal design of genetical genomics experiments\\
  \bold{BMC Bioinformatics}, 10:188 (2009)\\\\
  \authors{Yang Li, Rainer Breitling and Ritsert C. Jansen}\\
  Generalizing genetical genomics: getting added value from environmental perturbation\\
  \bold{Trends in Genetics}, 24:518-524 (2008)

\bibliographystyle{plain}
\addcontentsline{toc}{chapter}{Bibliography}
{\footnotesize
\bibliography{Thesis}
}
\end{document}

