\documentclass[11pt, twoside]{extbook}
\usepackage[paperwidth=170mm,paperheight=240mm, top=27mm, bottom=27mm, left=20mm, right=20mm]{geometry}
\special{papersize=170mm,240mm}
\usepackage[usenames, dvipsnames, svgnames, table]{xcolor}
\usepackage{titlesec, blindtext, color, graphicx}
\usepackage[light,condensed,math]{kurier}
\usepackage[T1]{fontenc}
\usepackage{tcolorbox}
\usepackage{lipsum}
\usepackage{fancyhdr}
\usepackage[font={small,it}]{caption}
\usepackage[square,sort,comma,numbers]{natbib}
\usepackage{subfigure}
\usepackage{wrapfig}
\usepackage{hyperref}       % Should be removed for printing, however is now handy for links
\captionsetup{singlelinecheck=off,labelfont=bf,labelsep=space}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}
\definecolor{GREEN}{rgb}{0, 0.5, 0.1}

%\renewcommand{\familydefault}{\sfdefault}

\usepackage{lmodern}        % Removes the warnings of size substitution
\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}

\definecolor{LightGray}{gray}{0.95}
\definecolor{DarkGray}{gray}{0.35}

\newenvironment{myexampleblock}[1]{%
    \tcolorbox[beamer,%
    noparskip,breakable,
    colback=LightGray,colframe=DarkGray,%
    colbacklower=DarkGray!75!LightGray,%
    title=#1]}%
    {\endtcolorbox}

\newcommand{\hsp}{\hspace{7pt}}
\newcommand{\hssp}{\hspace{3pt}}
\newcommand{\degree}{\ensuremath{^\circ}}

\titleformat{\chapter}[hang]{\vspace{-3cm}\huge\bfseries}{{\fontsize{50}{60}\selectfont \thechapter \hsp\textcolor{GREEN}{|}\hsp}}{0pt}{\huge\bfseries}
\titleformat{\section}[hang]{\Large}{\thesection\hssp}{0pt}{\Large\bfseries}

\newcommand{\authors}[1]{\small{#1}}
\newcommand{\italic}[1]{\textit{#1}}
\newcommand{\bold}[1]{{\bfseries #1}}
\newcommand{\code}[1]{{\tt #1}}
\newcommand{\mytitle}[1]{{\LARGE #1}}

\newcommand{\xqtlwb}{{\it x}QTL workbench }
\newcommand{\molgenis}{MOLGENIS} % don't use all caps, see journal guidelines
\newcommand{\xgap}{XGAP} % don't use all caps, see journal guidelines

\usepackage{fancyhdr}
\pagestyle{fancy}
\fancyhf{}
\fancyfoot[RO, LE]{\thepage}
\fancyhead[C]{\leftmark}

\fancypagestyle{plain}{%
  \fancyhf{}
  \fancyfoot[RO, LE]{\thepage}
}

\renewcommand{\contentsname}{Table of Contents}

\begin{document}
\pagenumbering{arabic}
  \thispagestyle{empty}
  \begin{center}
  \mytitle{ High throughput computational methods and software for systems genetics }\\
  \end{center}
\newpage
  \thispagestyle{empty}
  \noindent The work described in this thesis was carried out at the Groningen 
  Bioinformatics Centre, University of Groningen, The Netherlands. This research 
  was financially supported by the Centre for BioSystems Genomics (CBSG) and the 
  Netherlands Consortium of Systems Biology (NCSB), both of which are part of the 
  Netherlands Genomics Initiative / Netherlands Organisation for Scientific Research\\
  \vspace{110 mm}
  
  \noindent (c) 2013 by GBIC - Danny Arends\\
  Artwork by:\\
  Printed by:\\
  ISBN (Print):\\
  ISBN (Digital):\\
\newpage

\thispagestyle{empty}
\begin{center}
  \mytitle{ High throughput computational methods and software for systems genetics }\\
    \vspace{10 mm}
  \bold{Proefschrift}\\
    \vspace{10 mm}
  Ter verkrijging van het doctoraat in de \\
  Wiskunde en Natuurwetenschappen \\
  aan de Rijksuniversiteit Groningen \\
  op gezag van de \\
  Rector Magnificus, dr. E. Sterken\\ 
  in het openbaar te verdedigen op \\
  Maandag Xx XxXxX 2014 \\
  om 12.00 uur \\
    \vspace{10 mm}
  door\\
    \vspace{10 mm}
  Derk Arends\\
    \vspace{10 mm}
  Geboren op 15 juli 1983\\
  te Zwolle
\end{center}

\newpage
\thispagestyle{empty}
\begin{tabular}{ l l }
Promotor:             & Prof. dr. R. C. Jansen \\
                      & \\
Co-Promotors:         & dr. F. Johannes \\
                      & dr. M. A. Swertz \\
                      & \\
Beoordelingcommisie:  & Prof. dr. \\
                      & Prof. dr. \\
                      & Prof. dr. \\
\end{tabular}
\tableofcontents
\newpage
\chapter*{Summary}\vspace{-30pt}
\addcontentsline{toc}{chapter}{Summary}
Systems genetics is the inter disciplinairy field which deals with the consequences of genetic 
variation on all bio molecular levels of a biological system. The aim of systems genetics is to 
understand systems by partitioning variation into three major categories: Genetic, Environment 
and Error variation, and explain how complex phenotypes arise from a combination of these three 
major factors across bio molecular levels. 

Natural variation (or genetic pertubation) can be used to interogate the genetic component of 
phenotype variation. Combined with environmental pertubation we can investigate the influence 
different environments and the interaction between genetics and environment. Experimental design 
and statistics are used to minimize and estimate error variance. 

Chapter \ref{chap:introduction} contains a short introduction after which this thesis will 
continue by highlighting our contributions to the field of systems genetics.

Chapter \ref{chap:pheno2geno} shows Pheno2Geno an R package that deals with the creation of 
genetic maps from large scale omics data. The theory behind genetic map construction is around 
100 years old, most software was witen in the 1980s. Software available for genetic map construction 
has not been adapted yet to make use of new technologies such as multi threading or cluster 
computing. Pheno2Geno aims to provide analysis of data from tiling arrays or RNAseq to generate 
gene based expression markers (GEMs) and create high density genetic maps.

Chapter \ref{chap:mqm} describes the implementation of the Multiple QTL mapping routine 
by R. C. Jansen into R/qtl. Adding a 'new' algorithm to the R/qtl toolset which aims to 
provide a range of QTL mapping tools for inbred crosses. R/qtl is the basis of a toolset 
build around a unified data structure allowing easy adaptation and extension of the software. 
R/qtl allows researchers to analyse data from different sources or to quickly compare 
different approaches. This chapter showcases our contributions to the R/qtl package such as: 
MQM, visualizations and parallel computation of QTLs and permutations.

Chapter \ref{chap:ctlmapping} describes current work on using differences in correlation to 
generate interaction networks and detect cell type specific QTL effects. Correlated Traits 
Locus analysis (or CTL mapping) enables researchers to find genetic loci controlling correlation 
differences in segregating phenotypes. A variation on this method has proven valuable in 
discovering cell type specific eQTL effects. Using these effects it is possible 
to untangle cell mixtures seen in whole blood.

Chapter \ref{chap:xqtlwormbench} is to show case our ideas for a generic storage and 
computation platform for systems genetics. Our demo system xQTL workbench is currently 
being used as a backend to the WormQTL and WormQTL-HD database. xQTL workbench allows 
users to store and share their data in a local or web environment, and run analysis across 
datasets using the power of distributed computing. It comes standard with QTL mapping 
tools such as: R/qtl, PLINK and qtlbim but also provides a web inferfaces, data importers, 
APIs and visualizations.

I trust you enjoy reading this thesis as much as I enjoyed creating it during the last four years,\\\\

Danny Arends (Aug 2013)
\newpage
\thispagestyle{empty}
\input{chapter1.tex}

\input{chapter2.tex}

\input{chapter3.tex}

\input{chapter3.1.tex}

\input{chapter3.2.tex}

\input{chapter4.tex}

\input{chapter5.tex}

\input{additional.tex}

\bibliographystyle{unsrt85}
\addcontentsline{toc}{chapter}{Bibliography}
{\footnotesize
\bibliography{Thesis}
}
\end{document}

