\documentclass[8pt, twoside, a5paper]{report}
\usepackage[usenames, dvipsnames, svgnames]{xcolor}
\usepackage{titlesec, blindtext, color, graphicx}
\usepackage[light,condensed,math]{kurier}
\usepackage[T1]{fontenc}
\usepackage{tcolorbox}
\usepackage{lipsum}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}
\definecolor{GREEN}{rgb}{0, 0.5, 0.1}

%\usepackage{lmodern} % to remove warnings of size substitution
\usepackage[top=0.6in, bottom=0.6in, left=0.6in, right=0.4in]{geometry}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}

\newenvironment{myexampleblock}[1]{%
    \tcolorbox[beamer,%
    noparskip,breakable,
    colback=LightGreen,colframe=DarkGreen,%
    colbacklower=LimeGreen!75!LightGreen,%
    title=#1]}%
    {\endtcolorbox}

\newcommand{\hsp}{\hspace{7pt}}
\newcommand{\hssp}{\hspace{3pt}}

\titleformat{\chapter}[hang]{\vspace{-2 cm}\huge\bfseries}{{\fontsize{50}{60}\selectfont \thechapter \hsp\textcolor{GREEN}{|}\hsp}}{0pt}{\huge\bfseries}
\titleformat{\section}[hang]{\large}{\thesection\hssp}{0pt}{\large\bfseries}

\newcommand{\authors}[1]{\small{#1}}
\newcommand{\italic}[1]{\textit{#1}}
\newcommand{\bold}[1]{{\bfseries #1}}
\newcommand{\code}[1]{{\tt #1}}
\newcommand{\mytitle}[1]{{\LARGE #1}}

\newcommand{\xqtlwb}{{\it x}QTL workbench}
\newcommand{\molgenis}{Molgenis} % don't use all caps, see journal guidelines
\newcommand{\xgap}{XGAP} % don't use all caps, see journal guidelines

\pagestyle{headings}
\renewcommand{\contentsname}{Table of content\vspace{-30pt}}

\begin{document}
\pagenumbering{arabic}
  \thispagestyle{empty}
  \begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
  \end{center}
\newpage
  \thispagestyle{empty}
  \noindent The work described in this thesis was carried out at the Groningen Bioinformatics Centre, University of Groningen, The Netherlands.
  This research was financially supported by\\
  \vspace{130 mm}
  
  \noindent (c) 2013 by GBIC - Danny Arends\\
  Printed by:\\
  ISBN:\\
\newpage

\thispagestyle{empty}
\begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
    \vspace{10 mm}
  \bold{Proefschrift}\\
    \vspace{10 mm}
  Ter verkrijging van het doctoraat in de \\
  Wiskunde en Natuurwetenschappen \\
  aan de Rijksuniversiteit Groningen \\
  op gezag van de \\
  Rector Magnificus, dr. E. Sterken\\ 
  in het openbaar te verdedigen op \\
  Maandag 1 September 2013 \\
  om 12.00 uur \\
    \vspace{10 mm}
  door\\
    \vspace{10 mm}
  Derk Arends\\
    \vspace{10 mm}
  Geboren op 15 juli 1983\\
  te Zwolle
\end{center}

\newpage
\thispagestyle{empty}
\begin{tabular}{ l l }
Promotor:             & Prof. dr. R.C. Jansen \\
                      & \\
Co-Promotors:         & dr. F. Johannes \\
                      & dr. M. Swertz \\
                      & \\
Beoordelingcommisie:  & Prof. dr. \\
                      & Prof. dr. \\
                      & Prof. dr. \\
\end{tabular}
\tableofcontents

\chapter{Introduction}
\lipsum[1-3]

\chapter{Phenotypes and Genotypes}

\emph{Using prior knowledge about phenotypes and how they arise / mix in sexual reproduction allows us to saturate 
existing genetic maps or create them de-novo when enough data is available. Using a sensitive pre-selection 
and mixture modeling approach the resolution of the genetic map of A. thaliana was improved.}

\null
\vfill

\begin{myexampleblock}{Under review:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)
\end{myexampleblock}
\newpage

\section{Pheno2Geno}
Genetic markers and maps are instrumental for quantitative trait locus (QTL) mapping in segregating populations. The
resolution of QTL localisation depends on the number of informative recombinations in the population and how well 
these recombinations are tagged by markers. Thus larger populations and denser marker maps perform better at detecting 
and locating QTLs. In practice, marker maps are often still too sparse. However, maps can be saturated or even be derived 
\emph{de-novo} from high-throughput omics data, such as gene expression, protein or metabolite abundance data. This is 
because molecular phenotypes are influenced by genetic variation and they will show a clear multimodal distribution due 
to major QTL effects, such information can therefore be converted into useful genetic markers.

The Pheno2Geno R package is developed for high-throughput generation of genetic markers and maps from molecular 
phenotypes. Pheno2Geno selects suitable phenotypes that show clear differential expression in the founders. 
Pheno2Geno uses mixture modelling to select phenotypes showing segregation ratios close to the expected mendelian segregation 
ratios and transform them into genetic markers suitable for map construction and/or saturation. 
Pheno2Geno analyses the candidate genetic markers and excludes those showing multiple QTL, epistatically interacting QTL, 
and QTL by environment interactions to provide a set of robust markers for QTL mapping protecting against genetic markers 
from a non genetic origin.

We demonstrate our tool using gene expression data of 370.000 transcripts in 164 \emph{A. thaliana} Recombinant Inbred Lines (RILs). 
Pheno2Geno is able to saturate the existing genetic map decreasing the average distance between markers from 7.1 cM to 0.89 cM, 
close to the theoretical limit of 0.6 cM, pinpointing almost all of the informative recombinations in the population. Pheno2Geno 
is also able to created a \emph{de-novo} map from the gene expression data that is twice as dense as the original genetic map.

The Pheno2Geno package offers high-throughput \emph{de-novo} map construction and saturation of existing genetic maps. 
Processing of the showcase dataset takes less than 30 minutes on an average desktop PC. 
Pheno2Geno improves QTL mapping results at no additional laboratory cost and with minimum computational effort. Pheno2Geno 
results are formatted for direct use in R/qtl, the leading R package for QTL studies. Pheno2Geno is freely available on 
CRAN under GNU GPL version 3.

\subsection{Pheno2Geno - Background}
QTL mapping \cite{Lander:1989} is a powerful approach used in population analysis to link genetic variation to phenotypic 
variation between individuals. It requires polymorphic genetic markers positioned on a genetic map. Here we present Pheno2Geno, 
an R package to select and convert high-throughput phenotypes with major QTL into additional genetic markers, even when tens of 
thousands of molecular phenotypes (e.g. gene expression) are available, and then uses these markers to saturate a known genetic map.

Previously, gene expression data was used to derive genetic markers in an \emph{A. thaliana} RIL population 
\cite{west:2006} and a \emph{L. sativa} RIL population\cite{Truco:2013}. This work was only focussed only on RIL transcriptomics 
data, while Pheno2Geno is capable of handling any type of omics data and more cross types (backcross (BC), recombinant inbred 
lines (RIL) by selfing and sibling mating and F2 intercross). In addition, the method used\cite{west:2006, Truco:2013} is a simple method for 
genotype calling that works only for dichotome data, while Pheno2Geno uses statistical mixture models, posterior probabilities 
for genotype calling and error control that works for continuous data distributions\cite{jansen:1993, jansen:2001b}. 

In order to facilitate transition into the QTL mapping phase of the analysis, Pheno2Geno has output structures compatible with 
R/qtl, the leading R package for QTL analysis in segregating populations\cite{Broman:2003, Arends:2010}. The package also provides an option to output 
maps as GFF formatted file. The GFF file format is supported by most genome browsers, allowing users to use their favourite genome 
browser to explore and/or compare resulting maps (Fig. 4).

\subsection{Pheno2Geno - Features}
Pheno2Geno provides the following functionality for constructing and/or saturating genetic maps:

{\bf 1) Preprocessing of the data}: \emph{(optional)} Pheno2Geno offers a selection of data transformation functions (including: 
log, sqrt, reciprocal, probit and logit). Depending on the type of input data provided, the user can select a pre-processing step, 
e.g. gene expression data measured using microarrays are generally log \cite{Quackenbush:2002} or square root \cite{jansen:2001b, 
Gort:2010} transformed before further analysis. R already provides packages for pre-processing and data normalization. However 
Pheno2Geno provides these methods such that they can be used 'on the fly' and further decrease memory usage of the algorithm.

{\bf 2) Analysis of parental data}: \emph{(optional)} When parental data are available Pheno2Geno can pre-select phenotypes for 
analysis. Pheno2Geno will use a \emph{t}-test to select molecular phenotypes showing significant phenotypic differences between 
the inbred parental strains, to reduce the computational load in the follow up analysis. By being more restrictive Pheno2Geno 
keeps only the phenotypes with the most significant differences between the parents for further analysis. This again reduces 
the computational burden of the analysis.

{\bf 3) Analysis of segregating populations}:
Phenotypes with major QTL will show clear multimodal distributions in the segregating population. We fit a mixture model 
to the phenotype distribution \cite{jansen:1993, jansen:2001b, Benaglia:2009}. Phenotypes are selected as candidate markers, 
only when significant multimodality and mixing proportions close to the expected segregation frequency are observed, e.g. 
1:1 for a bimodal distribution of two homozygous classes in a RIL; 1:2:1 for a trimodal distribution of two homozygous and 
one heterozygous class in an F2 cross.

{\bf 4) Assigning genotypes}:
Posterior probabilities of belonging to an underlying component distributions in the mixture are calculated for each component
\cite{jansen:2001b, Benaglia:2009}. Using the posterior probabilities the continuous phenotype values are converted into 
discrete data (e.g. 0 or 1 for RILs; 0, 1 or 2 for F2). If parental data are available, Pheno2Geno uses the direction of 
the difference in the parental data to convert these discrete data into marker genotype data with a parental origin 
label (A or B for RILs, A, H or B for F2). If the posterior probability for a specific marker/individual combination is 
lower than a user-specified threshold, a missing value (*) or partly informative value (e.g. not A, but homozygous B or 
heterozygous H) is assigned to avoid introducing genotyping errors.

{\bf 5) Environment and epistasis}: 
West et al. \cite{west:2007} highlighted some central issues when generating genetic markers from gene expression data: 1) 
The influence of environmental interactions on gene expression and 2) The influence of multiple QTL and epistatic interactions 
between two genetic loci. When information about the environment is available, Pheno2Geno will either flag or remove markers 
affected by environment using a user defined significance threshold. Pheno2Geno performs interval mapping (using the R/qtl \emph{scanone} function) 
on the molecular phenotypes to find the best locations for the candidate markers. Additionally Pheno2Geno tests if candidate markers 
are affected by multiple QTLs or if they show an epistatic interaction. Depending on the user these are also flagged or removed 
from further analysis.

{\bf 6) Saturation of a known map}:
Pheno2Geno saturates a known map by placing all candidate markers at their most likely genetic positions and re-estimating 
the map distances. Duplicate candidate markers and markers located at the exact position of a known marker are removed before 
saturating the genetic map.

{\bf 6) \emph{De-novo} construction of genetic maps}: 
When no initial map is available, Pheno2Geno can be used to create an initial 'skeleton' map. This skeleton map is produced 
using very strict settings in the mixture model analysis to obtain a limited number of highly thrustworthy markers, which 
are assigned to linkage groups by using the R/qtl function \emph{formLinkageGroups}. Additional information provided by the user 
can be used in this step, e.g. known physical and/or genetic positions will be used by Pheno2Geno to assign physical 
chromosome IDs to linkage groups, and to determine the correct orientation of chromosomes. Then Pheno2Geno orders all the 
markers inside a linkage group by using the R/qtl \emph{orderMarkers} function. As a final step the skeleton map is saturated to 
improve resolution following the procedure outline above.

{\bf 7) Detection of errors}: \emph{(optional)}
After saturation or \emph{de-novo} construction of a genetic map, Pheno2Geno will detect and correct any genotyping errors 
(double recombinations, missing data, semi informative markers) using the R/qtl function \emph{fill.geno}. Furthermore when 
saturating a known map with available genotype data, Pheno2Geno is able to detect sample mix-ups in the original data using 
R/lineup (which is part of the R/qtl toolset). Users can also use external tools like MixupMapper \cite{Westra:2011} 
beforehand to detect and correct the original genotype data.

\subsection{Pheno2Geno - Results}
The original AFLP map created using a population of 420 RILs derived from a cross \emph{Arabidopsis thaliana} Bayreuth 
(Bay-0) x Shahdara (Sha) \cite{loudet:2002} contains 69 AFLP markers at an average map distance of 7.1 cM also derived 
using mixture models\cite{jansen:2001b, loudet:2002}. 

We use Pheno2Geno to saturate this genetic map using newly available gene expression data on 370,000 transcripts measured 
on 164 individuals from the \emph{A. thaliana} core population. Pheno2Geno detects 10.801 phenotypes differentially 
expressed between parents ($P < 0.01$), from which mixture modeling identifies 1230 potential markers showing a 1 to 1 
segregation ratio. We remove 267 markers which appear to be affected by one or more environments ($LOD >= 7.5$), 
After QTL mapping using the R/qtl \emph{scanone} function 286 candidate markers are removed which show none($LOD < 15$) (279 markers) 
 or multiple QTL (7 markers). Scanning for epistasis shows 77 candidate markers which appear to show pairwise 
epistatic interactions ($LOD >= 7.5$). Using the remaining 600 candidate markers we saturate 
the original map, and remove another 103 duplicate/co-localizing markers. This results in 497 new markers (720\% increase), 
decreasing the average map distance from 7.1 cM to 0.89 cM. Saturation of the \emph{A. thaliana} Bay-0 x Sha map led to a 
more than sevenfold improvement in marker density at no additional lab cost. Map distances were re-estimated using Kosambi 
function. Map expansion is observed on chromosomes 4 and 5 increasing total map length from 480.7 to 501.5 cM (Fig. 1). 

A \emph{de-novo} reconstruction on gene expression data only (ignoring the original markers and map) would have led to 
a skeleton map containing 227 markers with average distance of 2.2 cM. 

QTL mapping of our previously published classical phenotype dataset \cite{Joosen:2011} onto the new saturated map leads to 
an improvement in QTL likelihood for 56\% of previously detected QTLs. Additionally, 29 new QTLs are detected 
as being significant on the saturated map, while showing LOD scores close to the threshold when mapped to the original 
map (between 3.4 to 5 LOD, an example is found in Fig. 2).

Moreover, all the gene expression probes showing differential expression between parents (10801 probes) were scanned. 5837
had a significant ($LOD > 5$) QTL on an original map. 3943 (66\%) showed increase in likelihood on a new map(Fig. 3). 
Additionally, 210 new QTLs are detected as being significant on the saturated map.
  
\subsection{Pheno2Geno - Conclusions and Discussion}
Pheno2Geno is the first generic software package for generating genetic markers and maps from high-throughput molecular 
phenotypes for any inbred diploid populations (backcross, F2 intercross and recombinant inbred lines). Pheno2Geno selects 
phenotypes that show multimodal distributions with proportions corresponding to expected segregation ratios. When 
information about environment is available, it is used to eliminate markers possibly affected by environment or 
interaction between environment and genotype. Moreover, Pheno2Geno eliminates markers affected by multiple QTL or 
pairwise epistatic interactions between loci.

Pheno2Geno is able to process large volumes of different kinds of molecular phenotypes \cite{Trelles:2011}. Computational 
effort may also be decreased by only selecting phenotypes that are different between founders in the first step of the 
analysis. Memory requirments of the algorithm are decreased by reading in and processing files in chunks rather than at once. 
Complete analysis of the showcase data (370,000 transcripts) is performed in matter of hours on an average 
desktop PC (Intel Core i5, 4 GB of RAM). For even larger datasets, the Pheno2Geno package is embedded in xQTL workbench
\cite{Arends:2012a, arends:2012b} allowing for easy parallelization, use of cluster and cloud computing.\newline

Pheno2Geno was developed to improve and facilitate QTL studies. A leading package for QTL mapping is R/qtl. 
Pheno2Geno results are formatted for direct use in R/qtl, providing a smooth transition into the QTL mapping phase 
of the analysis. \newline

In gene expression studies, the molecular phenotypes can show significant major eQTL not only in a local eQTL, but also distant eQTLs. 
In the case of a local eQTL the derived marker will be located at the position of the original gene. Often such eQTL are caused by 
differential signal due to polymorphisms in the probe region \cite{Alberts:2005,Alberts:2007}. In the case of a distant eQTL the 
derived marker will not be located by Pheno2Geno at the (possibly known) position of the original probe but correctly at the 
position/region of the distant QTL. \newline

Currently Pheno2Geno uses parental data to reduce the computational burden and to assign the parental origin to the mixture-model 
derived genotype categories in the segregating population. When such information is not available, mixture-model based scores 
cannot be converted into scores showing the parental origin (e.g. scores 0 and 1 in RILs cannot be converted into scores A and B 
showing parental origin). Pheno2Geno solves this problem by forming twice as much linkage groups as the expected number of 
chromosomes and then merges anti-correlated pairs of linkage groups into a single chromosome. \newline

\subsubsection{Pheno2Geno - Availability and requirements}
\begin{itemize}
\item Project name: Pheno2Geno
\item Project home page: http://www.pheno2geno.nl
\item Operating system(s): Any platform for which the R software \cite{rgui} is implemented, including Microsoft Windows, Mac OS and Linux
\item Programming language: R
\item Other requirements: Packages installed in R: qtl, mixtools \cite{Benaglia:2009}
\item License: GNU General Public License version 3
\item Any restrictions to use by non-academics: None
\end{itemize}




\chapter{(Multiple) QTL mapping}

\emph{QTL mapping is the main analysis method used in the analysis of quantitative traits. This approach has been 
the analysis method to study quantitative traits since its discovery in 1980. Many variations to the basic single 
marker approach can be made, but none so fundamental as multiple QTL mapping using generalized lineair models. 
Computational and sample size problems were limiting the application of the approach. Novel experimental 
design and decreased costs have made sample size less of an issue, and by 'refreshing' the orginal algorithm 
more wide-spread application of this important algorithm is possible.}
\null
\vfill

\begin{myexampleblock}{Originally published as:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  \emph{R/qtl: High throughput Multiple QTL mapping}\\
  \bold{Bioinformatics} (2010) \\

  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\

  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, 
  Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using 
  Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)
\end{myexampleblock}

\newpage

\section{Single marker QTL mapping (R/qtl)}

R/qtl is an extensible, interactive environment for the mapping of quantitative trait loci (QTL) in experimental 
crosses. It is implemented as an add-on package for the freely available and widely used statistical language/
software R \cite{R:2009}. Since its introduction, R/qtl \cite{Broman:2003} has become a
reference implementation with an extensive guide on QTL mapping \cite{RQTLGuide:2009}. R/qtl development is 
continuous, with input from multiple collaborators and users.  We have introduced a full testing environment with 
regression testing, updated the license to the GPL version 3, and hosted the source code repository on Github,
which gives R/qtl software development high visibility and transparency. 

The development of R/qtl reflects trends in quantitative genetics, in particular the use of larger datasets, larger 
calculations and requirements for controlling the false discovery rate. These developments are partly driven by 
high-throughput genetical genomics---the name coined for the study of gene expression QTL (eQTL)\cite{Jansen:2001}, 
metabolite QTL (mQTL), protein QTL (pQTL).

\section{Multiple QTL mapping}

Multiple QTL Mapping (MQM) belongs to a family of QTL mapping methods, that include Haley-Knott regression
\cite{Haley:1992} and composite interval mapping CIM \cite{Zeng:1994}. MQM combines the strengths of generalized 
linear model regression ith those of interval mapping \cite{Jansen:1993, Jansen:1994b}. 

Recent developments in QTL mapping include Bayesian modelling of multiple QTL e.g. R/qtlbim package
\cite{Yandell:2007, Banerjee:2008}. Bayesian modelling, however, is computationally expensive, and arguably has 
little additional power when applied to high density maps, and (nearly) complete genotype data\cite{Handbook:Jansen:2007}. 
Still, we intend to combine the strengths of the different methods in future versions of R/qtl.

These days, with most experimenal setups and high density maps, improving precision may be achieved by increasing 
the population size first. For more information on QTL mapping and Bayesian analysis we refer to the `Handbook of 
Statisical Genetics` \cite{Handbook:2007}. MQM makes used of generalized linear models, thereby potentially 
providing unified analysis of non-normal traits.

MQM provides a practical, relevant and sensitive approach for mapping QTL in experimental populations. The 
theoretical framework of MQM was introduced and explored by R.C. Jansen\cite{Jansen:1994a} and explained in the 
`Handbook of Statistical Genetics' \cite{Handbook:Jansen:2007}. MQM has one known commercial implementation
\cite{Mapqtl:2002}, which has been used effectively in practical research, resulting in hundreds of papers, e.g., 
in mouse, plant, and fish, respectively\cite{DeMooij:2009, Jeuken:2009, Kitano:2009}.  Now, with MQM for R/qtl, 
we present the first free and open source implementation of MQM, that is multi-platform, scalable and suitable 
for automated procedures and large datasets.

\subsection{Features}
MQM for R/qtl is an automated three-stage procedure in which, in the first stage, missing genotype data is 'augmented'. 
In other words, rather than guessing one likely genotype, multiple genotypes are modelled with their estimated 
probabilities. In the second stage, important marker cofactors are selected by multiple regression and backward 
elimination. The third stage, a QTL is moved along the chromosomes using these pre-selected markers as cofactors. 
QTL are interval-mapped using the most informative model selected by either maximum likelihood or restricted maximum 
likelihood. A refined and automated procedure for cases with large numbers of marker cofactors is included. 

The method lets users test different QTL models by elimination of non-significant cofactors. MQM for R/qtl brings the 
following advantages to QTL mapping:
\begin{enumerate}\itemsep1pt
\item Higher power, as long as the QTL explain a reasonable amount of variation.
\item Protection against over-fitting, because MQM fixes the residual variance from the full model, which allows the 
use of more cofactors than may be used in, for example, composite interval mapping\cite{Zeng:1994}.
\item Prevention of ghost QTL detection (between two QTL in coupling phase).
\item Detection of negating QTL (QTL in repulsion phase). 
\item MQM gives (compared to CIM) a reduction in type I and type II error \cite{Handbook:Jansen:2007}.
\item A pragmatic permutation strategy for controlling the false discovery rate (FDR) and prevention of locating 
false QTL hot spots, as discussed in Breitling et al.\cite{Breitling:2008a}. Marker data is permuted, while keeping 
the correlation structure in the trait data.
\item High-performance computing by scaling on multi-CPU computers, as well as clustered computers, by calculating 
phenotypes in parallel, through the Message Passing Interface (MPI) of the SNOW package for R\cite{Tierney:2009}.
\item Visualizations for exploring interactions in a genomic circle plot (Fig. XXX) and cis- and trans-regulation (Fig. XXX).
\end{enumerate}

A 40-page tutorial for MQM explores, both the automated procedure, and the manual procedure of adding and removing 
cofactors, in an \emph{Arabidopsis thaliana} recombinant inbred line (RIL) metabolite (mQTL) dataset with 24 metabolites 
as phenotypes\cite{Fu:2007}. In addition, the tutorial visually explains the effects of data augmentation, cofactor 
selection, model selection, and tweaking of input parameters, such as cofactor significance. Genetic interactions 
(epistasis) are explored through effect plots, and an example is given of parallel computation. The tutorial is part 
of the software distribution of R/qtl and is available online.

\subsection{Conclusions}
MQM for R/qtl is a significant addition to the QTL mapper's toolbox. R/qtl provides the user with the most frequently 
used statistical analysis methods: single-marker analysis, interval mapping, Haley-Knott regression \cite{Haley:1992}, 
CIM \cite{Zeng:1994} and MQM \cite{Jansen:1994a}.  MQM has improved handling of missing data and allows more powerful 
and precise detection of QTL, compared to many other methods. Not only is this new implementation of MQM available in the
statistical R environment, which allows scripting for pipe-lined setups, it is also highly scalable through 
parallelisation and paves the way for high-throughput QTL analysis. With MQM, R/qtl is a free and high-performance 
comprehensive QTL mapping toolbox for the analysis of experimental populations. R/qtl now includes permutation strategies 
for determining thresholds of significance relevant for QTL and QTL hot spots; the first step towards causal inference and
network analysis.

\section{Classical phenotypes in a DesignGG experiment}
Perfect timing of germination is required to encounter optimal conditions for plant survival and it is the result of a complex 
interaction between molecular processes, seed characteristics and environmental cues. To detangle these processes we made use 
of natural genetic variation present in an Arabidopsis thaliana Bayreuth x Shahdara RIL population. For a detailed analysis of 
the germination response we characterized rate, uniformity and maximum germination and discussed the added value of such precise
measurements. The effects of after-ripening, stratification and controlled deterioration as well as the effect of salt (NaCl), 
mannitol, heat, cold and ABA with and without cold stratification were analyzed for these germination characteristics. Seed 
morphology (size, length) of both dry and imbibed seeds were quantified by using image analysis. For the overwhelming amount 
of data produced in this study we developed new approaches to perform and visualize high throughput QTL analysis. We show 
correlation of trait data, (shared) QTL positions and epistatic interactions. The detection of similar loci for different 
stresses indicate that often the molecular processes regulating environmental responses converge into similar pathways. 7 major 
QTL hotspots were confirmed using a HIF approach. QTLs co-locating with previously reported QTLs and well characterized mutants 
are discussed. A new connection between dormancy, ABA and a cripple mucilage formation due to a natural occurring mutation in the
MUM2 gene is proposed and this is an interesting lead for further research on the regulatory role of ABA in mucilage production 
and its multiple effects on germination parameters.

\subsection{Classical phenotypes - Background}
Colonizing plants are subject to a wide variety of environmental conditions. For successful adaptation to new habitats the timing 
of developmental transitions is especially important. Seed germination is one of these important transitions as it determines 
the seasonal environment experienced in further plant life \cite{Huang:2010}. Natural populations that develop under distinct 
environmental conditions may reveal genetic adaptation, which can be used to disentangle the signaling routes that are involved. 
Seed germination is described by three phases of water uptake. In phase I the seeds imbibes and reinitiates metabolic processes 
followed by a lag phase (phase II). Further water uptake results in protrusion of the radicle through the testa and endosperm 
(phase III). The moment of radicle protrusion through the endosperm is considered to be the moment of germination sensu stricto 
(Finch-Savage and Leubner-Metzger, 2006). To characterize the genetic variation of germination related traits we focused on the 
effect of the environment that a seed perceives during germination rather than the effect of the environment during maternal 
plant growth, which has been the subject of other studies (Gutterman, 2000; Dechaine et al., 2009; Elwell et al., 2011). Seed 
content (e.g. oil) is often used as commodity and modifications to the content can therefore be regarded as seed quality parameters 
as well. To prevent confusion we will use the term seed performance to indicate that the focus of our study was restricted to 
seed germination characteristics.

The production of high quality crop seed not only entails knowledge about maternal plant growth, harvesting and storage of seeds, 
but also of germination conditions \cite{Rivero-Lepinckas:2006}. To obtain better germination and field performance, many seed 
companies rely on enhancement methods, such as seed priming and coating and/or pelleting, but these methods are reaching their 
limits. Dissecting the molecular mechanisms underlying seed germination and its tolerance to the environment may unlock the full 
genetic potential and enable targeted breeding for seed performance. In this study we used a recombinant inbred line (RIL) population 
derived from two Arabidopsis thaliana ecotypes: Bayreuth (Bay-0) which originates from a fallow land habitat in Germany and Shahdara 
(Sha) which grows at high altitude in the Pamiro-Alay mountains in Tadjikistan (Loudet et al., 2002). The Bay-0 x Sha RIL population 
has been used in many previous studies to map QTL positions for root morphology (Loudet et al., 2005; Reymond et al., 2006), anion 
content (Loudet et al., 2003), nitrogen use efficiency (Loudet et al., 2003), cell wall digestibility (Barriere et al., 2005), 
carbohydrate content (Calenge et al., 2006), sulfate content (Loudet et al., 2007), leaf senescence (Diaz et al., 2006), 
morning-specific growth (Loudet et al., 2008) and cold-dark germination \cite{Meng:2008}. We have used the natural variation 
present in this RIL population to map the response of germination characteristics to environmental conditions to which a seed is exposed.

Freshly harvested viable Arabidopsis seeds often don’t germinate even when placed under conditions favorable for germination. This event, 
called primary dormancy, is shown to be subject to natural variation \cite{Bentsink:2010}. In many Arabidopsis ecotypes, this primary 
dormancy is released after a period of dry storage at room temperature. Another dormancy breaking treatment is cold stratification were 
seeds are imbibed in water and stored at 4°C in the dark for four days before putting them into optimal conditions for germination 
(Finch-Savage and Leubner-Metzger, 2006). Unfavorable conditions during seed germination may result in a changed rate or even failure 
of germination. In Arabidopsis, it has been shown that the responsiveness to temperature is closely related to the level of after-ripening 
(Tamura et al., 2006). High salt concentrations induce osmotic stress and ion toxicity resulting in both a delay and reduction of maximum 
germination \cite{Galpaz:2010}. Often, these different environmental stresses are interconnected and will cause osmotic and 
associated oxidative stress (Zhu, 2002; Chinnusamy et al., 2004). The plant hormone Abscisic Acid (ABA) plays a predominant role in 
plant responses to different environmental stresses and can activate various signal transduction pathways leading to a global change in 
transcription (Finkelstein et al., 2002; Xiong et al., 2002). Exogenous application of ABA during germination results in a distinction 
between testa and endosperm rupture. At certain concentrations the testa will rupture but germination sensu stricto (radicle protrusion 
through the endosperm) will be inhibited. This phenomenon, caused by reduced weakening of the endosperm cap, is the consequence of a 
complex interplay between ABA, GA and ethylene signals (Linkies et al., 2009). In this report, we determined germination sensu stricto 
for primary dormancy in freshly harvested seeds, germination of fully after- ripened seeds with and without a preceding cold stratification
period (see material and methods for conditions), and germination under various stress conditions (low/high temperature, salt/osmotic 
stress and ABA) to assess natural  variation in the Bay-0xSha RIL population. Additionally, seed morphology (size and length) and flowering 
time were phenotyped as they have been shown to be strong determinants of plant trait variation (Chiang et al., 2009; Orsi and Tanksley, 
2009; Elwell et al., 2011). We correlated these traits to our germination related traits to evaluate possible causality. In total this 
analysis resulted in 327 trait scores over different harvests. Evaluation of these high numbers of phenotypes demanded methods of QTL
analysis that extended beyond mapping of individual traits and that allowed comprehensive and comprehensible visualization.

Analysis of natural variation that is captured in well-defined recombinant inbred populations has shown to be a powerful tool to detect 
important loci that influence the traits under study \cite{Alonso-Blanco:2009}. To uncover the loci with genetic variation a statistical 
framework is needed. For this, any programming language can be used which supports statistics. In the life sciences the statistical language R 
is often the prime candidate. R is open source, contains the latest in statistical analysis methods and has a large community for help 
and support (http://www.r-project.org/). Furthermore, it has the R/qtl package \cite{Broman:2003}, which contains an array of different 
QTL mapping methods, including Single Marker Mapping, Interval Mapping and Multiple QTL Mapping (MQM) \cite{Arends:2010}. Although all 
possibilities to perform a detailed QTL analysis including data preprocessing and output formatting are present in R, it requires extensive
knowledge of the R-syntax to combine all necessary steps in a single analysis protocol that can loop through hundreds or thousands of traits. 

We present a script that can perform these tasks. This type of automated analysis combined with efficient data visualization is a necessary 
step to keep up with the increasing rate of biological data production. For using single trait mapping the effect of a certain treatment, 
e.g. germination at high temperature, must be corrected by the germination characteristics under control conditions. Here, we subtracted the 
observed germination under stress conditions from values for germination under control conditions. This correction can lead to complicated 
interpretation, especially when the environment under study affects loci with already strong effects under control conditions. Further, 
it can reduce statistical power due to summation of the error components. Therefore we performed an additionalanalysis using a QTL by 
environment (QTLxE) approach (Jansen et al., 1994; Malosetti et al., 2004; Moreau et al., 2004; Eeuwijk et al., 2006). Instead of considering 
individual responses, one can then treat the stress conditions as a set of environmental perturbations and evaluate a single trait (such as 
germination percentage). Because several environments are taken into account simultaneously, the statistical power to detect loci that are 
affected by several environments increases and interpretation becomes more intuitive as the need for correcting the stress response by the control
response is eliminated (Boer et al., 2007; Payne et al., 2011).

The Bay-0 x Sha RIL population consists of 420 lines that were genotyped in the F6. This relatively low degree of inbreeding provoked residual 
heterozygosity present at almost all genome positions. This residual heterozygosity can be used to confirm QTL positions, as it provides a 
possibility to study both parental alleles at the locus of interest in an elsewhere homozygous background (Tuinstra et al., 1997). In contrast to
conventional near isogenic lines (NILs) the genetic background of heterogeneous inbred lines (HIFs) consist of a mix of the two parental genomes. 
The availability of a genome wide set of HIF lines for the Bay-0xSha RIL population provides a fast and accurate mean to confirm QTL loci

\subsection{Classical phenotypes - Results}

\subsubsection{Single trait QTL mapping}
To evaluate the response of germination to a certain treatment, we first subtracted the observed germination at test conditions from germination 
at the proper control conditions. For example, the effect of NaCl on germination after cold stratification is determined by subtracting Gmax on 
NaCl from Gmax on water. This subtraction was reversed for the rate and uniformity parameters to correct the reversed nature of these parameters 
(e.g. slower germination results in a larger t10 and t50). Table 1 gives an overview of all corrections that have been applied. 

An analytical protocol was designed, using the popular R/qtl package of R to analyze trait data of recombinant inbred populations with the multiple 
QTL model approach \cite{Arends:2010}. When performing a detailed QTL analysis it is important that several steps are performed or checked. 
Missing genotypic data is imputed and a recombination frequency plot is generated (Figure 1A). In the next step, quality of the trait data is 
investigated. Outliers are detected and removed using a z-score transformation with a user defined threshold. As an extra control the results of 
MQM mapping were always compared to standard interval mapping, using the parametric model with Haley Knott regression (Haley and Knott, 1992) 
(Figure 1B). The whole genome additive effect was estimated based on the non-transformed data as half the difference between the phenotypic
averages for the two homozygotes (Figure 1C).

R/qtl MQM uses a backward elimination of cofactors. As a rule of thumb one can select a maximum of n-16 initial cofactors with this procedure (Jansen, 2008), with n
being the number of lines in the RIL population. In our script, a cofactor file can be provided with the selection of the initial cofactors. When no cofactor file is provided, the
analysis will be performed without cofactors resulting in an analysis comparable with the composite interval mapping (CIM) method. For the analysis of the Bay-0xSha population
we selected 39 out of 69 markers as possible cofactors. Cofactors were selected based on their quality (least amount of missing data or heterozygous status) and physical cM
position, attempting to obtain intervals of about 10 cM. Although the procedure allows the selection of all 69 markers as cofactors, this does not improve mapping and only lowers
statistical power due to the multiple testing correction in the permutation analysis. The provided cofactor file is used to perform automated backward elimination of cofactors.
Backward elimination is performed to remove cofactors that do not significantly contribute to the fit of the initial model. This is achieved by comparing Akaike’s information criterions
(AIC) of the different models (Jansen, 1993). Using the final selected QTL model, the mapping LOD scores are calculated for all genetic markers. Plots showing all significant
markers are produced automatically (figure 1D). We have used the procedure described to map all 327 individual measurements but to enhance
readability we only show average values for each trait (94 traits)

\subsubsection{QTLxQTL Interactions}
Epistatic interactions between QTL can help to elucidate meaningful co-localizations and will enable an efficient design of follow up experiments. Besides the
visualization of the epistatic interactions per trait (Figure 1F) our script creates an output that can help to visualize all detected epistatic interactions in a single plot. This output file
in sif format summarizes all detected epistatic interactions (Figure 5, Supplemental file "Interactionnetwork.cys"). Among others, clear hotspots of epistatic interactions between
QTL loci on chromosome 3, 4 and 5 (resp. ATHCHIB2 + MSAT332, MSAT435 and MSAT520037 + MSAT519) were observed for germination on salt (yellow lines) and
dormancy (blue lines). Next to the importance of detecting possible interacting loci this QTLxQTL analysis provides additional arguments for co-locating QTL to be of similar
genetic origin. Overall, the creation of this type of summarizing figures is greatly facilitating the interpretation of large datasets.

\subsubsection{QTLxEnvironment interaction}
To obtain a parameter for the response, we had to correct all values with their proper control condition values. This sometimes led to complex interpretation, which can
be circumvented by using the non-corrected germination parameters and model them over the various environmental conditions that were tested. Because several
environments are taken into account simultaneously the statistical power to detect loci that are affected by several environments increases and interpretation becomes more
intuitive as the need for correcting the stress response by the control response is eliminated. By using this approach the sensitivity of a specific QTL for environmental
conditions can be determined for each separate germination parameter. Details about the procedure are described in Material and Methods. Results are summarized in Figure 6.
The final model P-value profiles (top panel, Figure 6) clearly show the great consistency between the 5 germination parameters that we measured. However, a closer look also
reveals loci that are affecting different germination curve parameters. For example, the QTL on top chromosome 5 is not detected by measuring maximum germination but is
well defined when using t50 or t10 as parameter. As expected, the parameter AUC (Area Under the Curve) is outperforming the others as it represents a combined value for
maximum germination percentage, rate and uniformity. For comparison of the environment-specific QTL effects for the 5 different germination parameters (5 lower
panels, Figure 6) the effects could be compared with germination under control conditions. For example, after ripened seeds without stratification (AR.NS) can guide as
reference for the stress treatments (AR.NS.ABA, AR.NS.CD, AR.NS.Cold, AR.NS.Heat, AR.NS.Mannitol, AR.NS.NaCl). The same analogy holds true for after ripened seeds
without stratification (AR.NS) and freshly harvested seeds without stratification (Fresh.NS). In this way stress specific QTLs on chromosome II and top chromosome III
can easily be identified. Interestingly, some QTLs, including germination at low temperature (top chromosome I) and germination in the presence of exogenous ABA
(bottom chromosome V) displayed opposite effects on germination when compared to the other treatments. In Table 3 the environmental specific effect sizes are summarized
for the major loci. A complete overview of effect sizes and explained variances for all detected loci can be found in Supplemental table 6. 

\subsubsection{QTL confirmation}
Taking advantage of the residual heterozygosity present in the F6 generation of the Bay-0xSha population, combined with the large population size, 
we were able to confirm several QTL following the heterogeneous inbred family (HIF) approach. In short, RIL lines which are heterozygous at the 
locus of interest were selected in the next generation for lines homozygous for both parental alleles. These “families” are near isogenic lines 
(NIL) which can be used to confirm the observed allelic effects (Figure 7A). We applied this strategy for 7 of the major QTL that we detected in 
this study and tested the 5 germination parameters for 11 different conditions. For a single parameter (Gmax) and a single HIF (line HIF103) the 
analytical procedure is summarized in figure 7B. Traits that could be confirmed by one or several HIF lines are indicated in Table 4. An overview
of all HIF results can be found in Supplemental table 7. We detected a vast QTL for imbibed seed size at the bottom of chromosome 5, which could 
be confirmed by the use of HIF103. Upon imbibition seeds swell due to rapid water uptake and possibly because of the expansion of the inner 
mucilage layer. In Sha, which is a natural mutant for the MUM2 gene \cite{Macquet:2007}, this swelling did not occur. Also the HIF lines at the
MUM2 position showed a clear difference in swelling phenotype which was still significant 24 hours after imbibition (Figure 8).

\subsection{Classical phenotypes - Discussion}
When analyzing large (RIL) populations, it is hardly feasible to manually count all germination experiments several times a day to obtain germination curves. Therefore,
previous studies mostly restricted to counting end-point germination \cite{Quesada:2002, Alonso-Blanco:2003, Clerkx:2004, Laserna:2008, Meng:2008, Bentsink:2010, Galpaz:2010, Vallejo:2010}. 
A germination curve allows QTL mapping under conditions where rate and uniformity are
delayed, but maximum germination is not affected. Therefore, we used the Germinator package (Joosen et al., 2010) that enabled measurement of cumulative germination data
and extracting 5 germination parameters that describe the resulting germination curve. In the present study we describe several germination QTLs that were not detected before in
the Bay-0xSha population. We observed interesting co-localizations for several germination traits and identified the loci that show large effect epistatic interactions.
Among these were new loci and loci similar to the ones already found in other RIL populations as summarized in Table 4 for the major identified QTL loci.

\subsubsection{Dormancy}
Primary dormancy has been studied extensively in various RIL populations \cite{Bentsink:2010}. These authors quantified primary dormancy with the DSDS50
parameter (days of dry storage to reach 50\% germination), which is a good measure for after-ripening related dormancy breaking. Although we only compared the germination
characteristics of freshly harvested seeds with those of after ripened seeds and fresh seeds with and without stratification, we detected large genetic variation. Both dormancy
breaking treatments showed strong QTL at positions 3-2, 4-1 and 5-2, co-locating with DOG6, DOG18 and DOG1, respectively (Table 4).
DOG18 was not detected in a LerxSha population and showed a stronger dormancy in Ler as compared with An-1, Fei-0 and Kas-2 \cite{Bentsink:2010}. We detected stronger 
dormancy in Sha as compared to Bay-0 at the DOG18 locus. This suggests that both Ler and Sha contain an allele of similar strength which is stronger when compared to An-1, Fei-0, 
Kas-2 and Bay-0. Remarkably, for both the DOG6 and DOG18 location the sensitivity to ABA was higher in Bay-0, whereas dormancy was deeper in Sha, which resulted in a directional
change of the QTL effect. The more dormant Sha parent contains higher initial ABA levels (supplemental table 8) and apparently, after-ripening and stratification reduce the
ABA sensitivity to a greater extent as compared to the Bay-0 parent. This effect was not observed for the DOG1 locus. Further, we identified a strong effect of the dormancy-
breaking treatments on the initiation (t10) and rate (t50) of germination at the bottom of chromosome 5 (marker MSAT519, 85 cM). The same was observed for germination on
mannitol and germination at higher temperature. A QTL with opposite effect at this position was found for germination on ABA. Interestingly, these co-located with a QTL
found for imbibed seed size. 

\subsubsection{Water uptake}
Initiation and rate of germination are highly influenced by the overall water potential of the seed. The mucilage layer surrounding the seed appears to play an
important role in the process of water uptake \cite{Penfield:2001}. Sha is a natural mucilage mutant due to a mutation in the MUM2 gene, which changes the hydrophilic
potential of rhamnogalacturonan I \cite{Macquet:2007}. Although mucilage has been reported to be dispensable for germination and development under lab conditions
(Arsovski et al., 2010), a link with germination under reduced water potential conditions was shown by \cite{Penfield:2001}. They showed reduced maximum germination of a
mucilage-impaired mutant only on osmotic PEG solutions. In our study, other traits that co-located on the MUM2 locus were delayed initiation and rate of germination on osmotic
mannitol solution but also on water, which clearly shows the advantage of determining a detailed germination curve. We also observed a very strong QTL for swelling of the seed
in the first hours of imbibition (imbibed seed size) at the MUM2 location. Interestingly, exogenous ABA can be used to stimulate mucilage production and ABA-1 mutants are
affected in mucilage production (Karssen et al., 1983). This indicates a regulatory role of ABA in mucilage production and fits with our observation of the co-localization of a QTL
for initiation and rate of germination with a QTL with opposite effect for ABA sensitivity. Therefore, we hypothesize that Sha has a slower initiation and rate of germination,
combined with reduced ABA sensitivity due to its mutation in the MUM2 gene. This observation may open new research strategies to define the regulatory role of ABA in
mucilage production and its multiple effects on germination parameters.

\subsubsection{Salt, heat and ABA}
At the top of chromosome I, underlying marker F12M12, we detected a strong QTL for maximum germination in the presence of 100 mM NaCl or 0.5μm ABA. A similar
locus has been identified and fine-mapped in a LerxSha population \cite{Ren:2010}. They identified a premature stop codon in the Response to ABA and Salt 1
gene (RAS1; At1g09950) in Sha that led to a truncated protein and showed its role as a negative regulator of salt tolerance during seed germination and early seedling growth by
enhancing ABA sensitivity. Here we show that a similar locus is also inferring tolerance to germination at 30°C. This suggests an additional role for the RAS1
gene. Increased heat tolerance due to modulation of ABA sensitivity has been shown before for other loci, (Argyris et al., 2008; Lee et al., 2010). Interestingly, our present study showed a strong
effect of stratification which resulted in a strong reduction of significant linkage for NaCl, heat and ABA sensitivity at the F12M12 locus. A specific QTL for germination on NaCl
preceded by a cold stratification period was found at the middle of chromosome I (marker T27K12). Also at this locus we found colocation with sensitivity for germination on ABA
after stratification. Further fine-mapping at this locus might help to elucidate the effect of stratification on ABA mediated abiotic stress tolerance, as well as the apparent overlap of
dormancy and stress responses. Especially interesting is QTL 5-1 (Table 4, Figure 6) which mainly influences rate and initiation of germination. We detected this QTL for t50 in after 
ripened seeds with stratification treatment, but also for t10 and t50 for germination on salt, regardless of a preceding cold stratification and for maximum germination after an accelerated aging
treatment. One of the genes underlying this QTL interval is a nicotinamidase gene (NIC2, At5g23230), the mutant of which has retarded germination and impaired germination potential 
(Hunt et al., 2007). These authors suggested that NIC2 is normally metabolizing nicotinamide during moist chilling or after-ripening, which relieves inhibition of poly(ADP-ribose) 
polymerase (PARP enzyme) activity and allows DNA repair to occur prior to germination. Both accelerated aging and germination under salt stress conditions might require optimal 
functioning of this DNA repair mechanism. Further research is needed to determine whether NIC2 is causal for this QTL.

Detection of epistatic interactions in genetic studies can enhance the understanding of underlying molecular mechanisms. Recently, \cite{Galpaz:2010} showed strong epistasis 
in the genetic network controlling germination under salt stress in Arabidopsis. Due to careful dissection of the epistatic relationships they were able to show that three detected 
QTL rely on the presence of a Columbia allele at a QTL on top of chromosome 1. This observation led to the hypothesis that RAS1 \cite{Ren:2010} functions as a switch of the genetic 
network by regulating the expression of the other QTL. In another study it was found that epistasis significantly influences both fitness and germination in Arabidopsis \cite{Huang:2010} 
and novel allele combinations were identified that resulted in higher fitness. In our study we detected clear hotspots of epistatic interactions between QTL loci on chromosome 3, 4 and 5 (ATHCHIB2,
MSAT332, MSAT435, MSAT520037 + MSAT519, respectively). This observation strengthens the hypothesis that some of the traits with strong QTL co-localizations indeed rely on the same underlying genetic networks.
Conclusion

\subsubsection{In conclusion}
We analyzed natural variation for many seed germination characteristics and showed their correlation, (shared) QTL positions and epistatic interactions, using a high-
throughput phenotyping approach and subsequent high-throughput QTL mapping. Using the HIF approach, confirmation of some major QTL hotspots was demonstrated, which 
allows a fast but solid confirmation of a QTL position. Together with results from several other studies focusing on genetic variation in seed traits, this study has 
generated an extensive QTL database for Arabidopis and proposed a method of analysis to visualize the genetic landscape of seed performance. This database is a solid 
resource for further study. For most of the found loci in this and other studies further characterization, and in most cases fine mapping, must be undertaken to elucidate 
the causal molecular mechanisms. Further, we have designed a free available analysis protocol to perform detailed high-throughput QTL analysis based on the R/qtl MQM 
routine. In this era of large-scale phenotyping we regard a detailed analysis of QTL, QTLxQTL and QTLxEnvironment interaction as indispensable steps to allow visualization 
and interpretation of multiple traits.

\section{Metabolites in a DesignGG experiment}
\lipsum

\chapter{High throughput data analysis}

\emph{The major challenge in bioinformatics comes from the huge amount of data collected and the multitude of technologies used. We developed 
the xQTL workbench system\cite{Arends:2012} to store large amounts of phenotype and genotype data in the XGAP\cite{Swertz:2010a} 
dataformat. xQTL workbench can easily be adapted to a multitude of input and storage formats using the MOLGENIS\cite{Swertz:2004} 
generator. xQTL workbench uses the power of distibuted computing to allow massive parrallel QTL analysis and allows new analysis 
tools to be quickly added to the system, an application of the xQTL system is found in the WormQTL database\cite{Snoek:2012}.}

\null
\vfill

\begin{myexampleblock}{Originally published as:}
  \authors{Morris A Swertz, Martijn Dijkstra, ..., Danny Arends, George Byelas, et al}\\
  \emph{The MOLGENIS toolkit: rapid prototyping of biosoftware at the push of a button}\\
  \bold{BMC Bioinformatics} (2010)\\\\

  \authors{Morris A Swertz, K Joeri van der Velde, Bruno M Tesson, ..., Danny Arends, et al}\\
  \emph{XGAP: a uniform and extensible data model and software platform for genotype and phenotype experiments}\\
  \bold{Genome Biology} (2010)\\\\

  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  \emph{xQTL workbench: a scalable web environment for multi-level QTL analysis}\\
  \bold{Bioinformatics} (2012)\\\\

  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, et al.}\\
  \emph{WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp}\\
  \bold{Nucleic Acids Research} (2012)
\end{myexampleblock}

\newpage

\section{Reusable software (MOLGENIS)}
\subsection{MOLGENIS - Background}
High throughput technologies have boosted biological and medical research and the need for software infrastructures to 
manage and process the large datasets produced is widely accepted [1-3]. Bioinformaticians are under continuous pressure 
to both tackle the complexity and diversity of new biological systems and analytical methods and to translate these 
quickly into flexible informatics infrastructures, while keeping up with the unpredictable evolution of molecular 
biotechnologies and the increasing scale of experiments. While standardization of tools and data formats in open source 
projects like the Generic Model Organism Database, GMOD [4], and the Open Bioinformatics Foundation, OBF [5], have been 
indispensable in reducing the development efforts needed via reusable and easy to integrate components, new research 
must also be quickly accommodated, for which efficient software variation mechanisms are needed.

We present the evolution of MOLGENIS into a generic, model-driven toolkit for the rapid generation of bespoke, 
data-intensive biosoftware applications [10]. We demonstrate step-by-step how bioinformaticians can use a domain-specific 
language to efficiently model the biological details of their particular biological system, and use MOLGENIS software 
generation tools to automatically generate a web application tailored to the experiments of their biologists, building 
on reusable components. Next, we evaluate the results of these methods in the development of a range of MOLGENIS applications 
[9,11-15], that is, software applications generated using the MOLGENIS toolkit. We found up to 30 times efficiency improvement 
compared to hand-writing software, while providing a richness of features practically unfeasible to produce by hand but not 
yet provided by related projects. We conclude by inviting the bioinformatics community to add more MOLGENIS models, components 
and generators to quickly generate all the software infrastructures biologists want to have.

\subsection{MOLGENIS - Methods}

The MOLGENIS toolkit is based on the method of model-driven development which emerged in the 1990s from the computer industry. 
The key to success is the clear scope of the toolbox (i.e., what family of software applications should be produced with it) 
and separating which features should be fixed (e.g., reusable components common to all MOLGENIS applications) and which 
features should be variable (i.e. modeled and generated per MOLGENIS application instance), a process known as domain 
analysis [16]. Below we discuss MOLGENIS' initial domain analysis, its modeling language, generators and reusable components. 

\subsubsection{Domain analysis}

Table 1 summarizes the initial set of features we required from MOLGENIS information systems when we started; it explains 
why these features are indeed required, and describes what parts of the features are common and variable over experiment 
databases. To obtain this picture, we analyzed 20 existing microarray databases next to many requirements interviews, see 
Table 1 in [9].

Table 1. Common and variable features of MOLGENIS information systems.

The second step was to implement the common and variable parts, which we started with a prototype. Here we applied the 
don't repeat yourself principle (DRY) [17]: every piece of design knowledge must have a single, unambiguous, authoritative 
representation. We therefore searched through the prototype software code. If we found identical pieces then we put them 
into the library of reusable components. If we found very similar pieces of software code, we put the common parts into a 
generator and the variation points into the modeling language. In each subsequent step we evolved the MOLGENIS generator, 
only incorporating new functionality when we repeatedly needed it.

During the next six years of using the MOLGENIS generator we added numerous functions and optimizations, such as filters for 
the data, viewing data as a 'matrix', downloading data as CSV files, enabling programming interaction via R and web services, 
and so on. The generators ensure that 'old' MOLGENIS application variants can benefit from these improvements: when a MOLGENIS 
instance is re-generated, these improvements are automatically integrated into the new version.

\subsubsection{Modeling language}

Figure 2 shows how a custom MOLGENIS application can be defined in a single file. The file is written in MOLGENIS' modeling 
language. This enables compact specification of what experiment database is needed, i.e., to declare how an experiment is 
organized in terms of data types and their relationships and how these data are to be shown on the screen. Figure 2 shows 
the following features: Three data entities: (1)Experiment, Sample and Hybridization; the Experiment entity has six fields, 
including ID, Medium and Stress (because it needs to administrate microbe experiments). To minimize the modeling work we 
choose sensible defaults in the domain-specific language, a principle known as convention over configuration: each field has 
to be set to a value by the researcher unless specified to be nillable(2); field can be edited (updated) unless specified to be 
read only(3); each field is default of type 'string' (a variable character string of length 255) unless otherwise specified to 
e.g., 'decimal'(4); and fields can be defined as having a relation to fields in other entities via a cross-reference (xref)(5). 
The user interface consists of a plugin(6) that renders the MOLGENIS header and tool menu; one user interface form(7) to control 
Experiments, with a sub menu(8), consisting of two child forms for Samples and Hybridizations. Child forms are automatically 
linked to the parent form based on cross references, e.g., the field 'Experiment' of 'Sample' references to the 'ID' of an 
'Experiment'(9). By default, forms show each entity as one-record-per-screen unless specified as a list(10). The modeling 
language includes advanced object-orientation features like inheritance, as well as extensive help to document your model 
(not shown).

One can think of MOLGENIS' modeling language as a 'domain-specific language' (DSL) that is optimized to efficiently express 
a particular problem, task or area [18,19], in this case to compose biosoftware infrastructures. The level of abstraction is 
raised, so no lengthy, technical or redundant details on how each feature should be implemented in general programming 
languages have to be given [20,21]. Examples of other domain-specific languages include R/Splus for statistics, MatLab for 
mathematics, SQL for databases, HTML for layouting, and now MOLGENIS' modeling language for biological software infrastructures.

In most cases, knowledge of the DSL is all that is needed to produce a custom MOLGENIS application variant. The domain-specific 
language was implemented using XML so that model files can be edited using off-the-shelf XML editors. However, you may want 
to include hand-programmed components into a particular MOLGENIS instance. For example, for the eXtensible Genotype And 
Phenotype (XGAP) database application of MOLGENIS [11], we developed a 'MatrixViewer' that builds on the generated components, 
which saved us the work of writing the plug-in from scratch. This requires a model sentence that points to the 'plug-in' 
(allowing it to be seamlessly integrated) as well as hand-programming of the plug-in itself.

\subsubsection{Reusable components}

Each MOLGENIS application follows the widely accepted three-layered architecture design of web applications. Figure 3 summarizes 
some of MOLGENIS' reusable components and their variation mechanisms. MOLGENIS' reusable components provide building blocks with 
a modular structure, which allows them to be assembled in diverse combinations, similar to prefabricated houses that are built 
from modular walls instead of bricks. Some building blocks are semi-finished and need to be 'completed' before use (which is 
automated in MOLGENIS via the generators and inheritance). We based the design of MOLGENIS on industry-proven design patterns 
from the 'patterns for enterprise application architecture' (PEAA), a catalog of proven solutions for software design problems 
that we used as a guideline [22]. The logic of the reusable components is implemented using Java (http://java.sun.com); the 
HTML layout for the user interface is encoded in Freemarker templates (http://freemarker.sourceforge.net/); and the database 
back-end using MySQL, PostgreSQL or HSQLDB.

\subsubsection{Generators}

The generators are compact specifications of how each database feature should be implemented. The MOLGENIS toolkit now has 
over 20 generators, but normal users will never need to take a look inside. However, for readers wanting to create their own 
generators, Figure 4 provides an example of the simple, text-based, generators we use. Each generator consists of two files: 
a Freemarker template that describes the code to be generated (similar to that shown in Figure 4a) and a Java 'Generator' 
class that controls the generating process. A new generator can be developed as follows: first write some examples of the 
desired programs by hand, where possible using similar patterns (see Figure 4b) and mark which parts are variable between 
them. Then copy one of these examples into a generator template (text file) and replace all variable parts with 'holes' that 
are to be filled by the code generator based on parameters from DSL (see Figure 4a). At each generation, the template is then 
automatically copied and the 'holes' filled, based on parameters described in the domain-specific language, saving much 
laborious manual work. 

\subsection{MOLGENIS - Results}

To start generating your own MOLGENIS application, you can download a ready-to-use 'workspace' from http://www.molgenis.org, 
which can be edited using the commonly used Eclipse integrated development environment (IDE) tool (http://www.eclipse.org). 
Extensive manuals are available to help install the Java, MySQL, Tomcat and Eclipse software needed and to learn how to walk 
through the Eclipse workspace to edit models and generate and run MOLGENIS instances; most new users can complete this part 
in about three hours. Below we summarize the output you can expect as well as recent experiences from using this toolbox. 
Detailed examples on how these features can be used to support actual microarray or genetical genomics experiments can be 
found in [11,14,15].

\subsubsection{Expected output}

After completing a MOLGENIS model and running the generator as described above, you have a ready-to-use software application. 
Figure 5 summarizes the features you get when running the generated result as a web application: a fully functional system 
where researchers can upload, manage, browse and query their biological data that conform to the model, optionally enhanced 
with analysis tools to explore and annotate (depending on the plug-ins).

An important feature is human readable and printable documentation of your model, including a graphical overview showing 
relationships in UML(1) which is of great use when still designing and discussing the model in a team. The next step is 
typically using the web user interface to populate and test your application with real data(2). To enable batch loading 
from a spreadsheet application such as Excel, the system comes with a tab-delimited import/export tool tailored for your 
data which you can use from the user interface as well as via a command-line tool; i.e., the headers of your Excel file 
have to match the fields you have defined in the model, (3). In our experience, most computational biologists greatly 
appreciate the use of the R interface to load, analyze and re-store data from within the R statistical environment(4) with 
web services to connect to workflow tools(5). Finally, advanced programmers may want to customize the layout or integrate 
their own scripts into the user interface, that is, create plug-ins that are seamlessly integrated with the generated 
software(6). Typical examples here are the integration of R scripts that produce graphical overviews of the data, enabling 
them to be run by non-technical research colleagues. Alternatively, you can use SOAP, REST and RDF interfaces for 
integration with workflow tools like Taverna, or for use with commonly used JavaScript frameworks like jQuery to create 
'Web 2.0' interactive websites. When satisfied with your MOLGENIS system, it can be shared as a simple JAR executable 
using an embedded web server, or as a WAR file that can be run on public web servers.

\subsubsection{Applications}

Since the earliest MOLGENIS application [9], we have successfully evaluated use of the MOLGENIS toolkit to build a wide 
range of biomedical applications [11-15], ranging from sequencing to proteomics, including:

\begin{enumerate}\itemsep1pt
\item \italic{XGAP} - an eXtensible Genotype And Phenotype platform [11] for systems genetics (GWAS, GWL) to store all kinds of *omics 
data ranging from genotype to transcript and protein data. XGAP comes with plug-ins to view large data matrices and run 
processing tools on a cluster. See http://www.xgap.org
\item \italic{Pheno-OM} - to integrate any phenotype data from locus-specific annotations to rich biobank cohort reports with the help 
of the OntoCAT ontology toolkit to create semantic mappings between related data items [23]. See http://www.ebi.ac.uk/microarray-srv/pheno
\item \italic{FINDIS} - a mutation database for monogenic diseases belonging to the Finnish disease heritage. See http://www.findis.org/
\item \italic{HGVBaseG2P} - the data management and curation interface complement for HGVbaseG2P, a central database of genotype 
to phenotype association studies [12]. See http://www.hgvbaseg2p.org
\item \italic{MAGETAB-OM} - a microarray experiment data platform based on the MAGE-TAB data format standard to create a local 
microarray repository that is compatible with the public ArrayExpress and GEO repositories. See http://magetab-om.sourceforge.net/ 
\item \italic{NordicDB} - the database of high-density genome-wide SNP information from 5,000 controls originating from Finnish, 
Swedish and Danish studies [13]. See http://www.nordicdb.org
\item \italic{DesignGG} - a web tool to optimally design such genetical genomics experiments [14]. See http://gbic.biol.rug.nl/designGG/
\end{enumerate}

More MOLGENIS applications can be found at http://www.molgenis.org. Each of these MOLGENIS projects reported major 
benefits from the short cycle from model to running system to enable quick evaluation (500 lines of model XML replaces 
15,000 lines of programming code) and use of the batch loading of data to evaluate how the newly built system works with 
real data. More often than not, MOLGENIS helped in finding inconsistencies in existing data that would otherwise have gone 
unnoticed, leading to experimental errors. In our experience, a typical MOLGENIS generator run gives you about 90\% of 
the application that is desired 'for free', with the remaining 10\% typically filled in using plug-ins that are written 
by hand. The MOLGENIS toolkit has also been used to extend or replace existing software applications: the ExtractModel 
tool allows you to generate a MOLGENIS application from an existing database, which can then be run side-by-side with 
code developed previously, providing the best of both generated and hand-written worlds.

\subsubsection{Richness of features}

MOLGENIS provides a richness of features not yet provided by other projects: BioMart [10,24] and InterMine [25] generate 
powerful query interfaces for existing data but are not suited for bespoke data management; Omixed [26] generates 
programmatic interfaces onto databases, including a security layer, but lacks user interfaces; PEDRO/Pierre [27] generates 
data entry and retrieval user interfaces but lacks programmatic interfaces; and general generators such as AndroMDA [28] and 
Ruby-on-Rails [29] require much more programming and configuration efforts compared to tools specific to the biological 
domain. Turnkey [30] seems to come close to MOLGENIS, having GUI and SOAP interfaces but lacks auto-generation of R 
interfaces and file exchange format. 

\subsection{MOLGENIS - Conclusions}

In a recent perspective paper [1] we evaluated the general benefits and pitfalls of model-driven development, such as 
the ability to develop infrastructure in short cycles to get the application right, ensuring developers and biologists 
are thinking along the same lines and increasing quality and functionality for all. We further evaluated applying this 
method to both microarray and genetical genomics experiments [9], [11].

Here we have presented MOLGENIS in detail and reported the results of using this method against a wider range of 
applications. We conclude that using model-driven methods enables bioinformaticians to build biological software 
infrastructures faster than before, with the additional benefit of much easier sharing of models, data and components. 
Much less time is spent on customizing and gluing together individual components. The result is of higher quality 
because fewer incidental errors creep into the applications as a consequence of the automated procedures; best practices 
are applied instead of reinvented. And you do not need heavy-weight technology to implement a model-driven generator: 
simple text-based templates suffice to create biological software generators.

As a next step we want to expand the MOLGENIS toolkit to also generate data processing tools, including user friendly 
interaction, building on other 'model-driven bioinformatics' projects in this area, such as Taverna [6] to model/execute 
analysis workflows and Galaxy [7] to generate user interfaces for processing tools. We hope that many bioinformaticians 
will enforce our open source efforts and share their best models, plug-ins and generators at http://www.molgenis.org, 
so that, in time, every biologist may find a MOLGENIS variant that suits his/her needs.

\subsubsection{MOLGENIS - Availability and requirements}

Project name: MOLGENIS
Project homepage: http://www.molgenis.org
Operating systems: Windows, Linux, Apple
Programming language: Java JRE 1.5 or higher
Other requirements: MySQL or Postgresql, Tomcat or other J2EE container
License: GNU Lesses General Public License version 3 (GNU LGPLv3)
Any restrictions to use by non-academics: No

\section{Storing extensible data (XGAP)}
We present an extensible software model for the genotype and phenotype community, XGAP. Readers can down-
load a standard XGAP (http://www.xgap.org) or auto-generate a custom version using MOLGENIS with program-
ming interfaces to R-software and web-services or user interfaces for biologists. XGAP has simple load formats for
any type of genotype, epigenotype, transcript, protein, metabolite or other phenotype data. Current functionality
includes tools ranging from eQTL analysis in mouse to genome-wide association studies in humans.

\subsection{XGAP - A minimal and extensible object model}

We developed the XGAP objectmodel to uniformly capture the wide variety of (future) genotype and phenotype data, building 
on generic standard model FuGE (Functional Genomics Experiment) [38] for describing the experimental 'metadata' on samples, protocols and
experimental variables of functional genomics experiments, the OBO model (of the Open Biological and Biomedical Ontologies foundry for use of standard and
controlled vocabularies and ontologies that ease integration [39], and lessons learned from previous, profiling technology-specific modeling efforts [29].

Figure 1b shows the core components of a genotype-to-phenotype investigation: the biological subjects studied (for example, human individuals, mouse strains,
plant tissue samples), the biomolecular protocols used (for example, Affymetrix, Illumina, Qiagen, liquid chromatography-mass spectrometry (LC/MS), Orbitrap,
NMR), the trait data generated (usually data matrices with, for example, phenotype or transcript abundance data), the additional information on these traits (for
example, genome location of a transcript, masses of LC/MS peaks), the wet-lab or computational protocols used (for example, MetaNetwork [22] in the case of QTL and
network analysis) and the derived data (for example, QTL likelihood curves).

We describe these biological components using FuGE data types and XGAP extensions thereof. Investigation binds all details of an investigation. Each investigation
may apply a series of biomolecular [40] and computational [20-23] Protocols. The applications of such Protocols are termed ProtocolApplications, which in the case
of computational Protocols may require input Data and will deliver output Data.These data have the form of matrices, the DataElements of which have a row and a
column index. Each row and column refers to a DimensionElement, being a particular Subject or a particular Trait, Table 2 illustrates the usage of these core data
types. Figure 1a, c shows how the XGAP model can be extended to accommodate details on particular types of subjects and traits in a uniform way. A Trait can be a
classical phenotype (for example, flowering - the flowering time is stored in the DataElement) or a biomolecular phenotype (for example, Gene X it's transcript 
abundance is stored in the DataElement). A Trait can also be a genotype (for example, Marker Y is a genomic feature observation that is stored in the DataElement).
Genomic traits such as Gene, Marker and Probe all need additional information about their genome Locus to be provided. Similarly, a Subject can be a single Sample
(for example, a labeled biomaterial as put on a microarray) and such a sample may originate from one particular Individual. It may also be a PairedSample when 
biomaterials come from two individuals - for example, if biomaterial has been pooled as in two-color microarrays. An individual belongs to a particular Strain
.When new experiments are added new variants of Trait and Subject can be added in a similar way. Table 3 illustrates the generic usage of these extended data types.
Several standard data types were also inherited from FuGE to enable researchers to provide 'Minimum Information' for QTLs and Association Studies such as defined 
in the MIQAS checklist [41] - a member of the Minimum Information for Biological and Biomedical Investigations (MIBBI) guideline effort [42]

\subsection{XGAP - Conclusions}
In this paper we report a minimal and extensible data infrastructure for the management and exchange of genotype-to-phenotype experiments, including an object
model for genotype and phenotype data (XGAP-OM), a simple file format to exchange data using this model (XGAP-TAB) and easy-to-customize database software
(XGAP-DB) that will help groups to directly use and adapt XGAP as a platform for their particular experimental data and analysis protocols. We successfully 
evaluated the XGAP model and software in a broad range of experiments: array data (gene expression, including tiling arrays for detection of alternative splicing, 
ChIP-on-chip for methylation, and genotyping arrays for SNP detection); proteomics and metabolomics data (liquid chromatography time of flight mass spectrometry 
(LC-QTOF MS), NMR); classical phenotype assays [8,11,13,15,49,50,58,59]; other assays for detection of genetic markers; and annotation information for panel, gene, 
sample and clone. Non-technical partners successfully evaluated the practical utility by independently formatting and loading parts of their consortium data: 
EU-CASIMIR (for mouse; Table 7), EU-GEN2PHEN (for human; Table 7), EU-PANACEA (for C. elegans) and IOP-Brassica (for plants). A public subset of these data sets 
is available for download at [51]. When needed we could quickly add customizations to the model, building on the general schema, and then use MOLGENIS to generate 
a new version of the software at the push of a button, for example, to support NMR methods as an extended type of Trait[60]. Furthermore we successfully integrated 
processing tools, such as a two-way communication with R/QTL [24] enabling QTL mapping on XGAP stored genotypes and phenotypes with QTL results stored back into XGAP.

Based on these experiences, we expect use of XGAP to help the community of genome-to-phenome researchers to share data and tools, notwithstanding large variations 
in their research aims. The XGAP data format can be used to represent and exchange all raw, intermediate and result data associated with an investigation, and an 
XGAP database, for instance, can be used as a platform to share both data and computational protocols (for example, written in the R statistical language) associated 
with a research publication in an open format. We envision a directory service to which XGAP users can publish metadata on their investigations either manually or 
automatically by configuring this option in the XGAP administration user interface. This directory service can then be used as an entry point for federated querying 
between the community of XGAPs to share data and tools. Groups that already have an infrastructure can assimilate XGAP to ease evolution of their existing software.
Next to their existing user tools, they can 'rewire' algorithms and visual tools to also use the MOLGENIS APIs as data backend. Thus, researchers still have the
same features as before, plus the features provided by the generated infrastructure (for example, data management GUIs, R/API) and connected tools (for example, R
packages developed elsewhere). Moreover, much less software code needs to be maintained by hand when replacing hand-written parts by MOLGENIS-generated parts, 
allowing software engineers to add new features for researchers much more rapidly. We invite the broader community to join our efforts at the public XGAP.org wiki, 
mailing list and source code versioning system to evolve and share the best XGAP customizations and GUI/API 'plug-in' enhancements, to support the growing range 
of profiling technologies, create data pipelines between repositories, and to push developments in the directions that will most benefit research

\section{High throughput data analysis (xQTL workbench)}
xQTL workbench is a scalable web platform for the mapping of quantitative trait loci (QTLs) at multiple levels: 
for example gene expression (eQTL), protein abundance (pQTL), metabolite abundance (mQTL) and phenotype (phQTL) data. 
Popular QTL mapping methods for model organism and human populations are accessible via the web user interface. 
Large calculations scale easily on to multi-core computers, clusters and Cloud. All data involved can be uploaded 
and queried online: markers, genotypes, microarrays, NGS, LC-MS, GC-MS, NMR, etc. When new data types come available, 
xQTL workbench is quickly customized using the MOLGENIS software generator.

Modern high throughput technologies generate large amounts of genomic, transcriptomic, proteomic, and metabolomic 
data. However, existing open source web-based tools for QTL analysis, such as webQTL\cite{Wang:2003} and 
QTLNetwork \cite{Yang:2008}, are not easily extendable to different settings and computationally scalable 
for whole genome analyses. \xqtlwb makes it easy to analyse large and complex datasets using state-of-the-art QTL 
mapping tools and to apply these methods to millions of phenotypes using paralellized 'Big Data' 
solutions\cite{Trelles:2011}. \xqtlwb also supports storing of raw, intermediate and final result data, and 
analysis protocols and history for reproducibility and data provenance. Use of MOLGENIS\cite{Swertz:2010b} 
helps to customize the software. All is conveniently accessible via standard Internet browsers on Windows, 
Linux or Mac (and using Java, R for the server).

\subsection{xQTL workbench - Features}
\xqtlwb provides visualization of QTL profiles, single and multiple QTL mapping methods, easy addition of new QTL 
analyses, scalable data management and analysis tracking.

\begin{enumerate}\itemsep1pt
\item \italic{Explore QTL profiles} - Through the web-interface, users can explore mapped QTLs, and underlying information 
by viewing QTL plots in an interactive scrollable and zoomable window. \xqtlwb has support for other common image 
formats, such as PNG, JPG, SVG and embedded postscript; useful for publishing scientific results online, and on 
paper. From the output, main database identifiers, such as gene, protein and/or metabolite identifiers are 
automatically linked-out to matching external web pages of public database such as NCBI, KEGG, and Wormbase.

\item \italic{Single and multiple QTL mapping} - \xqtlwb wraps R/qtl\cite{Broman:2003, Arends:2010} in a web-based analysis
framework offering all important QTL mapping routines, such as the EM algorithm, imputation, Haley-Knott regression, 
the extended Haley-Knott method, marker regression, and Multiple QTL mapping. In addition, \xqtlwb includes 
R/qtlbim, a library which provides a Bayesian model selection approach for mapping multiple interacting QTL\cite{Yandell:2007} 
and Plink, a library for association QTL mapping on Single Nucleotide Polymorphisms (SNP) in natural 
populations\cite{Purcell:2007}.

\item \italic{Add new analysis tools} - \xqtlwb supports flexible adding of more QTL analysis software: any R-based, or 
command-line tool, can be plugged in. All analysis results are uploaded, stored and tracked, in the \xqtlwb database 
through an R-API. When new tools are added, they can build on the high-level multi-core computer, cluster and 
Cloud management functions, based on TORQUE/OpenPBS and BioNode\cite{Prins:2012}. \xqtlwb can be made part of a larger 
analysis pipeline using interfaces to R, Excel, REST and SOAP web services and Galaxy\cite{Goecks:2010}.

\item \italic{Track and trace} - When a new analysis protocol or R script is defined, this protocol can 
easily be applied to new data. Also, \xqtlwb keeps track of history. Re-use of analysis protocols can be done in an 
automated fashion. Previous analyses can be rerun without resetting parameters. \xqtlwb provides an online overview 
of past analyses e.g. which analyses were performed, by who, when and display settings applied.

\item \italic{Scalable data management} - \xqtlwb has a consistency checking database based on XGAP specification\cite{Swertz:2010a}, 
user interfaces to manage and query genotype and phenotype data sets, and support for various database back-ends including 
HSQL (standalone) and MySQL. Phenotype, genotype and genetic map data can be imported as text (TXT), comma separated (CSV), 
and Excel files. \xqtlwb handles and stores large data in a new and efficient binary edition of the XGAP format, named 
XGAPbin (extension .xbin), documented online. Such binary formats are essential when handling, storing and transporting 
multi-Gigabyte datasets.

\item \italic{Customize to research} - Additional modules for new data modalities can be added using MOLGENIS software 
generator\cite{Swertz:2010b}. The 'look and feel' of \xqtlwb is adaptable to institute or consortium style by changing a 
simple template which is described in the \xqtlwb documentation enabeling seamless integration into an existing web-site
or intranet site, such as recently for EU-PANACEA model organism project and LifeLines biobank.
\end{enumerate}

\subsection{xQTL workbench - Implementation}

We built \xqtlwb on top of MOLGENIS\cite{Swertz:2004}, a Java based software to generate tailored research infrastructure 
on-demand\cite{Swertz:2007}. From a single 'blueprint' describing the whole system, MOLGENIS auto-generates a full 
application including user interface, database infrastructure, application programming interfaces in R, REST and SOAP 
(APIs). MOLGENIS' flexibility and robustness is proven by the wide range of research projects, e.g. the Nordic GWAS 
Control database\cite{Leu:2010}, EB mutation database \cite{Akker:2011}, the Animal observation database\cite{Swertz:2010b}.

For data storage, the eXtensible Genotype and Phenotype (XGAP) data model was adopted\cite{Swertz:2010a} and extended 
for big data. To support the increased demand for computational resources for included mapping routines we added high-level 
cluster and cloud management functions for computation. The scalable QTL mapping routines of \xqtlwb are written in R 
and C. The choice of R ties in with the general practise of using R for QTL mapping. The user interface includes direct 
access to the R interpreter.  Both \xqtlwb and MOLGENIS are open source software, and source code is transparently
stored and tracked in online source control repositories.

\subsection{xQTL workbench - Discussion}

\xqtlwb provides a total-solution for web-based analysis: Major QTL mapping routines are integrated for use by experienced 
and inexperienced users. Researchers can upload raw data, run analyses, explore mapped QTL and underlying information, and 
link-out to important databases. New algorithms can be flexibly added, immediately available to all users. Large analyses 
can be easily executed on a cluster, or in the Cloud. Future work include visualizations and search options to explore the 
results. We also had an EU-SYSGENET workshop that envisioned further integration of xQTL with analysis tools like HAPPY, 
databases like GeneNetwork, and the workflow manager TIQS\cite{Durrant:2012}.

\section{A worm database (WormQTL)}
WormQTL is one of the application developed using the xQTL workbench system. It is a public web portal for the management 
of all these new data and integrated development of suitable analysis tools. The web server provides a rich set of analysis 
tools available to use directly, based on R/qtl \cite{Broman:2003, Arends:2010}. Users can upload and share new R scripts 
as 'plugin' for colleagues in the community to use directly. New data can be uploaded and downloaded using XGAP-extensible 
text format for genotype and phenotypes\cite{Swertz:2010a}. All data and tools can be accessed via web user interfaces and 
programming interfaces to R, REST, and SOAP web services. Large consortia as well as individual researchers, can have a 
private area that is under embargo for publication. All software is free for download as MOLGENIS 'app'\cite{Swertz:2010b}. 
WormQTL is freely accessible without registration and is hosted on a large computational cluster enabling high throughput 
analyses to all at http://www.wormqtl.org.

\subsection{WormQTL - Introduction}
Over the past 30 years, the metazoan Caenorhabditis elegans has become a premier animal model for determining the genetic 
basis of quantitative traits\cite{Gaertner:2010, Kammenga:2008}. The extensive knowledge of molecular, cellular and neural 
bases of complex phenotypes makes \italic{C. elegans} an ideal system for the next endeavour: determining the role of natural genetic 
variation on system variation. These efforts have resulted in an accumulation of a valuable amount of phenotypic, 
high-throughput molecular and genotypic data across different developmental worm stages and environments in hundreds of 
strains (3–19). In addition, a similar wealth has been produced on hundreds of different \italic{C. elegans} wild isolates and 
other species (20). For example, \italic{C. briggsae} is an emerging model organism that allows evolutionary comparisons with 
\italic{C. elegans} and quantitative genetic exploration of its own unique biological attributes (21).

This rapid increase in valuable data calls for an easily accessible database allowing for comparative analysis and meta-analysis 
within and across Caenorhabditis species (22). To facilitate this, we designed a public database repository for the worm community, 
WormQTL (http://www.wormqtl.org). Driven by the PANACEA project of the systems biology program of the EU, its design was tuned 
to the needs of \italic{C. elegans} researchers via an intensive series of interactive design and user evaluation sessions on 
a mission to integrate all available data within the project.

As a result, data that were scattered across different platforms and databases can now be stored, downloaded, analysed and 
visualized in an easily and comprehensive way in WormQTL. On top, the database provides a set of user interfaced analysis 
tools to search the database and explore genotype–phenotype mapping based on R/qtl\cite{Broman:2003, Arends:2010}. New data can be uploaded and 
downloaded using the extensible plain text format for genotype and phenotypes, XGAP\cite{Swertz:2010a}. There is no limit to the type of 
data (from gene expression to protein, metabolite or cellular data) that can be accommodated because of its extensible design. 
All data and tools can be accessed via a public web user interface and programming interfaces to R and REST web services, 
which were built using the MOLGENIS biosoftware toolkit\cite{Swertz:2010b}. Moreover, users can upload and share more R scripts as 'plugin' 
for the colleagues in the community to use directly and run those on a computer cluster using software modules from xQTL 
workbench\cite{Arends:2012}; this requires login to prevent abuse. All software can be downloaded for free to be used, for example as 
local mirror of the database, and/or to host new studies.

All the software was built as open source, reusing and building on existing open source components as much as possible. WormQTL 
is freely accessible without registration and is hosted on a large computational cluster enabling high-throughput analyses at 
http://www.wormqtl.org. Below we detail the results, methods used to implement the system and future plans.

\subsection{WormQTL - Results}
WormQTL is an online database platform for expression quantitative trait loci (eQTL) exploration to service the worm community and 
already provides many publicly available data sets (5,9-15,19). New data sets can be uploaded using the XGAP plain file data format. 
Suitable help pages are provided. Currently, 38 public data sets have been loaded, of which the bulk is xQTL data on 500 strains 
(introgression lines, recombinant inbred lines (RILs), recombinant inbred advanced intercross lines and natural isolates), 55,000 
transcripts, 1594 samples and 1579 markers (Table 1). With this combination of classical phenotypes, molecular profiles and genetics 
data sets, WormQTL contains all the 'genetical genomics' experiments published to our current knowledge (except for some tiling data). 
Using WormQTL, researchers can explore many xQTLs across the various studies in different conditions and ages and compare classical 
QTLs with xQTLs. The main interfaces are 'Find QTLs', 'Genome browser' and 'Browse data'.

\begin{enumerate}\itemsep1pt
\item \italic{Find QTLs} - QTL is genomic regions associated with phenotypic variation and can be used to study the genetic architecture of 
traits and to detect potential phenotypic regulators. Recently, the number of QTLs and especially eQTL studies in \italic{C. elegans} has 
increased greatly. These eQTL studies consist of large data sets that, before WormQTL, were very difficult to access and perform a 
combined meta-analysis. Therefore, we provide easy access to most of the eQTL studies published, by search, browse and plot 
functions (Figure 1). We support relatively simple questions like 'does my gene have an xQTL?' to more advanced ones like 'how do 
these genes fit into an xQTL network?'. All the matching genes, markers and traits found in the data sets are returned including 
links to WormBase and literature. Furthermore, WormQTL is the first portal for any species that allows comparison of eQTLs over 
multiple experiments and environments, giving insight in the plastic nature of genetic regulation.

\item \italic{Genome browser} - To find the genes that have a QTL on your favourite position, click 'Genome browser'. Here, you can select 
from all the different releases of the University of California, Santa Cruz genome releases. You can add tracks from the designated 
experiments of interest. Then navigate to your favourite location (tip: use open in new window) and collect significant probe 
identifiers from that region.

\item \italic{Browse data} - Complete data sets and accompanying gene, sample and trait identifier lists can be browsed via the 'browse data' 
user interface. External identifiers anywhere in the data are automatically recognized and enhanced as linkouts to background 
information, such as links to Wormbase, NCBI, KEGG or Ensembl. All the annotation lists and data matrices can be browsed and searched 
in a tabular form and can be downloaded as plain text or Excel files. Readers can also download data sets or submit new data sets 
using the XGAP data format following examples described in the WormQTL help section. Also all data can be accessed programmatically 
from with R (as whole matrix or per row) or using REST web services, including filtering of the annotations (genes, probes, markers 
and phenotypes) and services to 'slice' individual lines out of the complete data sets to speed up download and (parallel) analyses. 
Alternatively, readers can request a login to upload data and new analysis scripts directly.
\end{enumerate}

\subsection{WormQTL - Discussion}
\subsubsection{Implementation}
All the software was implemented using the open source Molecular Genetics Information Systems MOLGENIS toolkit (26), and in 
particular one previously existing MOLGENIS application, the extensible xQTL workbench (27) and the R/qtl QTL mapping and 
visualization package for the R language (23,24). The MOLGENIS toolkit is a Java-based software to generate tailored research 
infrastructure on demand (22). From a single 'blueprint' describing all biological data structures and user interfaces of the 
whole system, MOLGENIS autogenerates a full application including user interface, database infrastructure and application 
programming interfaces (APIs) in R, REST and SOAP.

At the push of a button, MOLGENIS 'generators' automatically translates these models into a database, standard user interfaces 
for data queries and updates, upload/download tools for tab-delimited data and scriptable interfaces for programmers to users 
from within R and via web services. This greatly speeded up the initial software development and also enables rapid extension 
when, for example, new data types arrive. On top of this foundation, we build the WormQTL specific user interactions such as 
the 'Find QTLs' and the 'Genome browser' using MOLGENIS 'plug-in' mechanism and the visualizations and plots using the R 
interface. xQTL workbench is a scalable web platform for the mapping of QTLs at multiple levels: for example, gene expression 
(xQTL), protein abundance (pQTL), metabolite abundance (mQTL) and phenotype (phQTL) data. The xQTL workbench provided a set of 
previously developed user interfaces to run R/qtl mapping methods directly from within the WormQTL user interface, the ability 
to add new analysis procedures in R, data management and data format conversions, all greatly speeding up the generation of 
new xQTL profiles.

All the data sets were downloaded from their original sources and then formatted using the XGAP data format. XGAP is a simple 
text file format that uses a directory of tab-delimited files or one Excel file with multiple sheets to load lists of annotations 
and data matrices. The annotations list all the background information needed to run and interpret the analysis including, 
for example, genome position information, such as markers, genes, probes and strains. The data matrices describe all the raw, 
intermediate and result data, such as gene expression, genotypes and QTL P-values, with the row names and column names cross 
linking to the annotations. For example, gene expression is a matrix of 'gene' X 'sample'. Subsequently these data sets were 
loaded using the MOLGENIS/xQTL data import wizards, which check the files for correctness and give informative feedback if the 
data are not yet in a format that WormQTL can understand (25). All the annotations are stored in tables in the database; the 
large data matrices are stored in a optimized binary format to speed up analyses and queries. This format is documented in 
the WormQTL manual to ease the submission of new data sets from the community. Finally, all the QTL profiles were recalculated 
according to the specification of the original, or slightly modified when needed, such as to include a previously missing wrongly 
labelled sample correction. In this process, we greatly benefitted from the integration with xQTL workbench, which enabled us to 
re-run all these analyses on the computer cluster and add new R analysis procedures when needed, simply from the user interface.

All software is available as open source on http://github.com/molgenis for others to reuse locally, and related technical 
documentation is available at http://www.xqtl.org and http://www.rqtl.org and http://www.molgenis.org.

\subsubsection{Future plans}
The current version of WormQTL (June 2012) is a comprehensive, versatile and flexible package. Follow-up plans of more extended 
versions with new tools and data depend on the demand by the users of WormQTL. We envisage that in the future, three types of 
new tools will be developed: (i) visualization tools, (ii) QTL mapping tools and (iii) candidate gene selection tools. Improved 
visualization tools might include plotting a phenotype against the marker at a certain position; so the two groups become visible 
at a QTL position. Also plots can be made showing transgression and heritability per microarray probe or gene or histograms of 
the phenotypic values (and include the parental values if available). Advanced QTL mapping tools might include multi-environment
/age mapping or genotype-by-environment analyses, developed in collaboration with the R/qtl team to enable automatic links to 
this software. The candidate gene selection tools would benefit from the most recent stable release of Wormbase (28), the most 
widely used platform for worm biology. But also other sources of information like MODENCODE (29) or Wormnet (30) are likely to 
be connected with WormQTL. A candidate gene selection tool might be implemented in a next version of WormQTL as it is less easy 
to implement and often needs information beyond WormQTL. One can think of (i) which SNPs/genes/polymorphic genes/transcription 
factor binding sites and so forth are underlying a eQTL; (ii) which gene, underlying my xQTL, is linked to most of the genes 
having an xQTL; (iii) which genes are polymorphic and (iv) which other genotypes show a difference in expression and do they 
share polymorphisms with the parental strains of the RIL population that the xQTL was mapped in. Moreover, WormQTL can be easily 
expanded to other Caenorhabditis species (21).

We believe that WormQTL, which will be continuously curated by the members of this international consortium, is a very 
attractive database for the growing community of quantitative genetics in worms researchers. We are committed to maintain data 
and software for the years to come and invite the community to add and share new data and ideas.

\chapter{Mapping correlation}

\emph{In this chapter we take one step further then in the previous chapters and develop a new methodology for
quantitative genetics called Correlated Traits Locus (CTL) mapping, a method complementairy to QTL mapping. 
Where QTL associates differences in mean, CTL, associate differences in correlation to genetic variation, i.e. 
CTL identify regions in the genome for which one genotype leads to correlated expression between a pair of 
traits, while the other genotype shows none (or significantly different) correlation.}

\null
\vfill

\begin{myexampleblock}{In press:}
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  \emph{CTL mapping}\\
  \bold{BMC Bioinformatics} (2013)
\end{myexampleblock}

\newpage

\section{What is a CTL?}
\lipsum[1]

\section{Combining CTL and QTL information}
\lipsum

\section{Examples of CTL mapping}
\lipsum
\subsection{CTL mapping in an \emph{A. thaliana} RIL population}

\subsection{CTL mapping using human GWA data}

\chapter{Thesis summary}
\lipsum[1-3]

\chapter{Additional for Dissertation}
\section*{Nederlandse Samenvatting / Dutch Summary}
\addcontentsline{toc}{section}{Nederlandse Samenvatting / Dutch Summary}
\lipsum[1]

\section*{Abbreviations and acronyms}
\addcontentsline{toc}{section}{Abbreviations and acronyms}
\begin{tabular}{ l l }
API:         & Application Programming Interface\\
BC:          & Backcross \\
bp:          & Base pair(s) \\
cM:          & centi Morgan \\
CPNN:        & Collaborative computing project for NMR\\
CSV:         & Comma separated values\\
CTL:         & Correlated traits locus \\
DesignGG:    & Experimental design of genetical genomics software\\
DRY:         & Principle of don’t repeat yourself\\
DSL:         & Domain Specific Language\\
EBI:         & European Bioinformatics Institute\\
FINDIS:      & Finish disease database\\
GEN2PHEN:    & EU project to unify human and model organism genetic variation databases\\
GMOD:        & Generic model organism database project\\
GUI:         & Graphical user interface\\
GWAS:        & Genome Wide Association Study\\
GWL:         & Genome Wide Linkage analysis\\
HGVBaseG2P:  & Human genome variation database of genotype-to-phenotype information\\
HTML:        & Hypertext markup language\\
IDE:         & Integrated Development Environment\\
JAR:         & Java Software Archive\\
LGPL:        & Lesser general public license\\
MAGE-TAB:    & Microarray gene expression tab delimited file format\\
Mbp:         & Mega base pairs = 1.000.000 bp \\
MOLGENIS:    & Molecular genetics information systems toolkit\\
NordicDB:    & Nordic Control Cohort Database\\
OBF:         & Open Bioinformatics Foundation\\
OntoCAT:     & Ontology common API toolkit\\
PEAA:        & Patterns for enterprise application architecture\\
QTL:         & Quantitative trait locus\\
RDF:         & Resource description format\\
REST:        & Representative state transfer web services\\
RIL:         & Recombinant inbred line \\
SNP:         & Single Nucleotide Polymorphism\\
SOAP:        & Simple Object Access Protocol\\
SQL:         & Structured Query Language\\
UML:         & Uniform data Modeling Language\\
WAR:         & Web Application aRchive file\\
XML:         & Extensible Markup Language\\
XGAP:        & Extensible genotype and phenotype software platform. 
\end{tabular}

\newpage

\section*{Acknowledgements}
\addcontentsline{toc}{section}{Acknowledgements}
\lipsum[1]

\newpage

\section*{List of publications}
\addcontentsline{toc}{section}{List of publications}
\subsection*{Authored:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  R/qtl: high throughput Multiple QTL mapping\\
  \bold{Bioinformatics} (2010)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\\\
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  xQTL workbench: a scalable web environment for multi-level QTL analysis\\
  \bold{Bioinformatics} (2012)\\\\
  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, Mark Elvin, et al.}\\
  WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp\\
  \bold{Nucleic Acids Research} (2012)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)

\subsection*{Co-Authored:}
  \authors{Morris A Swertz, Martijn Dijkstra, Tomasz Adamusiak, Danny Arends, et al.}\\
  The MOLGENIS toolkit: rapid prototyping of biosoftware at the push of a button\\
  \bold{BMC Bioinformatics} (2010)\\\\
  \authors{Morris A Swertz, K Joeri van der Velde, Bruno M Tesson, Danny Arends, et al.}\\
  XGAP: a uniform and extensible data model and software platform for genotype and phenotype experiments\\
  \bold{Genome Biology} (2010)\\\\
  \authors{Klaus Schughart, Danny Arends, P. Andreux, R. Balling, Pjotr Prins, et al.}\\
  SYSGENET: a meeting report from a new European network for systems genetics\\
  \bold{Mammalian Genome} (2010)\\\\
  \authors{Rudolf SN Fehrmann, Ritsert C. Jansen, Jan H. Veldink, Harm-Jan Westra, Danny Arends, et al.}\\
  Trans-eQTLs Reveal that Independent Genetic Variants Associated With a Complex Phenotype Converge on Intermediate Genes, with a Major Role for the HLA\\
  \bold{Plos Genetics} (2011)\\\\
  \authors{Caroline Durrant, Morris A. Swertz, Rudi Alberts, Danny Arends, Klaus Schughart, et al.}\\
  Bioinformatics tools and database resources for systems genetics analysis in mice - a short review and an evaluation of future needs\\
  \bold{Briefings in Bioinformatics} (2011)

\subsection*{In Press:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)\\\\
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  CTL mapping\\
  \bold{BMC Bioinformatics} (2013)

\subsection*{Acknowledged in:}
  \authors{Yang Li, Morris A Swertz, Gonzalo Vera, Jingyuan Fu, Rainer Breitling and Ritsert C Jansen}\\
  DesignGG: an R-package and web tool for the optimal design of genetical genomics experiments\\
  \bold{BMC Bioinformatics}, 10:188 (2009)\\\\
  \authors{Yang Li, Rainer Breitling and Ritsert C. Jansen}\\
  Generalizing genetical genomics: getting added value from environmental perturbation\\
  \bold{Trends in Genetics}, 24:518-524 (2008)

\bibliographystyle{plain}
\addcontentsline{toc}{chapter}{Bibliography}
\bibliography{Thesis}

\end{document}

