\documentclass[8pt, twoside, a5paper]{report}
\usepackage[usenames, dvipsnames, svgnames]{xcolor}
\usepackage{titlesec, blindtext, color, graphicx}
\usepackage[light,condensed,math]{kurier}
\usepackage[T1]{fontenc}
\usepackage{tcolorbox}
\usepackage{lipsum}
\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}
\definecolor{GREEN}{rgb}{0, 0.5, 0.1}

%\usepackage{lmodern} % to remove warnings of size substitution
\usepackage[top=0.6in, bottom=0.6in, left=0.6in, right=0.4in]{geometry}

\tcbuselibrary{skins,breakable}
\usetikzlibrary{shadings,shadows}

\newenvironment{myexampleblock}[1]{%
    \tcolorbox[beamer,%
    noparskip,breakable,
    colback=LightGreen,colframe=DarkGreen,%
    colbacklower=LimeGreen!75!LightGreen,%
    title=#1]}%
    {\endtcolorbox}

\newcommand{\hsp}{\hspace{7pt}}
\newcommand{\hssp}{\hspace{3pt}}

\titleformat{\chapter}[hang]{\vspace{-2 cm}\huge\bfseries}{{\fontsize{50}{60}\selectfont \thechapter \hsp\textcolor{GREEN}{|}\hsp}}{0pt}{\huge\bfseries}
\titleformat{\section}[hang]{\large}{\thesection\hssp}{0pt}{\large\bfseries}

\newcommand{\authors}[1]{\small{#1}}
\newcommand{\italic}[1]{\textit{#1}}
\newcommand{\bold}[1]{{\bfseries #1}}
\newcommand{\code}[1]{{\tt #1}}
\newcommand{\mytitle}[1]{{\LARGE #1}}

\newcommand{\xqtlwb}{{\it x}QTL workbench}
\newcommand{\molgenis}{Molgenis} % don't use all caps, see journal guidelines
\newcommand{\xgap}{XGAP} % don't use all caps, see journal guidelines

\pagestyle{headings}
\renewcommand{\contentsname}{Table of content\vspace{-30pt}}

\begin{document}
\pagenumbering{arabic}
  \thispagestyle{empty}
  \begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
  \end{center}
\newpage
  \thispagestyle{empty}
  \noindent The work described in this thesis was carried out at the Groningen Bioinformatics Centre, University of Groningen, The Netherlands.
  This research was financially supported by\\
  \vspace{130 mm}
  
  \noindent (c) 2013 by GBIC - Danny Arends\\
  Printed by:\\
  ISBN:\\
\newpage

\thispagestyle{empty}
\begin{center}
  \mytitle{Methods and Software for Systems Genetics}\\
    \vspace{10 mm}
  \bold{Proefschrift}\\
    \vspace{10 mm}
  Ter verkrijging van het doctoraat in de \\
  Wiskunde en Natuurwetenschappen \\
  aan de Rijksuniversiteit Groningen \\
  op gezag van de \\
  Rector Magnificus, dr. E. Sterken\\ 
  in het openbaar te verdedigen op \\
  Maandag 1 September 2013 \\
  om 12.00 uur \\
    \vspace{10 mm}
  door\\
    \vspace{10 mm}
  Derk Arends\\
    \vspace{10 mm}
  Geboren op 15 juli 1983\\
  te Zwolle
\end{center}

\newpage
\thispagestyle{empty}
\begin{tabular}{ l l }
Promotor:             & Prof. dr. R.C. Jansen \\
                      & \\
Co-Promotors:         & dr. F. Johannes \\
                      & dr. M. Swertz \\
                      & \\
Beoordelingcommisie:  & Prof. dr. \\
                      & Prof. dr. \\
                      & Prof. dr. \\
\end{tabular}
\tableofcontents

\chapter{Introduction}
\lipsum[1-3]

\chapter{Phenotypes and Genotypes}

\emph{Using prior knowledge about phenotypes and how they arise / mix in sexual reproduction allows us to saturate 
existing genetic maps or create them de-novo when enough data is available. Using a sensitive pre-selection 
and mixture modeling approach the resolution of the genetic map of A. thaliana was improved.}

\null
\vfill

\begin{myexampleblock}{Under review:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)
\end{myexampleblock}
\newpage

\section{Pheno2Geno}
Pheno2geno is a package for the computational generation of markers and construction of genetic maps from 
molecular phenotypes in segregating inbred line populations, what is possible because phenotypes with a 
clearly separated bimodal or trimodal expression distribution can be used as genetic markers. Pheno2geno 
offers: de novo map construction, saturation of existing maps and detection of sample mix-ups. 

Pheno2geno starts by finding the phenotypes that are suitable as markers. These phenotypes should show 
differential expression between founders and are selected using a Student's t-test or RankProd. In the 
next step mixture modeling is used to select phenotypes that show bimodal (e.g. for RIL, BC) or trimodal 
(e.g. for F2) expression patterns with mixing proportions comparable to the expected segregation 
frequencies (e.g. 1 to 1 in RIL) across all the offspring individuals. Only these phenotypes are 
transformed from continuous measurements into discrete markers.

After marker detection the markers are used to create de novo map. Additional information, like known physical 
and/or genetic positions for all/some of the markers could be used and it will improve the quality of the 
resulting map. When a genetic map is available pheno2geno can be used to saturate it. For each of the markers 
it is assessed whether it has a single significant QTL. If so, it is being placed on the map in the position 
of this peak, if not it is dropped from the analysis.

Pheno2geno can additionally be used to point out sample mix-ups and errors. Using QTL information the package 
compares the observed phenotype value with the expected value. This results in calculating a mismatch score, 
and obtaining phenotype-based idealized genotypes. 

\section{Phenotypes}
\lipsum

\section{Markers and Maps}
\lipsum

\chapter{(Multiple) QTL mapping}

\emph{QTL mapping is the main analysis method used in the analysis of quantitative traits. This approach has been 
the analysis method to study quantitative traits since its discovery in 1980. Many variations to the basic single 
marker approach can be made, but none so fundamental as multiple QTL mapping using generalized lineair models. 
Computational and sample size problems were limiting the application of the approach. Novel experimental 
design and decreased costs have made sample size less of an issue, and by 'refreshing' the orginal algorithm 
more wide-spread application of this important algorithm is possible.}
\null
\vfill

\begin{myexampleblock}{Originally published as:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  \emph{R/qtl: High throughput Multiple QTL mapping}\\
  \bold{Bioinformatics} (2010) \\

  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\

  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, 
  Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using 
  Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)
\end{myexampleblock}

\newpage

\section{Single marker QTL mapping (R/qtl)}

R/qtl is an extensible, interactive environment for the mapping of quantitative trait loci (QTL) in experimental 
crosses. It is implemented as an add-on package for the freely available and widely used statistical language/
software R \cite{R:2009}. Since its introduction, R/qtl \cite{Broman:2003} has become a
reference implementation with an extensive guide on QTL mapping \cite{RQTLGuide:2009}. R/qtl development is 
continuous, with input from multiple collaborators and users.  We have introduced a full testing environment with 
regression testing, updated the license to the GPL version 3, and hosted the source code repository on Github,
which gives R/qtl software development high visibility and transparency. 

The development of R/qtl reflects trends in quantitative genetics, in particular the use of larger datasets, larger 
calculations and requirements for controlling the false discovery rate. These developments are partly driven by 
high-throughput genetical genomics---the name coined for the study of gene expression QTL (eQTL)\cite{Jansen:2001}, 
metabolite QTL (mQTL), protein QTL (pQTL).

\section{Multiple QTL mapping}

Multiple QTL Mapping (MQM) belongs to a family of QTL mapping methods, that include Haley-Knott regression
\cite{Haley:1992} and composite interval mapping CIM \cite{Zeng:1994}. MQM combines the strengths of generalized 
linear model regression ith those of interval mapping \cite{Jansen:1993, Jansen:1994b}. 

Recent developments in QTL mapping include Bayesian modelling of multiple QTL e.g. R/qtlbim package
\cite{Yandell:2007, Banerjee:2008}. Bayesian modelling, however, is computationally expensive, and arguably has 
little additional power when applied to high density maps, and (nearly) complete genotype data\cite{Handbook:Jansen:2007}. 
Still, we intend to combine the strengths of the different methods in future versions of R/qtl.

These days, with most experimenal setups and high density maps, improving precision may be achieved by increasing 
the population size first. For more information on QTL mapping and Bayesian analysis we refer to the `Handbook of 
Statisical Genetics` \cite{Handbook:2007}. MQM makes used of generalized linear models, thereby potentially 
providing unified analysis of non-normal traits.

MQM provides a practical, relevant and sensitive approach for mapping QTL in experimental populations. The 
theoretical framework of MQM was introduced and explored by R.C. Jansen\cite{Jansen:1994a} and explained in the 
`Handbook of Statistical Genetics' \cite{Handbook:Jansen:2007}. MQM has one known commercial implementation
\cite{Mapqtl:2002}, which has been used effectively in practical research, resulting in hundreds of papers, e.g., 
in mouse, plant, and fish, respectively\cite{DeMooij:2009, Jeuken:2009, Kitano:2009}.  Now, with MQM for R/qtl, 
we present the first free and open source implementation of MQM, that is multi-platform, scalable and suitable 
for automated procedures and large datasets.

\subsection{Features}
MQM for R/qtl is an automated three-stage procedure in which, in the first stage, missing genotype data is 'augmented'. 
In other words, rather than guessing one likely genotype, multiple genotypes are modelled with their estimated 
probabilities. In the second stage, important marker cofactors are selected by multiple regression and backward 
elimination. The third stage, a QTL is moved along the chromosomes using these pre-selected markers as cofactors. 
QTL are interval-mapped using the most informative model selected by either maximum likelihood or restricted maximum 
likelihood. A refined and automated procedure for cases with large numbers of marker cofactors is included. 

The method lets users test different QTL models by elimination of non-significant cofactors. MQM for R/qtl brings the 
following advantages to QTL mapping:
\begin{enumerate}\itemsep1pt
\item Higher power, as long as the QTL explain a reasonable amount of variation.
\item Protection against over-fitting, because MQM fixes the residual variance from the full model, which allows the 
use of more cofactors than may be used in, for example, composite interval mapping\cite{Zeng:1994}.
\item Prevention of ghost QTL detection (between two QTL in coupling phase).
\item Detection of negating QTL (QTL in repulsion phase). 
\item MQM gives (compared to CIM) a reduction in type I and type II error \cite{Handbook:Jansen:2007}.
\item A pragmatic permutation strategy for controlling the false discovery rate (FDR) and prevention of locating 
false QTL hot spots, as discussed in Breitling et al.\cite{Breitling:2008a}. Marker data is permuted, while keeping 
the correlation structure in the trait data.
\item High-performance computing by scaling on multi-CPU computers, as well as clustered computers, by calculating 
phenotypes in parallel, through the Message Passing Interface (MPI) of the SNOW package for R\cite{Tierney:2009}.
\item Visualizations for exploring interactions in a genomic circle plot (Fig. XXX) and cis- and trans-regulation (Fig. XXX).
\end{enumerate}

A 40-page tutorial for MQM explores, both the automated procedure, and the manual procedure of adding and removing 
cofactors, in an \emph{Arabidopsis thaliana} recombinant inbred line (RIL) metabolite (mQTL) dataset with 24 metabolites 
as phenotypes\cite{Fu:2007}. In addition, the tutorial visually explains the effects of data augmentation, cofactor 
selection, model selection, and tweaking of input parameters, such as cofactor significance. Genetic interactions 
(epistasis) are explored through effect plots, and an example is given of parallel computation. The tutorial is part 
of the software distribution of R/qtl and is available online.

\subsection{Conclusions}
MQM for R/qtl is a significant addition to the QTL mapper's toolbox. R/qtl provides the user with the most frequently 
used statistical analysis methods: single-marker analysis, interval mapping, Haley-Knott regression \cite{Haley:1992}, 
CIM \cite{Zeng:1994} and MQM \cite{Jansen:1994a}.  MQM has improved handling of missing data and allows more powerful 
and precise detection of QTL, compared to many other methods. Not only is this new implementation of MQM available in the
statistical R environment, which allows scripting for pipe-lined setups, it is also highly scalable through 
parallelisation and paves the way for high-throughput QTL analysis. With MQM, R/qtl is a free and high-performance 
comprehensive QTL mapping toolbox for the analysis of experimental populations. R/qtl now includes permutation strategies 
for determining thresholds of significance relevant for QTL and QTL hot spots; the first step towards causal inference and
network analysis.

\section{Classical phenotypes in a DesignGG experiment}
\lipsum

\section{Metabolites in a DesignGG experiment}
\lipsum

\chapter{High throughput data analysis}

\emph{Problems in bioinformatics are the huge amount of data gathered and the multitude of technologies used. We 
developed the xQTL workbench system\cite{Arends:2012} to store large amounts of phenotype and genotype data 
in the XGAP\cite{Swertz:2010a} dataformat. xQTL workbench can easily be adapted to a multitude of input and 
storage formats using the MOLGENIS\cite{Swertz:2004} generator. xQTL workbench uses the power of distibuted 
computing to allow massive parrallel QTL analysis and allows new analysis tools to be quickly added to the 
system, an application of the xQTL system is found in the WormQTL database\cite{Snoek:2012}.}

\null
\vfill

\begin{myexampleblock}{Originally published as:}
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  \emph{xQTL workbench: a scalable web environment for multi-level QTL analysis}\\
  \bold{Bioinformatics} (2012)\\\\

  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, et al.}\\
  \emph{WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp}\\
  \bold{Nucleic Acids Research} (2012)
\end{myexampleblock}

\newpage

\section{Storage (MOLGENIS, XGAP)}
\lipsum

\section{Analysis (xQTL workbench)}
xQTL workbench is a scalable web platform for the mapping of quantitative trait loci (QTLs) at multiple levels: 
for example gene expression (eQTL), protein abundance (pQTL), metabolite abundance (mQTL) and phenotype (phQTL) data. 
Popular QTL mapping methods for model organism and human populations are accessible via the web user interface. 
Large calculations scale easily on to multi-core computers, clusters and Cloud. All data involved can be uploaded 
and queried online: markers, genotypes, microarrays, NGS, LC-MS, GC-MS, NMR, etc. When new data types come available, 
xQTL workbench is quickly customized using the MOLGENIS software generator.

Modern high throughput technologies generate large amounts of genomic, transcriptomic, proteomic, and metabolomic 
data. However, existing open source web-based tools for QTL analysis, such as webQTL\cite{Wang:2003} and 
QTLNetwork \cite{Yang:2008}, are not easily extendable to different settings and computationally scalable 
for whole genome analyses. \xqtlwb makes it easy to analyse large and complex datasets using state-of-the-art QTL 
mapping tools and to apply these methods to millions of phenotypes using paralellized 'Big Data' 
solutions\cite{Trelles:2011}. \xqtlwb also supports storing of raw, intermediate and final result data, and 
analysis protocols and history for reproducibility and data provenance. Use of MOLGENIS\cite{Swertz:2010b} 
helps to customize the software. All is conveniently accessible via standard Internet browsers on Windows, 
Linux or Mac (and using Java, R for the server).

\subsection{Features}
\xqtlwb provides visualization of QTL profiles, single and multiple QTL mapping methods, easy addition of new QTL 
analyses, scalable data management and analysis tracking.

\begin{enumerate}\itemsep1pt
\item \italic{Explore QTL profiles} - Through the web-interface, users can explore mapped QTLs, and underlying information 
by viewing QTL plots in an interactive scrollable and zoomable window. \xqtlwb has support for other common image 
formats, such as PNG, JPG, SVG and embedded postscript; useful for publishing scientific results online, and on 
paper. From the output, main database identifiers, such as gene, protein and/or metabolite identifiers are 
automatically linked-out to matching external web pages of public database such as NCBI, KEGG, and Wormbase.

\item \italic{Single and multiple QTL mapping} - \xqtlwb wraps R/qtl\cite{Broman:2003, Arends:2010} in a web-based analysis
framework offering all important QTL mapping routines, such as the EM algorithm, imputation, Haley-Knott regression, 
the extended Haley-Knott method, marker regression, and Multiple QTL mapping. In addition, \xqtlwb includes 
R/qtlbim, a library which provides a Bayesian model selection approach for mapping multiple interacting QTL\cite{Yandell:2007} 
and Plink, a library for association QTL mapping on Single Nucleotide Polymorphisms (SNP) in natural 
populations\cite{Purcell:2007}.

\item \italic{Add new analysis tools} - \xqtlwb supports flexible adding of more QTL analysis software: any R-based, or 
command-line tool, can be plugged in. All analysis results are uploaded, stored and tracked, in the \xqtlwb database 
through an R-API. When new tools are added, they can build on the high-level multi-core computer, cluster and 
Cloud management functions, based on TORQUE/OpenPBS and BioNode\cite{Prins:2012}. \xqtlwb can be made part of a larger 
analysis pipeline using interfaces to R, Excel, REST and SOAP web services and Galaxy\cite{Goecks:2010}.

\item \italic{Track and trace} - When a new analysis protocol or R script is defined, this protocol can 
easily be applied to new data. Also, \xqtlwb keeps track of history. Re-use of analysis protocols can be done in an 
automated fashion. Previous analyses can be rerun without resetting parameters. \xqtlwb provides an online overview 
of past analyses e.g. which analyses were performed, by who, when and display settings applied.

\item \italic{Scalable data management} - \xqtlwb has a consistency checking database based on XGAP specification\cite{Swertz:2010a}, 
user interfaces to manage and query genotype and phenotype data sets, and support for various database back-ends including 
HSQL (standalone) and MySQL. Phenotype, genotype and genetic map data can be imported as text (TXT), comma separated (CSV), 
and Excel files. \xqtlwb handles and stores large data in a new and efficient binary edition of the XGAP format, named 
XGAPbin (extension .xbin), documented online. Such binary formats are essential when handling, storing and transporting 
multi-Gigabyte datasets.

\item \italic{Customize to research} - Additional modules for new data modalities can be added using MOLGENIS software 
generator\cite{Swertz:2010b}. The 'look and feel' of \xqtlwb is adaptable to institute or consortium style by changing a 
simple template which is described in the \xqtlwb documentation enabeling seamless integration into an existing web-site
or intranet site, such as recently for EU-PANACEA model organism project and LifeLines biobank.
\end{enumerate}

\subsection{Implementation}

We built \xqtlwb on top of MOLGENIS\cite{Swertz:2004}, a Java based software to generate tailored research infrastructure 
on-demand\cite{Swertz:2007}. From a single 'blueprint' describing the whole system, MOLGENIS auto-generates a full 
application including user interface, database infrastructure, application programming interfaces in R, REST and SOAP 
(APIs). MOLGENIS' flexibility and robustness is proven by the wide range of research projects, e.g. the Nordic GWAS 
Control database\cite{Leu:2010}, EB mutation database \cite{Akker:2011}, the Animal observation database\cite{Swertz:2010b}.

For data storage, the eXtensible Genotype and Phenotype (XGAP) data model was adopted\cite{Swertz:2010a} and extended 
for big data. To support the increased demand for computational resources for included mapping routines we added high-level 
cluster and cloud management functions for computation. The scalable QTL mapping routines of \xqtlwb are written in R 
and C. The choice of R ties in with the general practise of using R for QTL mapping. The user interface includes direct 
access to the R interpreter.  Both \xqtlwb and MOLGENIS are open source software, and source code is transparently
stored and tracked in online source control repositories.

\subsection{Conclusion}

\xqtlwb provides a total-solution for web-based analysis: Major QTL mapping routines are integrated for use by experienced 
and inexperienced users. Researchers can upload raw data, run analyses, explore mapped QTL and underlying information, and 
link-out to important databases. New algorithms can be flexibly added, immediately available to all users. Large analyses 
can be easily executed on a cluster, or in the Cloud. Future work include visualizations and search options to explore the 
results. We also had an EU-SYSGENET workshop that envisioned further integration of xQTL with analysis tools like HAPPY, 
databases like GeneNetwork, and the workflow manager TIQS\cite{Durrant:2012}.

\section{WormQTL}
WormQTL is one of the application developed using the xQTL workbench system. It is a public web portal for the management 
of all these new data and integrated development of suitable analysis tools. The web server provides a rich set of analysis 
tools available to use directly, based on R/qtl \cite{Broman:2003, Arends:2010}. Users can upload and share new R scripts 
as 'plugin' for colleagues in the community to use directly. New data can be uploaded and downloaded using XGAP-extensible 
text format for genotype and phenotypes\cite{Swertz:2010a}. All data and tools can be accessed via web user interfaces and 
programming interfaces to R, REST, and SOAP web services. Large consortia as well as individual researchers, can have a 
private area that is under embargo for publication. All software is free for download as MOLGENIS 'app'\cite{Swertz:2010b}. 
WormQTL is freely accessible without registration and is hosted on a large computational cluster enabling high throughput 
analyses to all at http://www.wormqtl.org.

\subsection{WormQTL - introduction}
Over the past 30 years, the metazoan Caenorhabditis elegans has become a premier animal model for determining the genetic 
basis of quantitative traits\cite{Gaertner:2010, Kammenga:2008}. The extensive knowledge of molecular, cellular and neural 
bases of complex phenotypes makes \italic{C. elegans} an ideal system for the next endeavour: determining the role of natural genetic 
variation on system variation. These efforts have resulted in an accumulation of a valuable amount of phenotypic, 
high-throughput molecular and genotypic data across different developmental worm stages and environments in hundreds of 
strains (3–19). In addition, a similar wealth has been produced on hundreds of different \italic{C. elegans} wild isolates and 
other species (20). For example, \italic{C. briggsae} is an emerging model organism that allows evolutionary comparisons with 
\italic{C. elegans} and quantitative genetic exploration of its own unique biological attributes (21).

This rapid increase in valuable data calls for an easily accessible database allowing for comparative analysis and meta-analysis 
within and across Caenorhabditis species (22). To facilitate this, we designed a public database repository for the worm community, 
WormQTL (http://www.wormqtl.org). Driven by the PANACEA project of the systems biology program of the EU, its design was tuned 
to the needs of \italic{C. elegans} researchers via an intensive series of interactive design and user evaluation sessions on 
a mission to integrate all available data within the project.

As a result, data that were scattered across different platforms and databases can now be stored, downloaded, analysed and 
visualized in an easily and comprehensive way in WormQTL. On top, the database provides a set of user interfaced analysis 
tools to search the database and explore genotype–phenotype mapping based on R/qtl\cite{Broman:2003, Arends:2010}. New data can be uploaded and 
downloaded using the extensible plain text format for genotype and phenotypes, XGAP\cite{Swertz:2010a}. There is no limit to the type of 
data (from gene expression to protein, metabolite or cellular data) that can be accommodated because of its extensible design. 
All data and tools can be accessed via a public web user interface and programming interfaces to R and REST web services, 
which were built using the MOLGENIS biosoftware toolkit\cite{Swertz:2010b}. Moreover, users can upload and share more R scripts as ‘plugin’ 
for the colleagues in the community to use directly and run those on a computer cluster using software modules from xQTL 
workbench\cite{Arends:2012}; this requires login to prevent abuse. All software can be downloaded for free to be used, for example as 
local mirror of the database, and/or to host new studies.

All the software was built as open source, reusing and building on existing open source components as much as possible. WormQTL 
is freely accessible without registration and is hosted on a large computational cluster enabling high-throughput analyses at 
http://www.wormqtl.org. Below we detail the results, methods used to implement the system and future plans.

\subsection{WormQTL - Results}
WormQTL is an online database platform for expression quantitative trait loci (eQTL) exploration to service the worm community and 
already provides many publicly available data sets (5,9-15,19). New data sets can be uploaded using the XGAP plain file data format. 
Suitable help pages are provided. Currently, 38 public data sets have been loaded, of which the bulk is xQTL data on 500 strains 
(introgression lines, recombinant inbred lines (RILs), recombinant inbred advanced intercross lines and natural isolates), 55,000 
transcripts, 1594 samples and 1579 markers (Table 1). With this combination of classical phenotypes, molecular profiles and genetics 
data sets, WormQTL contains all the 'genetical genomics' experiments published to our current knowledge (except for some tiling data). 
Using WormQTL, researchers can explore many xQTLs across the various studies in different conditions and ages and compare classical 
QTLs with xQTLs. The main interfaces are 'Find QTLs', 'Genome browser' and 'Browse data'.

\begin{enumerate}\itemsep1pt
\item \italic{Find QTLs} - QTL is genomic regions associated with phenotypic variation and can be used to study the genetic architecture of 
traits and to detect potential phenotypic regulators. Recently, the number of QTLs and especially eQTL studies in \italic{C. elegans} has 
increased greatly. These eQTL studies consist of large data sets that, before WormQTL, were very difficult to access and perform a 
combined meta-analysis. Therefore, we provide easy access to most of the eQTL studies published, by search, browse and plot 
functions (Figure 1). We support relatively simple questions like 'does my gene have an xQTL?' to more advanced ones like 'how do 
these genes fit into an xQTL network?'. All the matching genes, markers and traits found in the data sets are returned including 
links to WormBase and literature. Furthermore, WormQTL is the first portal for any species that allows comparison of eQTLs over 
multiple experiments and environments, giving insight in the plastic nature of genetic regulation.

\item \italic{Genome browser} - To find the genes that have a QTL on your favourite position, click 'Genome browser'. Here, you can select 
from all the different releases of the University of California, Santa Cruz genome releases. You can add tracks from the designated 
experiments of interest. Then navigate to your favourite location (tip: use open in new window) and collect significant probe 
identifiers from that region.

\item \italic{Browse data} - Complete data sets and accompanying gene, sample and trait identifier lists can be browsed via the 'browse data' 
user interface. External identifiers anywhere in the data are automatically recognized and enhanced as linkouts to background 
information, such as links to Wormbase, NCBI, KEGG or Ensembl. All the annotation lists and data matrices can be browsed and searched 
in a tabular form and can be downloaded as plain text or Excel files. Readers can also download data sets or submit new data sets 
using the XGAP data format following examples described in the WormQTL help section. Also all data can be accessed programmatically 
from with R (as whole matrix or per row) or using REST web services, including filtering of the annotations (genes, probes, markers 
and phenotypes) and services to 'slice' individual lines out of the complete data sets to speed up download and (parallel) analyses. 
Alternatively, readers can request a login to upload data and new analysis scripts directly.
\end{enumerate}

\subsection{WormQTL - Discussion}
\subsubsection{Implementation}
All the software was implemented using the open source Molecular Genetics Information Systems MOLGENIS toolkit (26), and in 
particular one previously existing MOLGENIS application, the extensible xQTL workbench (27) and the R/qtl QTL mapping and 
visualization package for the R language (23,24). The MOLGENIS toolkit is a Java-based software to generate tailored research 
infrastructure on demand (22). From a single 'blueprint' describing all biological data structures and user interfaces of the 
whole system, MOLGENIS autogenerates a full application including user interface, database infrastructure and application 
programming interfaces (APIs) in R, REST and SOAP.

At the push of a button, MOLGENIS 'generators' automatically translates these models into a database, standard user interfaces 
for data queries and updates, upload/download tools for tab-delimited data and scriptable interfaces for programmers to users 
from within R and via web services. This greatly speeded up the initial software development and also enables rapid extension 
when, for example, new data types arrive. On top of this foundation, we build the WormQTL specific user interactions such as 
the ‘Find QTLs’ and the ‘Genome browser’ using MOLGENIS 'plug-in' mechanism and the visualizations and plots using the R 
interface. xQTL workbench is a scalable web platform for the mapping of QTLs at multiple levels: for example, gene expression 
(xQTL), protein abundance (pQTL), metabolite abundance (mQTL) and phenotype (phQTL) data. The xQTL workbench provided a set of 
previously developed user interfaces to run R/qtl mapping methods directly from within the WormQTL user interface, the ability 
to add new analysis procedures in R, data management and data format conversions, all greatly speeding up the generation of 
new xQTL profiles.

All the data sets were downloaded from their original sources and then formatted using the XGAP data format. XGAP is a simple 
text file format that uses a directory of tab-delimited files or one Excel file with multiple sheets to load lists of annotations 
and data matrices. The annotations list all the background information needed to run and interpret the analysis including, 
for example, genome position information, such as markers, genes, probes and strains. The data matrices describe all the raw, 
intermediate and result data, such as gene expression, genotypes and QTL P-values, with the row names and column names cross 
linking to the annotations. For example, gene expression is a matrix of 'gene' X 'sample'. Subsequently these data sets were 
loaded using the MOLGENIS/xQTL data import wizards, which check the files for correctness and give informative feedback if the 
data are not yet in a format that WormQTL can understand (25). All the annotations are stored in tables in the database; the 
large data matrices are stored in a optimized binary format to speed up analyses and queries. This format is documented in 
the WormQTL manual to ease the submission of new data sets from the community. Finally, all the QTL profiles were recalculated 
according to the specification of the original, or slightly modified when needed, such as to include a previously missing wrongly 
labelled sample correction. In this process, we greatly benefitted from the integration with xQTL workbench, which enabled us to 
re-run all these analyses on the computer cluster and add new R analysis procedures when needed, simply from the user interface.

All software is available as open source on http://github.com/molgenis for others to reuse locally, and related technical 
documentation is available at http://www.xqtl.org and http://www.rqtl.org and http://www.molgenis.org.

\subsubsection{Future plans}
The current version of WormQTL (June 2012) is a comprehensive, versatile and flexible package. Follow-up plans of more extended 
versions with new tools and data depend on the demand by the users of WormQTL. We envisage that in the future, three types of 
new tools will be developed: (i) visualization tools, (ii) QTL mapping tools and (iii) candidate gene selection tools. Improved 
visualization tools might include plotting a phenotype against the marker at a certain position; so the two groups become visible 
at a QTL position. Also plots can be made showing transgression and heritability per microarray probe or gene or histograms of 
the phenotypic values (and include the parental values if available). Advanced QTL mapping tools might include multi-environment
/age mapping or genotype-by-environment analyses, developed in collaboration with the R/qtl team to enable automatic links to 
this software. The candidate gene selection tools would benefit from the most recent stable release of Wormbase (28), the most 
widely used platform for worm biology. But also other sources of information like MODENCODE (29) or Wormnet (30) are likely to 
be connected with WormQTL. A candidate gene selection tool might be implemented in a next version of WormQTL as it is less easy 
to implement and often needs information beyond WormQTL. One can think of (i) which SNPs/genes/polymorphic genes/transcription 
factor binding sites and so forth are underlying a eQTL; (ii) which gene, underlying my xQTL, is linked to most of the genes 
having an xQTL; (iii) which genes are polymorphic and (iv) which other genotypes show a difference in expression and do they 
share polymorphisms with the parental strains of the RIL population that the xQTL was mapped in. Moreover, WormQTL can be easily 
expanded to other Caenorhabditis species (21).

We believe that WormQTL, which will be continuously curated by the members of this international consortium, is a very 
attractive database for the growing community of quantitative genetics in worms researchers. We are committed to maintain data 
and software for the years to come and invite the community to add and share new data and ideas.

\chapter{Mapping correlation}

\emph{In this chapter we take one step further then in the previous chapters and develop a new methodology for
quantitative genetics called Correlated Traits Locus (CTL) mapping, a method complementairy to QTL mapping. 
Where QTL associates differences in mean, CTL, associate differences in correlation to genetic variation, i.e. 
CTL identify regions in the genome for which one genotype leads to correlated expression between a pair of 
traits, while the other genotype shows none (or significantly different) correlation.}

\null
\vfill

\begin{myexampleblock}{In press:}
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  \emph{CTL mapping}\\
  \bold{BMC Bioinformatics} (2013)
\end{myexampleblock}

\newpage

\section{What is a CTL?}
\lipsum[1]

\section{Combining CTL and QTL information}
\lipsum

\section{Examples of CTL mapping}
\lipsum
\subsection{CTL mapping in an \emph{A. thaliana} RIL population}

\subsection{CTL mapping using human GWA data}

\chapter{Thesis summary}
\lipsum[1-3]

\chapter{Additional for Dissertation}
\section*{Nederlandse Samenvatting / Dutch Summary}
\addcontentsline{toc}{section}{Nederlandse Samenvatting / Dutch Summary}
\lipsum[1]

\section*{Abbreviations and acronyms}
\addcontentsline{toc}{section}{Abbreviations and acronyms}
\begin{tabular}{ l l }
BC:          & Backcross \\
bp:          & Base pair(s) \\
cM:          & centi Morgan \\
CTL:         & Correlated traits locus \\
Mbp:         & Mega base pairs = 1.000.000 bp \\
RIL:         & Recombinant inbred line \\
QTL:         & Quantitative trait locus \\
GWA:         & Genome Wide Association \\
\end{tabular}

\newpage

\section*{Acknowledgements}
\addcontentsline{toc}{section}{Acknowledgements}
\lipsum[1]

\newpage

\section*{List of publications}
\addcontentsline{toc}{section}{List of publications}
\subsection*{Authored:}
  \authors{Danny Arends*, Pjotr Prins*, Ritsert C. Jansen and Karl W. Broman}\\
  R/qtl: high throughput Multiple QTL mapping\\
  \bold{Bioinformatics} (2010)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Leo Willems, Wilco Ligterink, Henk Hilhorst, Ritsert C. Jansen}\\
  Visualizing the genetic landscape of Arabidopsis seed performance\\
  \bold{Plant Physiology} (2011)\\\\
  \authors{Danny Arends*, K. Joeri van der Velde*, Pjotr Prins, Karl W. Broman, Steffen Moller, et al.}\\
  xQTL workbench: a scalable web environment for multi-level QTL analysis\\
  \bold{Bioinformatics} (2012)\\\\
  \authors{L. Basten Snoek*, Joeri Van der Velde*, Danny Arends*, Yang Li*, Antje Beyer, Mark Elvin, et al.}\\
  WormQTL: Public archive and analysis web portal for natural variation data in Caenorhabditis spp\\
  \bold{Nucleic Acids Research} (2012)\\\\
  \authors{Ronny V. L. Joosen*, Danny Arends*, Yang Li*, Leo Willems, Joost J.B. Keurentjes, Wilco Ligterink, Ritsert C. Jansen, Henk Hilhorst}\\
  Identifying genotype-by-environment interactions in the metabolism of germinating Arabidopsis seeds using Generalized Genetical Genomics\\
  \bold{Plant Physiology} (2013)

\subsection*{Co-Authored:}
  \authors{Morris A Swertz, Martijn Dijkstra, Tomasz Adamusiak, Danny Arends, et al.}\\
  The MOLGENIS toolkit: rapid prototyping of biosoftware at the push of a button\\
  \bold{BMC Bioinformatics} (2010)\\\\
  \authors{Morris A Swertz, K Joeri van der Velde, Bruno M Tesson, Danny Arends, et al.}\\
  XGAP: a uniform and extensible data model and software platform for genotype and phenotype experiments\\
  \bold{Genome Biology} (2010)\\\\
  \authors{Klaus Schughart, Danny Arends, P. Andreux, R. Balling, Pjotr Prins, et al.}\\
  SYSGENET: a meeting report from a new European network for systems genetics\\
  \bold{Mammalian Genome} (2010)\\\\
  \authors{Rudolf SN Fehrmann, Ritsert C. Jansen, Jan H. Veldink, Harm-Jan Westra, Danny Arends, et al.}\\
  Trans-eQTLs Reveal that Independent Genetic Variants Associated With a Complex Phenotype Converge on Intermediate Genes, with a Major Role for the HLA\\
  \bold{Plos Genetics} (2011)\\\\
  \authors{Caroline Durrant, Morris A. Swertz, Rudi Alberts, Danny Arends, Klaus Schughart, et al.}\\
  Bioinformatics tools and database resources for systems genetics analysis in mice - a short review and an evaluation of future needs\\
  \bold{Briefings in Bioinformatics} (2011)

\subsection*{In Press:}
  \authors{Danny Arends*, Konrad Zych*, K. Joeri van der Velde, Ronny V. L. Joosen, Wilco Ligterink and Ritsert C Jansen}\\
  \emph{Pheno2Geno - High throughput generation of genetic markers and maps from molecular phenotypes}\\
  \bold{BMC Bioinformatics} (2013)\\\\
  \authors{Danny Arends, Pjotr Prins, Yang Li, Lude Franke and Ritsert C. Jansen}\\
  CTL mapping\\
  \bold{BMC Bioinformatics} (2013)

\subsection*{Acknowledged in:}
  \authors{Yang Li, Morris A Swertz, Gonzalo Vera, Jingyuan Fu, Rainer Breitling and Ritsert C Jansen}\\
  DesignGG: an R-package and web tool for the optimal design of genetical genomics experiments\\
  \bold{BMC Bioinformatics}, 10:188 (2009)\\\\
  \authors{Yang Li, Rainer Breitling and Ritsert C. Jansen}\\
  Generalizing genetical genomics: getting added value from environmental perturbation\\
  \bold{Trends in Genetics}, 24:518-524 (2008)

\bibliographystyle{plain}
\addcontentsline{toc}{chapter}{Bibliography}
\bibliography{Thesis}

\end{document}

